{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "296024c6-eb00-48f9-8508-17952a7c431e",
   "metadata": {},
   "source": [
    "## Notebook to recapture knowledge from the KG about NPDIs and compare with NaPDI repo data (in vitro experiments)\n",
    "\n",
    "* Author: Sanya B Taneja\n",
    "* Created: 2022-04-06\n",
    "* Last edited: 2022-04-06\n",
    "\n",
    "PheKnowLator and MR graphs are combined in the original Path Search notebook and loaded here as the combined graph."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 208,
   "id": "2fd2e387-eb09-4c59-9812-249446b5a028",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import os.path\n",
    "import networkx as nx\n",
    "import json\n",
    "import urllib\n",
    "import traceback\n",
    "from itertools import islice\n",
    "from rdflib import Graph, URIRef, BNode, Namespace, Literal\n",
    "from rdflib.namespace import RDF, OWL\n",
    "from tqdm import tqdm\n",
    "import json"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 209,
   "id": "2538e004-6b5f-4c3c-8b7d-f86c1eb18a44",
   "metadata": {},
   "outputs": [],
   "source": [
    "import hashlib"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 210,
   "id": "0ff6ba47-d078-45b3-8993-7c6adcd8d5c3",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pickle\n",
    "import pandas as pd\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 211,
   "id": "4f1d65a4-5e39-430b-85a2-0e3263711736",
   "metadata": {},
   "outputs": [],
   "source": [
    "#import pheknowlator kg_utils \n",
    "import sys\n",
    "sys.path.append('../')\n",
    "from pkt_kg.utils import *"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 212,
   "id": "dd17c6ad-4e4d-4380-a9f4-c2049a2fefb7",
   "metadata": {},
   "outputs": [],
   "source": [
    "KG_PATH = '/home/sanya/PheKnowLatorv2/resources/knowledge_graphs/'\n",
    "MR_PATH = '/home/sanya/PheKnowLatorv2/machine_read/output_graphs/'\n",
    "KG_PATH_MERGED = KG_PATH + 'merged_graphs/'\n",
    "NodeLabelsFile = KG_PATH_MERGED + 'nodeLabels_merged_20220406.pickle'\n",
    "KG_NAME_MERGED = 'PheKnowLator_machine_read_merged_instance_based_OWLNETS_20220406.gpickle'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 213,
   "id": "4e8ebc69-cc46-474b-b182-97116c35076b",
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(NodeLabelsFile, 'rb') as filep:\n",
    "    nodeLabels = pickle.load(filep)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 214,
   "id": "151a8d33-f0d7-4a31-9d48-98dfa1b3b61b",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_graph_stats(kg):\n",
    "    nodes = nx.number_of_nodes(kg)\n",
    "    edges = nx.number_of_edges(kg)\n",
    "    self_loops = nx.number_of_selfloops(kg)\n",
    "\n",
    "    print('There are {} nodes, {} edges, and {} self-loop(s)'.format(nodes, edges, self_loops))\n",
    "    # get degree information\n",
    "    avg_degree = float(edges) / nodes\n",
    "    print('The Average Degree is {}'.format(avg_degree))\n",
    "    \n",
    "    print('Nodes with highest degree:')\n",
    "    n_deg = sorted([(str(x[0]), x[1]) for x in  kg.degree], key=lambda x: x[1], reverse=1)[:6]\n",
    "\n",
    "    for x in n_deg:\n",
    "        print('Label: {}'.format(nodeLabels[x[0]]))\n",
    "        print('{} (degree={})'.format(x[0], x[1]))\n",
    "    # get network density\n",
    "    density = nx.density(kg)\n",
    "\n",
    "    print('The density of the graph is: {}'.format(density))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "ceda6e39-4f4a-4062-8aaa-7e254f0c99d5",
   "metadata": {},
   "outputs": [],
   "source": [
    "##READ MERGED GRAPH\n",
    "nx_graph = nx.read_gpickle(KG_PATH_MERGED+KG_NAME_MERGED)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 215,
   "id": "e237570c-0cf3-432a-81a8-983d200a0970",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "There are 745763 nodes, 7251937 edges, and 562 self-loop(s)\n",
      "The Average Degree is 9.72418449292872\n",
      "Nodes with highest degree:\n",
      "Label: transcript\n",
      "http://purl.obolibrary.org/obo/SO_0000673 (degree=190858)\n",
      "Label: SNV\n",
      "http://purl.obolibrary.org/obo/SO_0001483 (degree=121020)\n",
      "Label: Homo sapiens\n",
      "http://purl.obolibrary.org/obo/NCBITaxon_9606 (degree=116704)\n",
      "Label: protein_coding_gene\n",
      "http://purl.obolibrary.org/obo/SO_0001217 (degree=105046)\n",
      "Label: testis\n",
      "http://purl.obolibrary.org/obo/UBERON_0000473 (degree=43818)\n",
      "Label: lncRNA_with_retained_intron\n",
      "http://purl.obolibrary.org/obo/SO_0002113 (degree=29340)\n",
      "The density of the graph is: 1.3039259834811533e-05\n"
     ]
    }
   ],
   "source": [
    "get_graph_stats(nx_graph)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "029a7aa6-aa2d-44fa-a316-e26be747885f",
   "metadata": {},
   "outputs": [],
   "source": [
    "DIR_OUT = '/home/sanya/PheKnowLatorv2/output_files/'\n",
    "obo = Namespace('http://purl.obolibrary.org/obo/')\n",
    "napdi = Namespace('http://napdi.org/napdi_srs_imports:')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "cc597d91-1580-41c8-a768-88e74523b679",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'(-)-epigallocatechin 3-gallate'"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nodeLabels['http://purl.obolibrary.org/obo/CHEBI_4806']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "c2da4085-d5df-43b1-bf60-ea233aac7b26",
   "metadata": {},
   "outputs": [],
   "source": [
    "##Path between midazolam (CHEBI_) and 1'-hydroxymidazolam (CHEBI_145330)\n",
    "##Link to in vitro inhibition of mitragynine and midazolam - interaction occurs in intestine - pathway?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 216,
   "id": "b3e855a7-a9f7-4d2b-a2aa-48495eb3a35a",
   "metadata": {},
   "outputs": [],
   "source": [
    "import KG_path_searches"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 165,
   "id": "3fb0faaf-85ca-4431-a416-8ed16a3f6963",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "40"
      ]
     },
     "execution_count": 165,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "##read in pickle dictionary with obo identifiers\n",
    "import pickle\n",
    "with open(KG_PATH+'strToOBOdict.pickle','rb') as filep:\n",
    "    obodict = pickle.load(filep)\n",
    "len(obodict)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "22bfb0fa-8086-49a3-b1ad-a89a5bcf2722",
   "metadata": {},
   "outputs": [],
   "source": [
    "##Kratom with -- We present a case of a patient taking 150 mg venlafaxine (CYP2D6/3A substrate), 300 mg quetiapine (CYP3A substrate), \n",
    "#and a high amount of kratom (∼90 g) daily. The patient presented to the emergency department with serotonin syndrome and \n",
    "#corrected electrocardiogram abnormalities that may have been secondary to supratherapeutic exposure to venlafaxine and/or quetiapine."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "ce9cdb04-604d-4c55-b117-dd429b124454",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'raloxifene'"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "##Green tea and nadolol\n",
    "##Green tea and raloxifene\n",
    "\n",
    "raloxifene = obo.CHEBI_8772\n",
    "nodeLabels['http://purl.obolibrary.org/obo/CHEBI_8772']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "bd8f801c-0457-44a3-baa1-bca557c9a040",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{rdflib.term.URIRef('http://purl.obolibrary.org/obo/RO_0000087'): {'predicate_key': '6d3c663b6832d25f9f902311a4a222d1',\n",
       "  'weight': 0.0}}"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nx_graph.get_edge_data(raloxifene, obo.CHEBI_50837)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "1974c4b3-6343-4259-b317-0f4e51e5f2c9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'predicate_key': '6c12966e2e06c036d23c9289a8499790',\n",
       " 'weight': 0.0,\n",
       " 'pmid': '18371987',\n",
       " 'timestamp': '2008 Apr 23',\n",
       " 'source_graph': 'machine_read',\n",
       " 'belief': 0.65}"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nx_graph.get_edge_data(raloxifene, obo.PR_P78381)[URIRef('http://purl.obolibrary.org/obo/RO_0002449')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "7686ebae-5495-412a-b52c-861f2c02bcd1",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_neighbors(nx_graph, node):\n",
    "    neighs = nx.neighbors(nx_graph, node)\n",
    "    neighs_list = []\n",
    "    for item in neighs:\n",
    "        nodestr = nodeLabels[str(node)]\n",
    "        neighbor = nodeLabels[str(item)]\n",
    "        edge_result = nx_graph.get_edge_data(node, item)\n",
    "        edge_keys = list(edge_result.keys())\n",
    "        for edge in edge_keys:\n",
    "            edgestr = nodeLabels[str(edge)]\n",
    "            edgevalue = edge_result[edge]\n",
    "            result = [nodestr, edgestr, neighbor]\n",
    "            if 'source_graph' in edgevalue:\n",
    "                result.append(edgevalue['source_graph'])\n",
    "            if 'timestamp' in edgevalue:\n",
    "                result.append(edgevalue['timestamp'])\n",
    "            neighs_list.append(result)\n",
    "    return neighs_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "8ff88302-98ad-4a33-9c28-48857f383296",
   "metadata": {},
   "outputs": [],
   "source": [
    "inhibit_preds = [obo.RO_0002449, obo.RO_0002212, obo.RO_0002599, obo.RO_0011010]\n",
    "active_preds = [obo.RO_0002448, obo.RO_0002213, obo.RO_0011002, obo.RO_0002596, obo.RO_0002598, obo.RO_0011009, obo.RO_0002566]\n",
    "interact_preds = [obo.DIDEO_00000041, obo.RO_0002434, obo.RO_0002436, obo.RO_0002020, obo.RO_0003000, obo.RO_0000085, \n",
    "                  URIRef('http://purl.obolibrary.org/obo/chebi#has_functional_parent')]\n",
    "protein_preds = []"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 193,
   "id": "69a540cc-0544-4baa-9709-1cfa8a166bb2",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_neighbors_filtered(nx_graph, node):\n",
    "    neighs = nx.neighbors(nx_graph, node)\n",
    "    neighs_list = []\n",
    "    nodestr = ''\n",
    "    neighbor = ''\n",
    "    for item in neighs:\n",
    "        if str(node) in nodeLabels:\n",
    "            nodestr = nodeLabels[str(node)]\n",
    "        if str(item) in nodeLabels:\n",
    "            neighbor = nodeLabels[str(item)]\n",
    "        edge_result = nx_graph.get_edge_data(node, item)\n",
    "        edge_keys = list(edge_result.keys())\n",
    "        for edge in edge_keys:\n",
    "            if edge in inhibit_preds or edge in active_preds or edge in interact_preds:\n",
    "                edgestr = nodeLabels[str(edge)]\n",
    "                edgevalue = edge_result[edge]\n",
    "                result = [nodestr, edgestr, neighbor]\n",
    "                if 'source_graph' in edgevalue:\n",
    "                    result.append(edgevalue['source_graph'])\n",
    "                if 'timestamp' in edgevalue:\n",
    "                    result.append(edgevalue['timestamp'])\n",
    "                neighs_list.append(result)\n",
    "    return neighs_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "c05c8be4-043d-4b7b-a6ed-40346e28cff4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[rdflib.term.URIRef('http://purl.obolibrary.org/obo/RO_0011009'),\n",
       " rdflib.term.URIRef('http://purl.obolibrary.org/obo/RO_0002449'),\n",
       " rdflib.term.URIRef('http://purl.obolibrary.org/obo/RO_0002448'),\n",
       " rdflib.term.URIRef('http://purl.obolibrary.org/obo/RO_0002434')]"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "list(nx_graph.get_edge_data(obodict['mitragynine'][0], obodict['CYP3A4'][0]).keys())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "e80ac55f-5bcf-49a8-9f94-9bc94a00db5d",
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'raloxifene' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-20-8439a71c3dd2>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mget_neighbors\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mnx_graph\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mraloxifene\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m: name 'raloxifene' is not defined"
     ]
    }
   ],
   "source": [
    "get_neighbors(nx_graph, raloxifene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "548c58ad-fecd-4a66-b96d-339cab0fe947",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 1796 entries, 0 to 1795\n",
      "Data columns (total 14 columns):\n",
      " #   Column         Non-Null Count  Dtype  \n",
      "---  ------         --------------  -----  \n",
      " 0   pmid           1796 non-null   int64  \n",
      " 1   subject_cui    1755 non-null   object \n",
      " 2   subject_type   1796 non-null   object \n",
      " 3   object_cui     1719 non-null   object \n",
      " 4   object_type    1796 non-null   object \n",
      " 5   year           1790 non-null   float64\n",
      " 6   subject_name   1755 non-null   object \n",
      " 7   predicate      1796 non-null   object \n",
      " 8   object_name    1719 non-null   object \n",
      " 9   source_text    1796 non-null   object \n",
      " 10  Unnamed: 10    0 non-null      float64\n",
      " 11  predicate_obo  0 non-null      float64\n",
      " 12  subject_obo    1194 non-null   object \n",
      " 13  object_obo     831 non-null    object \n",
      "dtypes: float64(3), int64(1), object(10)\n",
      "memory usage: 196.6+ KB\n"
     ]
    }
   ],
   "source": [
    "##read in negated predications\n",
    "df_neg = pd.read_csv('../machine_read/semrep_data/semrep_negation_predications_mapped_20220407.csv')\n",
    "df_neg.info()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f090727d-a583-475d-bb28-6e190efe380d",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "## Kratom - Mechanistic Exploration Part 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "956a6e6a-31c5-4bef-8400-33e9b843e044",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "b3005f27-2630-4aa2-ac6f-bafb4a196413",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "AdjacencyView({rdflib.term.URIRef('http://purl.obolibrary.org/obo/PO_0025131'): {rdflib.term.URIRef('http://www.w3.org/2000/01/rdf-schema#subClassOf'): {'predicate_key': '958521878a132c4bfad53e7ed70ac7f3', 'weight': 0.0}}, rdflib.term.URIRef('http://purl.obolibrary.org/obo/CHEBI_6956'): {rdflib.term.URIRef('http://purl.obolibrary.org/obo/RO_0002180'): {'predicate_key': '92919f4d7a39d7db676c369dd45a1c1c', 'weight': 0.0}}, rdflib.term.URIRef('http://napdi.org/napdi_srs_imports:mitragyna_speciosa_whole'): {rdflib.term.URIRef('http://purl.obolibrary.org/obo/BFO_0000050'): {'predicate_key': '2aaba9f764274816c387a21ce0acb793', 'weight': 0.0}}, rdflib.term.URIRef('http://napdi.org/napdi_srs_imports:7_hydroxy_mitragynine'): {rdflib.term.URIRef('http://purl.obolibrary.org/obo/RO_0002180'): {'predicate_key': '3ebb29585d901f255e32b08d4a6a61a2', 'weight': 0.0}}})"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nx_graph.__getitem__(obodict['kratom'][0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "6d7ea4c2-2589-490d-b18b-2df4877a3eab",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "121"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mit_neighbors = get_neighbors(nx_graph, obodict['mitragynine'][0])\n",
    "len(mit_neighbors)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "3805c67d-901e-439f-a787-d952001fa684",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "98"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mit_neighbors_filt = get_neighbors_filtered(nx_graph, obodict['mitragynine'][0])\n",
    "len(mit_neighbors_filt)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 166,
   "id": "d94f2948-b069-4068-88b2-8abb6760b0ab",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[rdflib.term.URIRef('http://purl.obolibrary.org/obo/PR_P000001891'),\n",
       " rdflib.term.URIRef('http://purl.obolibrary.org/obo/PR_P08183'),\n",
       " rdflib.term.URIRef('http://purl.obolibrary.org/obo/GO_0008559'),\n",
       " rdflib.term.URIRef('http://purl.obolibrary.org/obo/PR_000012619')]"
      ]
     },
     "execution_count": 166,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "obodict['PGP']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "de79ddc0-8d13-417a-88ac-23f48ee916f0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'entity_type': 'RELATIONS', 'label': 'IncreaseAmount'}\n",
      "{'predicate_key': '04792bbeb7346991d4206904a9cd8164', 'weight': 0.0, 'pmid': '23274770', 'timestamp': '2013 Mar', 'source_graph': 'machine_read', 'belief': 0.65}\n",
      "directly negatively regulates activity of\n",
      "{'predicate_key': '35db33b594d5b68623d10e9a02d4c781', 'weight': 0.0, 'pmid': '24174816', 'timestamp': '2013 Oct', 'source_graph': 'machine_read', 'belief': 0.86}\n",
      "{'entity_type': 'RELATIONS', 'label': 'Activation'}\n",
      "{'predicate_key': '53929eb7232ad497ae25cf0f8b3f31b7', 'weight': 0.0, 'pmid': '23274770', 'timestamp': '2013 Mar', 'source_graph': 'machine_read', 'belief': 0.65}\n",
      "interacts with\n",
      "{'predicate_key': 'd7cc44c1009eda758102414ff4eb1e6a', 'weight': 0.0}\n"
     ]
    }
   ],
   "source": [
    "edges = nx_graph.get_edge_data(obodict['mitragynine'][0], obodict['CYP3A4'][0])\n",
    "if edges:\n",
    "    for item in edges:\n",
    "        print(nodeLabels[str(item)])\n",
    "        print(edges[item])\n",
    "edges = nx_graph.get_edge_data(obodict['mitragynine'][0], obodict['CYP3A4'][1])\n",
    "if edges:\n",
    "    for item in edges:\n",
    "        print(nodeLabels[str(item)])\n",
    "        print(edges[item])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "ff3525ec-7c03-4dde-a6c1-7a1a566d24be",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "directly negatively regulates activity of\n",
      "{'predicate_key': '95f12736d1211905a358aef7ffba851b', 'weight': 0.0, 'pmid': '29071751', 'timestamp': '2017.0', 'source_graph': 'machine_read', 'belief': 0.8}\n",
      "interacts with\n",
      "{'predicate_key': '21e30bf3c3f213311b2c497b8b198e06', 'weight': 0.0}\n",
      "directly negatively regulates activity of\n",
      "{'predicate_key': '788f0f936532db4f71a8a647692cab3e', 'weight': 0.0, 'pmid': '24174816', 'timestamp': '2013.0', 'source_graph': 'machine_read', 'belief': 0.8}\n",
      "positively regulates\n",
      "{'predicate_key': '40e4364894f99e7cb08fd0b20b6ace3f', 'weight': 0.0, 'pmid': '23274770', 'timestamp': '2013.0', 'source_graph': 'machine_read', 'belief': 0.8}\n",
      "interacts with\n",
      "{'predicate_key': '64c44bee7a60dc54ada041615f0b0371', 'weight': 0.0, 'pmid': '30547698', 'timestamp': '2019.0', 'source_graph': 'machine_read_INF', 'belief': 0.8}\n"
     ]
    }
   ],
   "source": [
    "edges = nx_graph.get_edge_data(obodict['mitragynine'][0], obodict['CYP2D6'][0])\n",
    "if edges:\n",
    "    for item in edges:\n",
    "        print(nodeLabels[str(item)])\n",
    "        print(edges[item])\n",
    "edges = nx_graph.get_edge_data(obodict['mitragynine'][0], obodict['CYP2D6'][1])\n",
    "if edges:\n",
    "    for item in edges:\n",
    "        print(nodeLabels[str(item)])\n",
    "        print(edges[item])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "f3ffb8e0-a691-4cb6-bdb7-fe7289e9ea07",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "interacts with\n",
      "{'predicate_key': 'a2f1439cda4a0a2da0b67705d23facce', 'weight': 0.0}\n"
     ]
    }
   ],
   "source": [
    "edges = nx_graph.get_edge_data(obodict['mitragynine'][0], obodict['CYP2C19'][0])\n",
    "if edges:\n",
    "    for item in edges:\n",
    "        print(nodeLabels[str(item)])\n",
    "        print(edges[item])\n",
    "edges = nx_graph.get_edge_data(obodict['mitragynine'][0], obodict['CYP2C19'][1])\n",
    "if edges:\n",
    "    for item in edges:\n",
    "        print(nodeLabels[str(item)])\n",
    "        print(edges[item])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "cf7d1fa9-955c-4d90-be2a-8d0204e3a2a6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "directly negatively regulates activity of\n",
      "{'predicate_key': '5044e2ac537ebe9f92096235427026f3', 'weight': 0.0, 'pmid': '24174816', 'timestamp': '2013 Oct', 'source_graph': 'machine_read', 'belief': 0.86}\n",
      "directly negatively regulates activity of\n",
      "{'predicate_key': 'e593974d2818ab169fbbe255d52646fd', 'weight': 0.0, 'pmid': '24174816', 'timestamp': '2013.0', 'source_graph': 'machine_read', 'belief': 0.8}\n"
     ]
    }
   ],
   "source": [
    "edges = nx_graph.get_edge_data(obodict['mitragynine'][0], obodict['CYP2C9'][0])\n",
    "if edges:\n",
    "    for item in edges:\n",
    "        print(nodeLabels[str(item)])\n",
    "        print(edges[item])\n",
    "edges = nx_graph.get_edge_data(obodict['mitragynine'][0], obodict['CYP2C9'][1])\n",
    "if edges:\n",
    "    for item in edges:\n",
    "        print(nodeLabels[str(item)])\n",
    "        print(edges[item])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "387e0b99-870c-4c8d-b15e-e1d85a1ff823",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'entity_type': 'RELATIONS', 'label': 'IncreaseAmount'}\n",
      "{'predicate_key': 'd7351c8523bd7cbf34d43b6a412a0bbe', 'weight': 0.0, 'pmid': '23274770', 'timestamp': '2013 Mar', 'source_graph': 'machine_read', 'belief': 0.86}\n",
      "{'entity_type': 'RELATIONS', 'label': 'Activation'}\n",
      "{'predicate_key': '77c46a33e2f9a70ee383dfd247785a23', 'weight': 0.0, 'pmid': '29071751', 'timestamp': '2017 Dec', 'source_graph': 'machine_read', 'belief': 0.86}\n",
      "interacts with\n",
      "{'predicate_key': '0553e1d0f9b0b6920c61bbcbc1c04db8', 'weight': 0.0}\n",
      "interacts with\n",
      "{'predicate_key': '572622b80f7e7965769037a2c42ec144', 'weight': 0.0, 'pmid': '23274770', 'timestamp': '2013.0', 'source_graph': 'machine_read_INF', 'belief': 0.8}\n",
      "positively regulates\n",
      "{'predicate_key': '1b698e5a28f61543a1014e3adf0998c2', 'weight': 0.0, 'pmid': '23274770', 'timestamp': '2013.0', 'source_graph': 'machine_read_INF', 'belief': 0.8}\n"
     ]
    }
   ],
   "source": [
    "edges = nx_graph.get_edge_data(obodict['mitragynine'][0], obodict['CYP1A2'][0])\n",
    "if edges:\n",
    "    for item in edges:\n",
    "        print(nodeLabels[str(item)])\n",
    "        print(edges[item])\n",
    "edges = nx_graph.get_edge_data(obodict['mitragynine'][0], obodict['CYP1A2'][1])\n",
    "if edges:\n",
    "    for item in edges:\n",
    "        print(nodeLabels[str(item)])\n",
    "        print(edges[item])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "4d43863e-1804-44bf-be17-298801dda1b5",
   "metadata": {},
   "outputs": [],
   "source": [
    "edges = nx_graph.get_edge_data(obodict['mitragynine'][0], obodict['CYP2C8'][0])\n",
    "if edges:\n",
    "    for item in edges:\n",
    "        print(nodeLabels[str(item)])\n",
    "        print(edges[item])\n",
    "edges = nx_graph.get_edge_data(obodict['mitragynine'][0], obodict['CYP2C8'][1])\n",
    "if edges:\n",
    "    for item in edges:\n",
    "        print(nodeLabels[str(item)])\n",
    "        print(edges[item])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "33903ff9-077b-48c0-8390-04d32487b864",
   "metadata": {},
   "outputs": [],
   "source": [
    "edges = nx_graph.get_edge_data(obodict['mitragynine'][0], obodict['CYP3A5'][0])\n",
    "if edges:\n",
    "    for item in edges:\n",
    "        print(nodeLabels[str(item)])\n",
    "        print(edges[item])\n",
    "edges = nx_graph.get_edge_data(obodict['mitragynine'][0], obodict['CYP3A5'][1])\n",
    "if edges:\n",
    "    for item in edges:\n",
    "        print(nodeLabels[str(item)])\n",
    "        print(edges[item])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 129,
   "id": "933489fb-ad4a-4fd1-a38b-d33b9110bc8c",
   "metadata": {},
   "outputs": [],
   "source": [
    "edges = nx_graph.get_edge_data(obodict['mitragynine'][0], obodict['UGT'][0])\n",
    "if edges:\n",
    "    for item in edges:\n",
    "        print(nodeLabels[str(item)])\n",
    "        print(edges[item])\n",
    "edges = nx_graph.get_edge_data(obodict['mitragynine'][0], obodict['UGT'][1])\n",
    "if edges:\n",
    "    for item in edges:\n",
    "        print(nodeLabels[str(item)])\n",
    "        print(edges[item])\n",
    "edges = nx_graph.get_edge_data(obodict['mitragynine'][0], obodict['UGT'][2])\n",
    "if edges:\n",
    "    for item in edges:\n",
    "        print(nodeLabels[str(item)])\n",
    "        print(edges[item])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "ff601fd8-29d7-4dc5-adfa-ccd23df9834f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "directly negatively regulates activity of\n",
      "{'predicate_key': '77b223d7adda942f0f53c2d6c25717cb', 'weight': 0.0, 'pmid': '26692748', 'timestamp': '2014.0', 'source_graph': 'machine_read', 'belief': 0.8}\n"
     ]
    }
   ],
   "source": [
    "edges = nx_graph.get_edge_data(obodict['mitragynine'][0], obodict['UGT1A1'][0])\n",
    "if edges:\n",
    "    for item in edges:\n",
    "        print(nodeLabels[str(item)])\n",
    "        print(edges[item])\n",
    "edges = nx_graph.get_edge_data(obodict['mitragynine'][0], obodict['UGT1A1'][1])\n",
    "if edges:\n",
    "    for item in edges:\n",
    "        print(nodeLabels[str(item)])\n",
    "        print(edges[item])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "dc2418ed-2b29-447f-a989-6a38ee32a319",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "directly negatively regulates activity of\n",
      "{'predicate_key': '6825c6627114a1d163b63d7f024d0289', 'weight': 0.0, 'pmid': '26692748', 'timestamp': '2014.0', 'source_graph': 'machine_read', 'belief': 0.8}\n"
     ]
    }
   ],
   "source": [
    "edges = nx_graph.get_edge_data(obodict['mitragynine'][0], obodict['UGT2B7'][0])\n",
    "if edges:\n",
    "    for item in edges:\n",
    "        print(nodeLabels[str(item)])\n",
    "        print(edges[item])\n",
    "edges = nx_graph.get_edge_data(obodict['mitragynine'][0], obodict['UGT2B7'][1])\n",
    "if edges:\n",
    "    for item in edges:\n",
    "        print(nodeLabels[str(item)])\n",
    "        print(edges[item])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 167,
   "id": "fe07ad04-ed9f-4c39-83f1-22d76b9b46ac",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'entity_type': 'RELATIONS', 'label': 'DecreaseAmount'}\n",
      "{'predicate_key': '7b16891a8684bc49ca03f724b93f2266', 'weight': 0.0, 'pmid': '30604191', 'timestamp': '2019 Apr', 'source_graph': 'machine_read', 'belief': 0.9419}\n",
      "directly negatively regulates activity of\n",
      "{'predicate_key': '1d2db53249d0e93cedb141dc363e65bb', 'weight': 0.0, 'pmid': '30604191', 'timestamp': '2019 Apr', 'source_graph': 'machine_read', 'belief': 0.65}\n",
      "directly negatively regulates activity of\n",
      "{'predicate_key': '3cd523ea48bad379fe3031cf46d9747f', 'weight': 0.0, 'pmid': '34003732', 'timestamp': '2021.0', 'source_graph': 'machine_read', 'belief': 0.8}\n"
     ]
    }
   ],
   "source": [
    "edges = nx_graph.get_edge_data(obodict['mitragynine'][0], obodict['PGP'][0])\n",
    "if edges:\n",
    "    for item in edges:\n",
    "        print(nodeLabels[str(item)])\n",
    "        print(edges[item])\n",
    "edges = nx_graph.get_edge_data(obodict['mitragynine'][0], obodict['PGP'][1])\n",
    "if edges:\n",
    "    for item in edges:\n",
    "        print(nodeLabels[str(item)])\n",
    "        print(edges[item])\n",
    "edges = nx_graph.get_edge_data(obodict['mitragynine'][0], obodict['PGP'][2])\n",
    "if edges:\n",
    "    for item in edges:\n",
    "        print(nodeLabels[str(item)])\n",
    "        print(edges[item])\n",
    "edges = nx_graph.get_edge_data(obodict['mitragynine'][0], obodict['PGP'][3])\n",
    "if edges:\n",
    "    for item in edges:\n",
    "        print(nodeLabels[str(item)])\n",
    "        print(edges[item])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 153,
   "id": "1024d812-422f-4ab6-8669-1770413fbec0",
   "metadata": {},
   "outputs": [],
   "source": [
    "edges = nx_graph.get_edge_data(obodict['mitragynine'][0], obodict['glycoprotein'][0])\n",
    "if edges:\n",
    "    for item in edges:\n",
    "        print(nodeLabels[str(item)])\n",
    "        print(edges[item])\n",
    "edges = nx_graph.get_edge_data(obodict['mitragynine'][0], obodict['glycoprotein'][1])\n",
    "if edges:\n",
    "    for item in edges:\n",
    "        print(nodeLabels[str(item)])\n",
    "        print(edges[item])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "92d461bb-83b1-4050-8cec-8e5f994c5090",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[['7-hydroxy-mitragynine', 'subClassOf', 'chemical entity'],\n",
       " ['7-hydroxy-mitragynine', 'has role', 'metabolite'],\n",
       " ['7-hydroxy-mitragynine', 'has functional parent', 'Mitragynine']]"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "get_neighbors(nx_graph, obodict['hydroxy_mitragynine'][0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "84656b45-7061-46d9-b646-58a0711d4626",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[[{'entity_type': 'NODES', 'label': '7-hydroxymitragynine', 'cui': 'C1435927'},\n",
       "  'directly negatively regulates activity of',\n",
       "  '4-methylumbelliferone beta-D-glucuronide',\n",
       "  'machine_read',\n",
       "  '2014.0'],\n",
       " [{'entity_type': 'NODES', 'label': '7-hydroxymitragynine', 'cui': 'C1435927'},\n",
       "  'interacts with',\n",
       "  '4-methylumbelliferone beta-D-glucuronide',\n",
       "  'machine_read_INF',\n",
       "  '2014.0'],\n",
       " [{'entity_type': 'NODES', 'label': '7-hydroxymitragynine', 'cui': 'C1435927'},\n",
       "  'existence overlaps',\n",
       "  'methanol',\n",
       "  'machine_read',\n",
       "  '2014.0'],\n",
       " [{'entity_type': 'NODES', 'label': '7-hydroxymitragynine', 'cui': 'C1435927'},\n",
       "  'directly negatively regulates activity of',\n",
       "  'Mitragynine',\n",
       "  'machine_read',\n",
       "  '2014.0'],\n",
       " [{'entity_type': 'NODES', 'label': '7-hydroxymitragynine', 'cui': 'C1435927'},\n",
       "  'directly negatively regulates activity of',\n",
       "  'UDP-glucuronosyltransferase 2B7',\n",
       "  'machine_read',\n",
       "  '2014.0'],\n",
       " [{'entity_type': 'NODES', 'label': '7-hydroxymitragynine', 'cui': 'C1435927'},\n",
       "  'existence overlaps',\n",
       "  'dimethyl sulfoxide',\n",
       "  'machine_read',\n",
       "  '2014.0'],\n",
       " [{'entity_type': 'NODES', 'label': '7-hydroxymitragynine', 'cui': 'C1435927'},\n",
       "  'existence overlaps',\n",
       "  'pre-mRNA-processing factor 6',\n",
       "  'machine_read',\n",
       "  '2019.0']]"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "get_neighbors(nx_graph, obodict['hydroxy_mitragynine'][1])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 131,
   "id": "841812b9-cd2f-4404-8177-455d50c61696",
   "metadata": {},
   "outputs": [],
   "source": [
    "edges = nx_graph.get_edge_data(obodict['hydroxy_mitragynine'][1], obodict['UGT'][0])\n",
    "if edges:\n",
    "    for item in edges:\n",
    "        print(nodeLabels[str(item)])\n",
    "        print(edges[item])\n",
    "edges = nx_graph.get_edge_data(obodict['hydroxy_mitragynine'][1], obodict['UGT'][2])\n",
    "if edges:\n",
    "    for item in edges:\n",
    "        print(nodeLabels[str(item)])\n",
    "        print(edges[item])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "a036defe-62bb-42db-a10b-5a5a828fc8d8",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "has functional parent\n",
      "{'predicate_key': '500db2060a025b3ab27fcc2602acb6c2', 'weight': 0.0}\n",
      "directly negatively regulates activity of\n",
      "{'predicate_key': 'e8de6bae777ac5a1078eb563887b72a8', 'weight': 0.0, 'pmid': '26692748', 'timestamp': '2014.0', 'source_graph': 'machine_read', 'belief': 0.8}\n"
     ]
    }
   ],
   "source": [
    "edges = nx_graph.get_edge_data(obodict['hydroxy_mitragynine'][0], obodict['mitragynine'][0])\n",
    "if edges:\n",
    "    for item in edges:\n",
    "        print(nodeLabels[str(item)])\n",
    "        print(edges[item])\n",
    "edges = nx_graph.get_edge_data(obodict['hydroxy_mitragynine'][1], obodict['mitragynine'][0])\n",
    "if edges:\n",
    "    for item in edges:\n",
    "        print(nodeLabels[str(item)])\n",
    "        print(edges[item])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "30431d86-7971-4ad3-a725-ed970e6378a6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'metabolite'"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nodeLabels['http://purl.obolibrary.org/obo/CHEBI_25212']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "6a8e2550-07c4-4466-abd5-f0068c007b95",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pmid</th>\n",
       "      <th>subject_cui</th>\n",
       "      <th>subject_type</th>\n",
       "      <th>object_cui</th>\n",
       "      <th>object_type</th>\n",
       "      <th>year</th>\n",
       "      <th>subject_name</th>\n",
       "      <th>predicate</th>\n",
       "      <th>object_name</th>\n",
       "      <th>source_text</th>\n",
       "      <th>Unnamed: 10</th>\n",
       "      <th>predicate_obo</th>\n",
       "      <th>subject_obo</th>\n",
       "      <th>object_obo</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>19353999</td>\n",
       "      <td>C0059438</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>C1332826</td>\n",
       "      <td>gngm,aapp</td>\n",
       "      <td>2009.0</td>\n",
       "      <td>epigallocatechin gallate</td>\n",
       "      <td>neg_interacts_with</td>\n",
       "      <td>CYP1A2 gene</td>\n",
       "      <td>19353999_ascii.txt.tx.1 A recent report has fo...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>napdi_srs_imports:epigallocatechin_gallate</td>\n",
       "      <td>PR_000006102</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>12659723</td>\n",
       "      <td>C0007404</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>C1420188</td>\n",
       "      <td>gngm,aapp</td>\n",
       "      <td>2003.0</td>\n",
       "      <td>Catechin</td>\n",
       "      <td>neg_interacts_with</td>\n",
       "      <td>SLC35A2 gene</td>\n",
       "      <td>12659723_ascii.txt.tx.2 Lhoste et al./ Food an...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CHEBI_23053</td>\n",
       "      <td>PR_000015090</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>12659723</td>\n",
       "      <td>C0022203</td>\n",
       "      <td>npop</td>\n",
       "      <td>C0011135</td>\n",
       "      <td>orgf</td>\n",
       "      <td>2003.0</td>\n",
       "      <td>Isomerism (chemical phenomenon)</td>\n",
       "      <td>neg_affects</td>\n",
       "      <td>Defecation</td>\n",
       "      <td>12659723_ascii.txt.tx.2 Other polyphenols, e.g...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>32645157</td>\n",
       "      <td>C0041912</td>\n",
       "      <td>dsyn</td>\n",
       "      <td>C1154978</td>\n",
       "      <td>orgf</td>\n",
       "      <td>2020.0</td>\n",
       "      <td>Upper Respiratory Infections</td>\n",
       "      <td>neg_affects</td>\n",
       "      <td>cold acclimation</td>\n",
       "      <td>32645157_ascii.txt.tx.1 After undergoing cold ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MONDO_0024355</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>32645157</td>\n",
       "      <td>C1167041</td>\n",
       "      <td>celc</td>\n",
       "      <td>C0120285</td>\n",
       "      <td>aapp,gngm,irda</td>\n",
       "      <td>2020.0</td>\n",
       "      <td>vacuolar membrane</td>\n",
       "      <td>neg_location_of</td>\n",
       "      <td>Green Fluorescent Proteins</td>\n",
       "      <td>32645157_ascii.txt.tx.1 Our results showed tha...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       pmid subject_cui subject_type object_cui     object_type    year  \\\n",
       "0  19353999    C0059438    orch,phsu   C1332826       gngm,aapp  2009.0   \n",
       "1  12659723    C0007404    orch,phsu   C1420188       gngm,aapp  2003.0   \n",
       "2  12659723    C0022203         npop   C0011135            orgf  2003.0   \n",
       "3  32645157    C0041912         dsyn   C1154978            orgf  2020.0   \n",
       "4  32645157    C1167041         celc   C0120285  aapp,gngm,irda  2020.0   \n",
       "\n",
       "                      subject_name           predicate  \\\n",
       "0         epigallocatechin gallate  neg_interacts_with   \n",
       "1                         Catechin  neg_interacts_with   \n",
       "2  Isomerism (chemical phenomenon)         neg_affects   \n",
       "3     Upper Respiratory Infections         neg_affects   \n",
       "4                vacuolar membrane     neg_location_of   \n",
       "\n",
       "                  object_name  \\\n",
       "0                 CYP1A2 gene   \n",
       "1                SLC35A2 gene   \n",
       "2                  Defecation   \n",
       "3            cold acclimation   \n",
       "4  Green Fluorescent Proteins   \n",
       "\n",
       "                                         source_text  Unnamed: 10  \\\n",
       "0  19353999_ascii.txt.tx.1 A recent report has fo...          NaN   \n",
       "1  12659723_ascii.txt.tx.2 Lhoste et al./ Food an...          NaN   \n",
       "2  12659723_ascii.txt.tx.2 Other polyphenols, e.g...          NaN   \n",
       "3  32645157_ascii.txt.tx.1 After undergoing cold ...          NaN   \n",
       "4  32645157_ascii.txt.tx.1 Our results showed tha...          NaN   \n",
       "\n",
       "   predicate_obo                                 subject_obo    object_obo  \n",
       "0            NaN  napdi_srs_imports:epigallocatechin_gallate  PR_000006102  \n",
       "1            NaN                                 CHEBI_23053  PR_000015090  \n",
       "2            NaN                                         NaN           NaN  \n",
       "3            NaN                               MONDO_0024355           NaN  \n",
       "4            NaN                                         NaN           NaN  "
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_neg.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "b3149836-dba5-43dc-bce4-f15eb46c6ea4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pmid</th>\n",
       "      <th>subject_cui</th>\n",
       "      <th>subject_type</th>\n",
       "      <th>object_cui</th>\n",
       "      <th>object_type</th>\n",
       "      <th>year</th>\n",
       "      <th>subject_name</th>\n",
       "      <th>predicate</th>\n",
       "      <th>object_name</th>\n",
       "      <th>source_text</th>\n",
       "      <th>Unnamed: 10</th>\n",
       "      <th>predicate_obo</th>\n",
       "      <th>subject_obo</th>\n",
       "      <th>object_obo</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1679</th>\n",
       "      <td>27260674</td>\n",
       "      <td>C0066619</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>C1416572</td>\n",
       "      <td>gngm,aapp</td>\n",
       "      <td>2016.0</td>\n",
       "      <td>mitragynine</td>\n",
       "      <td>neg_stimulates</td>\n",
       "      <td>KCNH2 gene</td>\n",
       "      <td>27260674_ascii.txt.tx.2 Mitragynine did not in...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CHEBI_6956</td>\n",
       "      <td>PR_000000791</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1717</th>\n",
       "      <td>21876481</td>\n",
       "      <td>C0066619</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>C0026549</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>2011.0</td>\n",
       "      <td>mitragynine</td>\n",
       "      <td>neg_same_as</td>\n",
       "      <td>Morphine</td>\n",
       "      <td>21876481_ascii.txt.tx.2 Each  point  was  Mitr...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CHEBI_6956</td>\n",
       "      <td>CHEBI_17303</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1746</th>\n",
       "      <td>16107269</td>\n",
       "      <td>C0066619</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>C0034801</td>\n",
       "      <td>aapp,gngm,rcpt</td>\n",
       "      <td>2005.0</td>\n",
       "      <td>mitragynine</td>\n",
       "      <td>neg_interacts_with</td>\n",
       "      <td>Opioid Receptor</td>\n",
       "      <td>16107269_ascii.txt.tx.1 This does not necessar...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CHEBI_6956</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1761</th>\n",
       "      <td>23274770</td>\n",
       "      <td>C0066619</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gngm,aapp</td>\n",
       "      <td>2013.0</td>\n",
       "      <td>mitragynine</td>\n",
       "      <td>neg_interacts_with</td>\n",
       "      <td>NaN</td>\n",
       "      <td>23274770_ascii.txt.tx.1 Interestingly, and unl...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CHEBI_6956</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1762</th>\n",
       "      <td>23274770</td>\n",
       "      <td>C0066619</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>C1515670</td>\n",
       "      <td>genf</td>\n",
       "      <td>2013.0</td>\n",
       "      <td>mitragynine</td>\n",
       "      <td>neg_affects</td>\n",
       "      <td>mRNA Expression</td>\n",
       "      <td>23274770_ascii.txt.tx.1 Mitragynine does not a...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CHEBI_6956</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1763</th>\n",
       "      <td>23274770</td>\n",
       "      <td>C0066619</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>C0544357</td>\n",
       "      <td>phsu</td>\n",
       "      <td>2013.0</td>\n",
       "      <td>mitragynine</td>\n",
       "      <td>neg_coexists_with</td>\n",
       "      <td>human plasma</td>\n",
       "      <td>23274770_ascii.txt.tx.1 Logically, concentrati...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CHEBI_6956</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          pmid subject_cui subject_type object_cui     object_type    year  \\\n",
       "1679  27260674    C0066619    orch,phsu   C1416572       gngm,aapp  2016.0   \n",
       "1717  21876481    C0066619    orch,phsu   C0026549       orch,phsu  2011.0   \n",
       "1746  16107269    C0066619    orch,phsu   C0034801  aapp,gngm,rcpt  2005.0   \n",
       "1761  23274770    C0066619    orch,phsu        NaN       gngm,aapp  2013.0   \n",
       "1762  23274770    C0066619    orch,phsu   C1515670            genf  2013.0   \n",
       "1763  23274770    C0066619    orch,phsu   C0544357            phsu  2013.0   \n",
       "\n",
       "     subject_name           predicate      object_name  \\\n",
       "1679  mitragynine      neg_stimulates       KCNH2 gene   \n",
       "1717  mitragynine         neg_same_as         Morphine   \n",
       "1746  mitragynine  neg_interacts_with  Opioid Receptor   \n",
       "1761  mitragynine  neg_interacts_with              NaN   \n",
       "1762  mitragynine         neg_affects  mRNA Expression   \n",
       "1763  mitragynine   neg_coexists_with     human plasma   \n",
       "\n",
       "                                            source_text  Unnamed: 10  \\\n",
       "1679  27260674_ascii.txt.tx.2 Mitragynine did not in...          NaN   \n",
       "1717  21876481_ascii.txt.tx.2 Each  point  was  Mitr...          NaN   \n",
       "1746  16107269_ascii.txt.tx.1 This does not necessar...          NaN   \n",
       "1761  23274770_ascii.txt.tx.1 Interestingly, and unl...          NaN   \n",
       "1762  23274770_ascii.txt.tx.1 Mitragynine does not a...          NaN   \n",
       "1763  23274770_ascii.txt.tx.1 Logically, concentrati...          NaN   \n",
       "\n",
       "      predicate_obo subject_obo    object_obo  \n",
       "1679            NaN  CHEBI_6956  PR_000000791  \n",
       "1717            NaN  CHEBI_6956   CHEBI_17303  \n",
       "1746            NaN  CHEBI_6956           NaN  \n",
       "1761            NaN  CHEBI_6956           NaN  \n",
       "1762            NaN  CHEBI_6956           NaN  \n",
       "1763            NaN  CHEBI_6956           NaN  "
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_neg.loc[df_neg['subject_name'] == 'mitragynine']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "d30ea8f3-5db1-4659-a62d-851e34600fe7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pmid</th>\n",
       "      <th>subject_cui</th>\n",
       "      <th>subject_type</th>\n",
       "      <th>object_cui</th>\n",
       "      <th>object_type</th>\n",
       "      <th>year</th>\n",
       "      <th>subject_name</th>\n",
       "      <th>predicate</th>\n",
       "      <th>object_name</th>\n",
       "      <th>source_text</th>\n",
       "      <th>Unnamed: 10</th>\n",
       "      <th>predicate_obo</th>\n",
       "      <th>subject_obo</th>\n",
       "      <th>object_obo</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [pmid, subject_cui, subject_type, object_cui, object_type, year, subject_name, predicate, object_name, source_text, Unnamed: 10, predicate_obo, subject_obo, object_obo]\n",
       "Index: []"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "##7-hydroxy-mitragynine\n",
    "df_neg.loc[df_neg['object_obo'] == 'CHEBI_180536']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "id": "93a18e67-911d-4fd8-8e1f-94ebd97d7b20",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "98"
      ]
     },
     "execution_count": 77,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(get_neighbors_filtered(nx_graph, obodict['mitragynine'][0]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f8043bb6-f5c3-425b-8ffd-09048ac4d1cc",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "ee3cc19e-3194-4b7a-98fe-d064f671d2e2",
   "metadata": {
    "tags": []
   },
   "source": [
    "## Green Tea - Mechanistic Exploration Part 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "id": "6f41d347-4542-4283-a204-a0b4c86e9dd7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "87"
      ]
     },
     "execution_count": 74,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "##read through - looks like useful information and a lot from machine read and inferred \n",
    "len(get_neighbors_filtered(nx_graph, obodict['greentea'][0]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "id": "a9db3bb9-7c20-401c-afee-fc4a407bce2f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[[{'entity_type': 'NODES', 'label': '7-hydroxymitragynine', 'cui': 'C1435927'},\n",
       "  'directly negatively regulates activity of',\n",
       "  '4-methylumbelliferone beta-D-glucuronide',\n",
       "  'machine_read',\n",
       "  '2014.0'],\n",
       " [{'entity_type': 'NODES', 'label': '7-hydroxymitragynine', 'cui': 'C1435927'},\n",
       "  'interacts with',\n",
       "  '4-methylumbelliferone beta-D-glucuronide',\n",
       "  'machine_read_INF',\n",
       "  '2014.0'],\n",
       " [{'entity_type': 'NODES', 'label': '7-hydroxymitragynine', 'cui': 'C1435927'},\n",
       "  'directly negatively regulates activity of',\n",
       "  'Mitragynine',\n",
       "  'machine_read',\n",
       "  '2014.0'],\n",
       " [{'entity_type': 'NODES', 'label': '7-hydroxymitragynine', 'cui': 'C1435927'},\n",
       "  'directly negatively regulates activity of',\n",
       "  'UDP-glucuronosyltransferase 2B7',\n",
       "  'machine_read',\n",
       "  '2014.0']]"
      ]
     },
     "execution_count": 79,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "get_neighbors_filtered(nx_graph, obodict['hydroxy_mitragynine'][1])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 149,
   "id": "aca7946b-8438-4807-b687-cd537ddaa62a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pmid</th>\n",
       "      <th>subject_cui</th>\n",
       "      <th>subject_type</th>\n",
       "      <th>object_cui</th>\n",
       "      <th>object_type</th>\n",
       "      <th>year</th>\n",
       "      <th>subject_name</th>\n",
       "      <th>predicate</th>\n",
       "      <th>object_name</th>\n",
       "      <th>source_text</th>\n",
       "      <th>Unnamed: 10</th>\n",
       "      <th>predicate_obo</th>\n",
       "      <th>subject_obo</th>\n",
       "      <th>object_obo</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>19353999</td>\n",
       "      <td>C0059438</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>C1332826</td>\n",
       "      <td>gngm,aapp</td>\n",
       "      <td>2009.0</td>\n",
       "      <td>epigallocatechin gallate</td>\n",
       "      <td>neg_interacts_with</td>\n",
       "      <td>CYP1A2 gene</td>\n",
       "      <td>19353999_ascii.txt.tx.1 A recent report has fo...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>napdi_srs_imports:epigallocatechin_gallate</td>\n",
       "      <td>PR_000006102</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>12659723</td>\n",
       "      <td>C0007404</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>C1420188</td>\n",
       "      <td>gngm,aapp</td>\n",
       "      <td>2003.0</td>\n",
       "      <td>Catechin</td>\n",
       "      <td>neg_interacts_with</td>\n",
       "      <td>SLC35A2 gene</td>\n",
       "      <td>12659723_ascii.txt.tx.2 Lhoste et al./ Food an...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CHEBI_23053</td>\n",
       "      <td>PR_000015090</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>12659723</td>\n",
       "      <td>C0022203</td>\n",
       "      <td>npop</td>\n",
       "      <td>C0011135</td>\n",
       "      <td>orgf</td>\n",
       "      <td>2003.0</td>\n",
       "      <td>Isomerism (chemical phenomenon)</td>\n",
       "      <td>neg_affects</td>\n",
       "      <td>Defecation</td>\n",
       "      <td>12659723_ascii.txt.tx.2 Other polyphenols, e.g...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>32645157</td>\n",
       "      <td>C0041912</td>\n",
       "      <td>dsyn</td>\n",
       "      <td>C1154978</td>\n",
       "      <td>orgf</td>\n",
       "      <td>2020.0</td>\n",
       "      <td>Upper Respiratory Infections</td>\n",
       "      <td>neg_affects</td>\n",
       "      <td>cold acclimation</td>\n",
       "      <td>32645157_ascii.txt.tx.1 After undergoing cold ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MONDO_0024355</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>32645157</td>\n",
       "      <td>C1167041</td>\n",
       "      <td>celc</td>\n",
       "      <td>C0120285</td>\n",
       "      <td>aapp,gngm,irda</td>\n",
       "      <td>2020.0</td>\n",
       "      <td>vacuolar membrane</td>\n",
       "      <td>neg_location_of</td>\n",
       "      <td>Green Fluorescent Proteins</td>\n",
       "      <td>32645157_ascii.txt.tx.1 Our results showed tha...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       pmid subject_cui subject_type object_cui     object_type    year  \\\n",
       "0  19353999    C0059438    orch,phsu   C1332826       gngm,aapp  2009.0   \n",
       "1  12659723    C0007404    orch,phsu   C1420188       gngm,aapp  2003.0   \n",
       "2  12659723    C0022203         npop   C0011135            orgf  2003.0   \n",
       "3  32645157    C0041912         dsyn   C1154978            orgf  2020.0   \n",
       "4  32645157    C1167041         celc   C0120285  aapp,gngm,irda  2020.0   \n",
       "\n",
       "                      subject_name           predicate  \\\n",
       "0         epigallocatechin gallate  neg_interacts_with   \n",
       "1                         Catechin  neg_interacts_with   \n",
       "2  Isomerism (chemical phenomenon)         neg_affects   \n",
       "3     Upper Respiratory Infections         neg_affects   \n",
       "4                vacuolar membrane     neg_location_of   \n",
       "\n",
       "                  object_name  \\\n",
       "0                 CYP1A2 gene   \n",
       "1                SLC35A2 gene   \n",
       "2                  Defecation   \n",
       "3            cold acclimation   \n",
       "4  Green Fluorescent Proteins   \n",
       "\n",
       "                                         source_text  Unnamed: 10  \\\n",
       "0  19353999_ascii.txt.tx.1 A recent report has fo...          NaN   \n",
       "1  12659723_ascii.txt.tx.2 Lhoste et al./ Food an...          NaN   \n",
       "2  12659723_ascii.txt.tx.2 Other polyphenols, e.g...          NaN   \n",
       "3  32645157_ascii.txt.tx.1 After undergoing cold ...          NaN   \n",
       "4  32645157_ascii.txt.tx.1 Our results showed tha...          NaN   \n",
       "\n",
       "   predicate_obo                                 subject_obo    object_obo  \n",
       "0            NaN  napdi_srs_imports:epigallocatechin_gallate  PR_000006102  \n",
       "1            NaN                                 CHEBI_23053  PR_000015090  \n",
       "2            NaN                                         NaN           NaN  \n",
       "3            NaN                               MONDO_0024355           NaN  \n",
       "4            NaN                                         NaN           NaN  "
      ]
     },
     "execution_count": 149,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_neg.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "78c97ee6-c266-43bc-bca2-b5bce4d34012",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "#### Green tea leaf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "id": "d71b847d-3d7a-43be-aa55-b827c732c128",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "directly negatively regulates activity of\n",
      "{'predicate_key': '330824948c6308aba5c71305c24a46ab', 'weight': 0.0, 'pmid': '15319329', 'timestamp': '2004.0', 'source_graph': 'machine_read', 'belief': 0.8}\n"
     ]
    }
   ],
   "source": [
    "##there is an edge with glycoprotein (CHEBI_)\n",
    "\n",
    "edges = nx_graph.get_edge_data(obodict['greentea'][0], obo.CHEBI_17089)\n",
    "if edges:\n",
    "    for item in edges:\n",
    "        print(nodeLabels[str(item)])\n",
    "        print(edges[item])\n",
    "edges = nx_graph.get_edge_data(obodict['greentea'][0], obo.PR_000037069)\n",
    "if edges:\n",
    "    for item in edges:\n",
    "        print(nodeLabels[str(item)])\n",
    "        print(edges[item])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 132,
   "id": "7142ef42-3756-4f24-83c7-3ccd0c26a9bc",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "directly negatively regulates activity of\n",
      "{'predicate_key': '2da8cfaf56cc552a738b5007546fb60b', 'weight': 0.0, 'pmid': '29467215', 'timestamp': '2018.0', 'source_graph': 'machine_read', 'belief': 0.8}\n"
     ]
    }
   ],
   "source": [
    "edges = nx_graph.get_edge_data(obodict['greentea'][0], obodict['UGT'][0])\n",
    "if edges:\n",
    "    for item in edges:\n",
    "        print(nodeLabels[str(item)])\n",
    "        print(edges[item])\n",
    "edges = nx_graph.get_edge_data(obodict['greentea'][0], obodict['UGT'][1])\n",
    "if edges:\n",
    "    for item in edges:\n",
    "        print(nodeLabels[str(item)])\n",
    "        print(edges[item])\n",
    "edges = nx_graph.get_edge_data(obodict['greentea'][0], obodict['UGT'][2])\n",
    "if edges:\n",
    "    for item in edges:\n",
    "        print(nodeLabels[str(item)])\n",
    "        print(edges[item])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 169,
   "id": "9e1f019b-50ee-4f66-abf1-a1ec05195689",
   "metadata": {},
   "outputs": [],
   "source": [
    "edges = nx_graph.get_edge_data(obodict['greentea'][0], obodict['PGP'][3])\n",
    "if edges:\n",
    "    for item in edges:\n",
    "        print(nodeLabels[str(item)])\n",
    "        print(edges[item])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 174,
   "id": "e8889587-2794-4069-a47d-d3eadd3183cc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pmid</th>\n",
       "      <th>subject_cui</th>\n",
       "      <th>subject_type</th>\n",
       "      <th>object_cui</th>\n",
       "      <th>object_type</th>\n",
       "      <th>year</th>\n",
       "      <th>subject_name</th>\n",
       "      <th>predicate</th>\n",
       "      <th>object_name</th>\n",
       "      <th>source_text</th>\n",
       "      <th>Unnamed: 10</th>\n",
       "      <th>predicate_obo</th>\n",
       "      <th>subject_obo</th>\n",
       "      <th>object_obo</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>20471964</td>\n",
       "      <td>C1704263</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>C0030012</td>\n",
       "      <td>moft</td>\n",
       "      <td>2010.0</td>\n",
       "      <td>Green Tea Extract</td>\n",
       "      <td>neg_affects</td>\n",
       "      <td>Oxidation-Reduction</td>\n",
       "      <td>20471964_ascii.txt.tx.3 Catechins are also kno...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>napdi_srs_imports:camellia_sinensis_leaf</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>103</th>\n",
       "      <td>25418056</td>\n",
       "      <td>C0039400</td>\n",
       "      <td>food</td>\n",
       "      <td>C0031327</td>\n",
       "      <td>phsf</td>\n",
       "      <td>2015.0</td>\n",
       "      <td>Tea</td>\n",
       "      <td>neg_affects</td>\n",
       "      <td>Drug Kinetics</td>\n",
       "      <td>25418056_ascii.txt.tx.1 With the continuing ef...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>napdi_srs_imports:camellia_sinensis_leaf</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>205</th>\n",
       "      <td>29570984</td>\n",
       "      <td>C0039400</td>\n",
       "      <td>food</td>\n",
       "      <td>C0017638</td>\n",
       "      <td>neop</td>\n",
       "      <td>2018.0</td>\n",
       "      <td>Tea</td>\n",
       "      <td>neg_predisposes</td>\n",
       "      <td>Glioma</td>\n",
       "      <td>29570984_ascii.txt.tx.1 335 glioma cases NIH-A...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>napdi_srs_imports:camellia_sinensis_leaf</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>206</th>\n",
       "      <td>29570984</td>\n",
       "      <td>C0039400</td>\n",
       "      <td>food</td>\n",
       "      <td>C0017638</td>\n",
       "      <td>neop</td>\n",
       "      <td>2018.0</td>\n",
       "      <td>Tea</td>\n",
       "      <td>neg_predisposes</td>\n",
       "      <td>Glioma</td>\n",
       "      <td>29570984_ascii.txt.tx.3 With the first quantil...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>napdi_srs_imports:camellia_sinensis_leaf</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>420</th>\n",
       "      <td>31100973</td>\n",
       "      <td>C0039400</td>\n",
       "      <td>food</td>\n",
       "      <td>C0013470</td>\n",
       "      <td>orgf</td>\n",
       "      <td>2019.0</td>\n",
       "      <td>Tea</td>\n",
       "      <td>neg_affects</td>\n",
       "      <td>Eating</td>\n",
       "      <td>31100973_ascii.txt.tx.3 However, supplementati...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>napdi_srs_imports:camellia_sinensis_leaf</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>421</th>\n",
       "      <td>31100973</td>\n",
       "      <td>C0039400</td>\n",
       "      <td>food</td>\n",
       "      <td>C0020615</td>\n",
       "      <td>dsyn</td>\n",
       "      <td>2019.0</td>\n",
       "      <td>Tea</td>\n",
       "      <td>neg_causes</td>\n",
       "      <td>Hypoglycemia</td>\n",
       "      <td>31100973_ascii.txt.tx.2 Green tea was then adm...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>napdi_srs_imports:camellia_sinensis_leaf</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>422</th>\n",
       "      <td>31100973</td>\n",
       "      <td>C0039400</td>\n",
       "      <td>food</td>\n",
       "      <td>C0178665</td>\n",
       "      <td>phsf</td>\n",
       "      <td>2019.0</td>\n",
       "      <td>Tea</td>\n",
       "      <td>neg_affects</td>\n",
       "      <td>glucose tolerance</td>\n",
       "      <td>31100973_ascii.txt.tx.1 Since the blood glucos...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>napdi_srs_imports:camellia_sinensis_leaf</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>503</th>\n",
       "      <td>23106150</td>\n",
       "      <td>C0039400</td>\n",
       "      <td>food</td>\n",
       "      <td>C0013470</td>\n",
       "      <td>orgf</td>\n",
       "      <td>2012.0</td>\n",
       "      <td>Tea</td>\n",
       "      <td>neg_affects</td>\n",
       "      <td>Eating</td>\n",
       "      <td>23106150_ascii.txt.tx.3 During the feeding per...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>napdi_srs_imports:camellia_sinensis_leaf</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>514</th>\n",
       "      <td>24656388</td>\n",
       "      <td>C1704263</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>C2698419</td>\n",
       "      <td>gngm,aapp</td>\n",
       "      <td>2014.0</td>\n",
       "      <td>Green Tea Extract</td>\n",
       "      <td>neg_interacts_with</td>\n",
       "      <td>NR0B1 wt Allele</td>\n",
       "      <td>24656388_ascii.txt.tx.1 / Journal of Nutrition...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>napdi_srs_imports:camellia_sinensis_leaf</td>\n",
       "      <td>PR_000011390</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>549</th>\n",
       "      <td>19170155</td>\n",
       "      <td>C1704263</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>C0002333</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>2009.0</td>\n",
       "      <td>Green Tea Extract</td>\n",
       "      <td>neg_interacts_with</td>\n",
       "      <td>Alprazolam</td>\n",
       "      <td>19170155_ascii.txt.tx.1 One pharmacokinetic st...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>napdi_srs_imports:camellia_sinensis_leaf</td>\n",
       "      <td>CHEBI_2611</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>756</th>\n",
       "      <td>17108060</td>\n",
       "      <td>C1704263</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gngm,aapp</td>\n",
       "      <td>2007.0</td>\n",
       "      <td>Green Tea Extract</td>\n",
       "      <td>neg_interacts_with</td>\n",
       "      <td>NaN</td>\n",
       "      <td>17108060_ascii.txt.tx.1 However, in healthy vo...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>napdi_srs_imports:camellia_sinensis_leaf</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>757</th>\n",
       "      <td>17108060</td>\n",
       "      <td>C1704263</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>C1332830</td>\n",
       "      <td>gngm,aapp</td>\n",
       "      <td>2007.0</td>\n",
       "      <td>Green Tea Extract</td>\n",
       "      <td>neg_interacts_with</td>\n",
       "      <td>CYP2D6 gene</td>\n",
       "      <td>17108060_ascii.txt.tx.1 However, in healthy vo...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>napdi_srs_imports:camellia_sinensis_leaf</td>\n",
       "      <td>PR_000006121</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>853</th>\n",
       "      <td>15729621</td>\n",
       "      <td>C1704263</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>C0069906</td>\n",
       "      <td>aapp,gngm,bacs</td>\n",
       "      <td>2005.0</td>\n",
       "      <td>Green Tea Extract</td>\n",
       "      <td>neg_stimulates</td>\n",
       "      <td>P-Glycoproteins</td>\n",
       "      <td>15729621_ascii.txt.tx.3 Furthermore, the funct...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>napdi_srs_imports:camellia_sinensis_leaf</td>\n",
       "      <td>GO_0008559</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>854</th>\n",
       "      <td>15729621</td>\n",
       "      <td>C1704263</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>C0525410</td>\n",
       "      <td>aapp,gngm,bacs</td>\n",
       "      <td>2005.0</td>\n",
       "      <td>Green Tea Extract</td>\n",
       "      <td>neg_stimulates</td>\n",
       "      <td>multidrug resistance-associated protein 2</td>\n",
       "      <td>15729621_ascii.txt.tx.3 Furthermore, the funct...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>napdi_srs_imports:camellia_sinensis_leaf</td>\n",
       "      <td>PR_000003558</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1050</th>\n",
       "      <td>25098399</td>\n",
       "      <td>C0039400</td>\n",
       "      <td>food</td>\n",
       "      <td>C0221102</td>\n",
       "      <td>phsf</td>\n",
       "      <td>2014.0</td>\n",
       "      <td>Tea</td>\n",
       "      <td>neg_affects</td>\n",
       "      <td>Excretory function</td>\n",
       "      <td>25098399_ascii.txt.tx.1 The total polyphenol c...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>napdi_srs_imports:camellia_sinensis_leaf</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1086</th>\n",
       "      <td>29368187</td>\n",
       "      <td>C1704263</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gngm,aapp</td>\n",
       "      <td>2018.0</td>\n",
       "      <td>Green Tea Extract</td>\n",
       "      <td>neg_interacts_with</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29368187_ascii.txt.tx.1 On the contrary, a pre...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>napdi_srs_imports:camellia_sinensis_leaf</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1397</th>\n",
       "      <td>34206736</td>\n",
       "      <td>C1704263</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>C0005938</td>\n",
       "      <td>clna</td>\n",
       "      <td>2021.0</td>\n",
       "      <td>Green Tea Extract</td>\n",
       "      <td>neg_affects</td>\n",
       "      <td>Bone Density</td>\n",
       "      <td>34206736_ascii.txt.tx.1 Elmassry et al.[79] ca...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>napdi_srs_imports:camellia_sinensis_leaf</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1509</th>\n",
       "      <td>19170155</td>\n",
       "      <td>C1704263</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>C0002333</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>2009.0</td>\n",
       "      <td>Green Tea Extract</td>\n",
       "      <td>neg_interacts_with</td>\n",
       "      <td>Alprazolam</td>\n",
       "      <td>19170155_ascii.txt.tx.1 One pharmacokinetic st...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>napdi_srs_imports:camellia_sinensis_leaf</td>\n",
       "      <td>CHEBI_2611</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1625</th>\n",
       "      <td>29368187</td>\n",
       "      <td>C1704263</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gngm,aapp</td>\n",
       "      <td>2018.0</td>\n",
       "      <td>Green Tea Extract</td>\n",
       "      <td>neg_interacts_with</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29368187_ascii.txt.tx.3 With respect to CYP2C9...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>napdi_srs_imports:camellia_sinensis_leaf</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          pmid subject_cui subject_type object_cui     object_type    year  \\\n",
       "13    20471964    C1704263    orch,phsu   C0030012            moft  2010.0   \n",
       "103   25418056    C0039400         food   C0031327            phsf  2015.0   \n",
       "205   29570984    C0039400         food   C0017638            neop  2018.0   \n",
       "206   29570984    C0039400         food   C0017638            neop  2018.0   \n",
       "420   31100973    C0039400         food   C0013470            orgf  2019.0   \n",
       "421   31100973    C0039400         food   C0020615            dsyn  2019.0   \n",
       "422   31100973    C0039400         food   C0178665            phsf  2019.0   \n",
       "503   23106150    C0039400         food   C0013470            orgf  2012.0   \n",
       "514   24656388    C1704263    orch,phsu   C2698419       gngm,aapp  2014.0   \n",
       "549   19170155    C1704263    orch,phsu   C0002333       orch,phsu  2009.0   \n",
       "756   17108060    C1704263    orch,phsu        NaN       gngm,aapp  2007.0   \n",
       "757   17108060    C1704263    orch,phsu   C1332830       gngm,aapp  2007.0   \n",
       "853   15729621    C1704263    orch,phsu   C0069906  aapp,gngm,bacs  2005.0   \n",
       "854   15729621    C1704263    orch,phsu   C0525410  aapp,gngm,bacs  2005.0   \n",
       "1050  25098399    C0039400         food   C0221102            phsf  2014.0   \n",
       "1086  29368187    C1704263    orch,phsu        NaN       gngm,aapp  2018.0   \n",
       "1397  34206736    C1704263    orch,phsu   C0005938            clna  2021.0   \n",
       "1509  19170155    C1704263    orch,phsu   C0002333       orch,phsu  2009.0   \n",
       "1625  29368187    C1704263    orch,phsu        NaN       gngm,aapp  2018.0   \n",
       "\n",
       "           subject_name           predicate  \\\n",
       "13    Green Tea Extract         neg_affects   \n",
       "103                 Tea         neg_affects   \n",
       "205                 Tea     neg_predisposes   \n",
       "206                 Tea     neg_predisposes   \n",
       "420                 Tea         neg_affects   \n",
       "421                 Tea          neg_causes   \n",
       "422                 Tea         neg_affects   \n",
       "503                 Tea         neg_affects   \n",
       "514   Green Tea Extract  neg_interacts_with   \n",
       "549   Green Tea Extract  neg_interacts_with   \n",
       "756   Green Tea Extract  neg_interacts_with   \n",
       "757   Green Tea Extract  neg_interacts_with   \n",
       "853   Green Tea Extract      neg_stimulates   \n",
       "854   Green Tea Extract      neg_stimulates   \n",
       "1050                Tea         neg_affects   \n",
       "1086  Green Tea Extract  neg_interacts_with   \n",
       "1397  Green Tea Extract         neg_affects   \n",
       "1509  Green Tea Extract  neg_interacts_with   \n",
       "1625  Green Tea Extract  neg_interacts_with   \n",
       "\n",
       "                                    object_name  \\\n",
       "13                          Oxidation-Reduction   \n",
       "103                               Drug Kinetics   \n",
       "205                                      Glioma   \n",
       "206                                      Glioma   \n",
       "420                                      Eating   \n",
       "421                                Hypoglycemia   \n",
       "422                           glucose tolerance   \n",
       "503                                      Eating   \n",
       "514                             NR0B1 wt Allele   \n",
       "549                                  Alprazolam   \n",
       "756                                         NaN   \n",
       "757                                 CYP2D6 gene   \n",
       "853                             P-Glycoproteins   \n",
       "854   multidrug resistance-associated protein 2   \n",
       "1050                         Excretory function   \n",
       "1086                                        NaN   \n",
       "1397                               Bone Density   \n",
       "1509                                 Alprazolam   \n",
       "1625                                        NaN   \n",
       "\n",
       "                                            source_text  Unnamed: 10  \\\n",
       "13    20471964_ascii.txt.tx.3 Catechins are also kno...          NaN   \n",
       "103   25418056_ascii.txt.tx.1 With the continuing ef...          NaN   \n",
       "205   29570984_ascii.txt.tx.1 335 glioma cases NIH-A...          NaN   \n",
       "206   29570984_ascii.txt.tx.3 With the first quantil...          NaN   \n",
       "420   31100973_ascii.txt.tx.3 However, supplementati...          NaN   \n",
       "421   31100973_ascii.txt.tx.2 Green tea was then adm...          NaN   \n",
       "422   31100973_ascii.txt.tx.1 Since the blood glucos...          NaN   \n",
       "503   23106150_ascii.txt.tx.3 During the feeding per...          NaN   \n",
       "514   24656388_ascii.txt.tx.1 / Journal of Nutrition...          NaN   \n",
       "549   19170155_ascii.txt.tx.1 One pharmacokinetic st...          NaN   \n",
       "756   17108060_ascii.txt.tx.1 However, in healthy vo...          NaN   \n",
       "757   17108060_ascii.txt.tx.1 However, in healthy vo...          NaN   \n",
       "853   15729621_ascii.txt.tx.3 Furthermore, the funct...          NaN   \n",
       "854   15729621_ascii.txt.tx.3 Furthermore, the funct...          NaN   \n",
       "1050  25098399_ascii.txt.tx.1 The total polyphenol c...          NaN   \n",
       "1086  29368187_ascii.txt.tx.1 On the contrary, a pre...          NaN   \n",
       "1397  34206736_ascii.txt.tx.1 Elmassry et al.[79] ca...          NaN   \n",
       "1509  19170155_ascii.txt.tx.1 One pharmacokinetic st...          NaN   \n",
       "1625  29368187_ascii.txt.tx.3 With respect to CYP2C9...          NaN   \n",
       "\n",
       "      predicate_obo                               subject_obo    object_obo  \n",
       "13              NaN  napdi_srs_imports:camellia_sinensis_leaf           NaN  \n",
       "103             NaN  napdi_srs_imports:camellia_sinensis_leaf           NaN  \n",
       "205             NaN  napdi_srs_imports:camellia_sinensis_leaf           NaN  \n",
       "206             NaN  napdi_srs_imports:camellia_sinensis_leaf           NaN  \n",
       "420             NaN  napdi_srs_imports:camellia_sinensis_leaf           NaN  \n",
       "421             NaN  napdi_srs_imports:camellia_sinensis_leaf           NaN  \n",
       "422             NaN  napdi_srs_imports:camellia_sinensis_leaf           NaN  \n",
       "503             NaN  napdi_srs_imports:camellia_sinensis_leaf           NaN  \n",
       "514             NaN  napdi_srs_imports:camellia_sinensis_leaf  PR_000011390  \n",
       "549             NaN  napdi_srs_imports:camellia_sinensis_leaf    CHEBI_2611  \n",
       "756             NaN  napdi_srs_imports:camellia_sinensis_leaf           NaN  \n",
       "757             NaN  napdi_srs_imports:camellia_sinensis_leaf  PR_000006121  \n",
       "853             NaN  napdi_srs_imports:camellia_sinensis_leaf    GO_0008559  \n",
       "854             NaN  napdi_srs_imports:camellia_sinensis_leaf  PR_000003558  \n",
       "1050            NaN  napdi_srs_imports:camellia_sinensis_leaf           NaN  \n",
       "1086            NaN  napdi_srs_imports:camellia_sinensis_leaf           NaN  \n",
       "1397            NaN  napdi_srs_imports:camellia_sinensis_leaf           NaN  \n",
       "1509            NaN  napdi_srs_imports:camellia_sinensis_leaf    CHEBI_2611  \n",
       "1625            NaN  napdi_srs_imports:camellia_sinensis_leaf           NaN  "
      ]
     },
     "execution_count": 174,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_neg.loc[df_neg['subject_obo'] == 'napdi_srs_imports:camellia_sinensis_leaf']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 126,
   "id": "6c736cb6-9b7e-4087-b3fb-4ff7469f3322",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[['Camellia_sinensis_leaf',\n",
       "  'positively regulates',\n",
       "  'cytochrome P450 1A2',\n",
       "  'machine_read',\n",
       "  '2005.0'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  'directly negatively regulates activity of',\n",
       "  'cytochrome P450 1A2',\n",
       "  'machine_read',\n",
       "  '2009.0'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  'interacts with',\n",
       "  'cytochrome P450 1A2',\n",
       "  'machine_read_INF',\n",
       "  '2003.0'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  'interacts with',\n",
       "  'ion',\n",
       "  'machine_read_INF',\n",
       "  '2013.0'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  'directly negatively regulates activity of',\n",
       "  'N-methyl-4-phenylpyridinium',\n",
       "  'machine_read',\n",
       "  '2006.0'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  'positively regulates',\n",
       "  'insulin receptor substrate 2 (human)',\n",
       "  'machine_read_INF',\n",
       "  '2013.0'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'affects'},\n",
       "  'activation of protein kinase C activity',\n",
       "  'machine_read',\n",
       "  '2006.0'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  'directly negatively regulates activity of',\n",
       "  'glycoprotein',\n",
       "  'machine_read',\n",
       "  '2004.0'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'affects'},\n",
       "  'breast cancer',\n",
       "  'machine_read',\n",
       "  '2018.0'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  'interacts with',\n",
       "  'flavonoids',\n",
       "  'machine_read_INF',\n",
       "  '2021.0'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  'interacts with',\n",
       "  'microtubule-associated protein tau',\n",
       "  'machine_read_INF',\n",
       "  '2018.0'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  'positively regulates',\n",
       "  'peroxisome proliferator-activated receptor gamma coactivator 1-alpha',\n",
       "  'machine_read_INF',\n",
       "  '2014.0'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'affects'},\n",
       "  'diabetes mellitus',\n",
       "  'machine_read',\n",
       "  '2017.0'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  'interacts with',\n",
       "  'polyphenol',\n",
       "  'machine_read_INF',\n",
       "  '2003.0'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  'interacts with',\n",
       "  'canalicular multispecific organic anion transporter 1',\n",
       "  'machine_read_INF',\n",
       "  '2018.0'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  'positively regulates',\n",
       "  'rosoxacin',\n",
       "  'machine_read_INF',\n",
       "  '2015.0'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  'directly negatively regulates activity of',\n",
       "  'rosoxacin',\n",
       "  'machine_read',\n",
       "  '2015.0'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  'positively regulates',\n",
       "  'very-long-chain enoyl-CoA reductase',\n",
       "  'machine_read',\n",
       "  '2004.0'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'affects'},\n",
       "  'import into cell',\n",
       "  'machine_read',\n",
       "  '2017.0'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  'negatively regulates',\n",
       "  'import into cell',\n",
       "  'machine_read',\n",
       "  '2020.0'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  'directly negatively regulates activity of',\n",
       "  'sodium-dependent dopamine transporter (human)',\n",
       "  'machine_read',\n",
       "  '2006.0'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  'directly negatively regulates activity of',\n",
       "  'UDP-galactose translocator',\n",
       "  'machine_read',\n",
       "  '2018.0'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'affects'},\n",
       "  'heart',\n",
       "  'machine_read',\n",
       "  '2018.0'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  'interacts with',\n",
       "  '(-)-epigallocatechin gallate',\n",
       "  'machine_read_INF',\n",
       "  '2008.0'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  'positively regulates',\n",
       "  '(-)-epigallocatechin gallate',\n",
       "  'machine_read_INF'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  'directly negatively regulates activity of',\n",
       "  'fructose',\n",
       "  'machine_read',\n",
       "  '2017.0'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  'directly negatively regulates activity of',\n",
       "  '4-nitrophenol',\n",
       "  'machine_read',\n",
       "  '2001.0'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  'directly negatively regulates activity of',\n",
       "  'cytochrome P450 2E1 (human)',\n",
       "  'machine_read',\n",
       "  '2007.0'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  'interacts with',\n",
       "  'catechin',\n",
       "  'machine_read_INF',\n",
       "  '2011.0'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  'directly negatively regulates activity of',\n",
       "  'catechin',\n",
       "  'machine_read',\n",
       "  '2005.0'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  'interacts with',\n",
       "  'atorvastatin',\n",
       "  'machine_read_INF',\n",
       "  '2020.0'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  'directly negatively regulates activity of',\n",
       "  'atorvastatin',\n",
       "  'machine_read',\n",
       "  '2020.0'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  'directly negatively regulates activity of',\n",
       "  'reactive oxygen species',\n",
       "  'machine_read',\n",
       "  '2020.0'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  'interacts with',\n",
       "  'cytochrome P450',\n",
       "  'machine_read_INF',\n",
       "  '2001.0'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  'positively regulates',\n",
       "  'midazolam',\n",
       "  'machine_read',\n",
       "  '2007.0'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  'negatively regulates',\n",
       "  'oxide',\n",
       "  'machine_read',\n",
       "  '2014.0'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'augments'},\n",
       "  'oxide',\n",
       "  'machine_read',\n",
       "  '2005.0'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  'interacts with',\n",
       "  'peroxisome proliferator-activated receptor gamma',\n",
       "  'machine_read_INF',\n",
       "  '2012.0'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  'interacts with',\n",
       "  '7,12-dimethyltetraphene',\n",
       "  'machine_read_INF',\n",
       "  '2021.0'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  'positively regulates',\n",
       "  'glycogen synthase kinase 3',\n",
       "  'machine_read',\n",
       "  '2013.0'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'affects'},\n",
       "  'glucose metabolism disorder',\n",
       "  'machine_read',\n",
       "  '2018.0'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  'positively regulates',\n",
       "  'antioxidant',\n",
       "  'machine_read',\n",
       "  '2001.0'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  'directly negatively regulates activity of',\n",
       "  'solute carrier family 2, facilitated glucose transporter member 1',\n",
       "  'machine_read',\n",
       "  '2014.0'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  'positively regulates',\n",
       "  'transcription factor A, mitochondrial',\n",
       "  'machine_read',\n",
       "  '2014.0'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  'interacts with',\n",
       "  'ABC-type transporter activity',\n",
       "  'machine_read_INF',\n",
       "  '2005.0'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'affects'},\n",
       "  'blood vessel',\n",
       "  'machine_read',\n",
       "  '2012.0'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  'positively regulates',\n",
       "  'ATP',\n",
       "  'machine_read',\n",
       "  '2018.0'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  'interacts with',\n",
       "  'dextromethorphan',\n",
       "  'machine_read_INF',\n",
       "  '2004.0'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  'positively regulates',\n",
       "  'cytochrome P450 1A1',\n",
       "  'machine_read',\n",
       "  '2005.0'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  'interacts with',\n",
       "  'cytochrome P450 1A1',\n",
       "  'machine_read_INF',\n",
       "  '2006.0'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  'directly negatively regulates activity of',\n",
       "  'cytochrome P450 1A1',\n",
       "  'machine_read',\n",
       "  '2000.0'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  'interacts with',\n",
       "  'triglyceride',\n",
       "  'machine_read_INF',\n",
       "  '2006.0'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  'directly negatively regulates activity of',\n",
       "  'nadolol',\n",
       "  'machine_read',\n",
       "  '2018.0'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  'causally influences',\n",
       "  'microinvasive gastric cancer',\n",
       "  'machine_read',\n",
       "  '2005.0'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  'directly negatively regulates activity of',\n",
       "  'glucose transmembrane transport',\n",
       "  'machine_read',\n",
       "  '2002.0'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  'positively regulates',\n",
       "  'imatinib',\n",
       "  'machine_read',\n",
       "  '2020.0'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  'negatively regulates',\n",
       "  {'entity_type': 'NODES',\n",
       "   'label': 'Secondary malignant neoplasm of lung',\n",
       "   'cui': 'C0153676'},\n",
       "  'machine_read',\n",
       "  '2012.0'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  'positively regulates',\n",
       "  'ryanodine receptor 1',\n",
       "  'machine_read',\n",
       "  '2010.0'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  'directly negatively regulates activity of',\n",
       "  'metalloproteinase inhibitor 3',\n",
       "  'machine_read',\n",
       "  '2019.0'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  'directly negatively regulates activity of',\n",
       "  'dopamine',\n",
       "  'machine_read',\n",
       "  '2006.0'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  'positively regulates',\n",
       "  '40S ribosomal protein SA',\n",
       "  'machine_read',\n",
       "  '2014.0'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  'interacts with',\n",
       "  'caveolin-1',\n",
       "  'machine_read_INF',\n",
       "  '2013.0'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'affects'},\n",
       "  'vascular disease',\n",
       "  'machine_read',\n",
       "  '2012.0'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  'interacts with',\n",
       "  'ABC-type xenobiotic transporter activity',\n",
       "  'machine_read_INF',\n",
       "  '2002.0'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  'positively regulates',\n",
       "  'Rho-related GTP-binding protein RhoB',\n",
       "  'machine_read_INF'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  'positively regulates',\n",
       "  'L-ascorbic acid',\n",
       "  'machine_read_INF'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  'positively regulates',\n",
       "  'programmed cell death protein 1',\n",
       "  'machine_read_INF'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  'positively regulates',\n",
       "  'nanos homolog 3 (human)',\n",
       "  'machine_read_INF'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  'positively regulates',\n",
       "  'hydrogen peroxide',\n",
       "  'machine_read_INF'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  'positively regulates',\n",
       "  'DNA-binding transcription factor activity, RNA polymerase II-specific',\n",
       "  'machine_read_INF'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  'positively regulates',\n",
       "  'mitogen-activated protein kinase 3',\n",
       "  'machine_read_INF'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  'positively regulates',\n",
       "  'calcitonin receptor',\n",
       "  'machine_read_INF'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  'positively regulates',\n",
       "  'NAD(P)H oxidase H2O2-forming activity',\n",
       "  'machine_read_INF'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  'positively regulates',\n",
       "  'phospholipid-transporting ATPase IB',\n",
       "  'machine_read_INF'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  'positively regulates',\n",
       "  'succinate--CoA ligase [ADP-forming] subunit beta, mitochondrial',\n",
       "  'machine_read_INF'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  'positively regulates',\n",
       "  'caspase-3',\n",
       "  'machine_read_INF'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  'positively regulates',\n",
       "  'narrow pore, gated channel activity',\n",
       "  'machine_read_INF'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  'positively regulates',\n",
       "  'neuronitis',\n",
       "  'machine_read_INF'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  'positively regulates',\n",
       "  '3,4-Methylenedioxyamphetamine',\n",
       "  'machine_read_INF'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  'positively regulates',\n",
       "  'superoxide',\n",
       "  'machine_read_INF'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  'positively regulates',\n",
       "  'warfarin',\n",
       "  'machine_read_INF'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  'positively regulates',\n",
       "  'sulfites',\n",
       "  'machine_read_INF'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  'positively regulates',\n",
       "  \"5'-AMP-activated protein kinase catalytic subunit alpha-1\",\n",
       "  'machine_read_INF'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  'positively regulates',\n",
       "  'mitomycin',\n",
       "  'machine_read_INF'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  'positively regulates',\n",
       "  'potassium-transporting ATPase alpha chain 1',\n",
       "  'machine_read_INF'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  'positively regulates',\n",
       "  'carcinogenic agent',\n",
       "  'machine_read_INF'],\n",
       " ['Camellia_sinensis_leaf',\n",
       "  'positively regulates',\n",
       "  'N-methyl-D-aspartic acid',\n",
       "  'machine_read_INF']]"
      ]
     },
     "execution_count": 126,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "get_neighbors_filtered(nx_graph, obodict['greentea'][0])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7d4405fb-df32-4e83-8bc0-42fb1b7d9c8c",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "#### EGCG"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "id": "bae04fa8-7f5d-4ab4-bdeb-cd9a853b3c7e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "positively regulates\n",
      "{'predicate_key': '9bd25ce60b782ba99fadae48b130a80c', 'weight': 0.0, 'pmid': '33387731', 'timestamp': '2021.0', 'source_graph': 'machine_read', 'belief': 0.8}\n"
     ]
    }
   ],
   "source": [
    "edges = nx_graph.get_edge_data(obodict['EGCG'][0], obodict['CYP3A4'][0])\n",
    "if edges:\n",
    "    for item in edges:\n",
    "        print(nodeLabels[str(item)])\n",
    "        print(edges[item])\n",
    "edges = nx_graph.get_edge_data(obodict['EGCG'][1], obodict['CYP3A4'][0])\n",
    "if edges:\n",
    "    for item in edges:\n",
    "        print(nodeLabels[str(item)])\n",
    "        print(edges[item])\n",
    "edges = nx_graph.get_edge_data(obodict['EGCG'][0], obodict['CYP3A4'][1])\n",
    "if edges:\n",
    "    for item in edges:\n",
    "        print(nodeLabels[str(item)])\n",
    "        print(edges[item])\n",
    "edges = nx_graph.get_edge_data(obodict['EGCG'][1], obodict['CYP3A4'][1])\n",
    "if edges:\n",
    "    for item in edges:\n",
    "        print(nodeLabels[str(item)])\n",
    "        print(edges[item])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "id": "3c854dcb-dc78-4dc4-9a22-a9ee7f256dd9",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "directly negatively regulates activity of\n",
      "{'predicate_key': '6c8b1cda2d2b3a1275555285284e3356', 'weight': 0.0, 'pmid': '29275296', 'timestamp': '2018 Jan-Feb', 'source_graph': 'machine_read', 'belief': 0.86}\n"
     ]
    }
   ],
   "source": [
    "edges = nx_graph.get_edge_data(obodict['EGCG'][0], obodict['CYP2C19'][0])\n",
    "if edges:\n",
    "    for item in edges:\n",
    "        print(nodeLabels[str(item)])\n",
    "        print(edges[item])\n",
    "edges = nx_graph.get_edge_data(obodict['EGCG'][1], obodict['CYP2C19'][0])\n",
    "if edges:\n",
    "    for item in edges:\n",
    "        print(nodeLabels[str(item)])\n",
    "        print(edges[item])\n",
    "edges = nx_graph.get_edge_data(obodict['EGCG'][0], obodict['CYP2C19'][1])\n",
    "if edges:\n",
    "    for item in edges:\n",
    "        print(nodeLabels[str(item)])\n",
    "        print(edges[item])\n",
    "edges = nx_graph.get_edge_data(obodict['EGCG'][1], obodict['CYP2C19'][1])\n",
    "if edges:\n",
    "    for item in edges:\n",
    "        print(nodeLabels[str(item)])\n",
    "        print(edges[item])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "id": "6694153c-1e42-4a97-9bf3-b24c7300c930",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "positively regulates\n",
      "{'predicate_key': '21515ea83ceaa7c5cbf4c5ec518fd8c1', 'weight': 0.0, 'pmid': '15629189', 'timestamp': '2005.0', 'source_graph': 'machine_read', 'belief': 0.8}\n",
      "directly negatively regulates activity of\n",
      "{'predicate_key': 'f098cb7f799d0373b0086f70b29998ef', 'weight': 0.0, 'pmid': '17490800', 'timestamp': '2007.0', 'source_graph': 'machine_read', 'belief': 0.8}\n",
      "interacts with\n",
      "{'predicate_key': 'cb2be9eab32ac6ccde3ad5a8202b186c', 'weight': 0.0, 'pmid': '17490800', 'timestamp': '2007.0', 'source_graph': 'machine_read_INF', 'belief': 0.8}\n"
     ]
    }
   ],
   "source": [
    "edges = nx_graph.get_edge_data(obodict['EGCG'][0], obodict['CYP2E1'][0])\n",
    "if edges:\n",
    "    for item in edges:\n",
    "        print(nodeLabels[str(item)])\n",
    "        print(edges[item])\n",
    "edges = nx_graph.get_edge_data(obodict['EGCG'][1], obodict['CYP2E1'][0])\n",
    "if edges:\n",
    "    for item in edges:\n",
    "        print(nodeLabels[str(item)])\n",
    "        print(edges[item])\n",
    "edges = nx_graph.get_edge_data(obodict['EGCG'][0], obodict['CYP2E1'][1])\n",
    "if edges:\n",
    "    for item in edges:\n",
    "        print(nodeLabels[str(item)])\n",
    "        print(edges[item])\n",
    "edges = nx_graph.get_edge_data(obodict['EGCG'][1], obodict['CYP2E1'][1])\n",
    "if edges:\n",
    "    for item in edges:\n",
    "        print(nodeLabels[str(item)])\n",
    "        print(edges[item])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "id": "663068fe-6c48-4499-a094-4c2f72965b35",
   "metadata": {},
   "outputs": [],
   "source": [
    "edges = nx_graph.get_edge_data(obodict['EGCG'][0], obodict['UGT'][0])\n",
    "if edges:\n",
    "    for item in edges:\n",
    "        print(nodeLabels[str(item)])\n",
    "        print(edges[item])\n",
    "edges = nx_graph.get_edge_data(obodict['EGCG'][1], obodict['UGT'][0])\n",
    "if edges:\n",
    "    for item in edges:\n",
    "        print(nodeLabels[str(item)])\n",
    "        print(edges[item])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "id": "fb373f03-28f0-4926-9ba4-24c530289321",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'entity_type': 'RELATIONS', 'label': 'IncreaseAmount'}\n",
      "{'predicate_key': '859ab4ce500a5fc5cf7c1090e1b99e54', 'weight': 0.0, 'pmid': '17657175', 'timestamp': '2007', 'source_graph': 'machine_read', 'belief': 0.65}\n"
     ]
    }
   ],
   "source": [
    "edges = nx_graph.get_edge_data(obodict['EGCG'][0], obodict['UGT1A8'][0])\n",
    "if edges:\n",
    "    for item in edges:\n",
    "        print(nodeLabels[str(item)])\n",
    "        print(edges[item])\n",
    "edges = nx_graph.get_edge_data(obodict['EGCG'][1], obodict['UGT1A8'][0])\n",
    "if edges:\n",
    "    for item in edges:\n",
    "        print(nodeLabels[str(item)])\n",
    "        print(edges[item])\n",
    "edges = nx_graph.get_edge_data(obodict['EGCG'][0], obodict['UGT1A8'][1])\n",
    "if edges:\n",
    "    for item in edges:\n",
    "        print(nodeLabels[str(item)])\n",
    "        print(edges[item])\n",
    "edges = nx_graph.get_edge_data(obodict['EGCG'][1], obodict['UGT1A8'][1])\n",
    "if edges:\n",
    "    for item in edges:\n",
    "        print(nodeLabels[str(item)])\n",
    "        print(edges[item])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "id": "2ac33163-b3af-4c83-8ab0-954357d9cb4f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "directly negatively regulates activity of\n",
      "{'predicate_key': '7ffd5f6ef41efd68711693dde83633d2', 'weight': 0.0, 'pmid': '21278283', 'timestamp': '2011 May', 'source_graph': 'machine_read', 'belief': 0.65}\n",
      "{'entity_type': 'RELATIONS', 'label': 'Activation'}\n",
      "{'predicate_key': '6578c214afe21ae7a260029628026e6e', 'weight': 0.0, 'pmid': '23327877', 'timestamp': '2013 Mar 22', 'source_graph': 'machine_read', 'belief': 0.949999985651093}\n",
      "directly negatively regulates activity of\n",
      "{'predicate_key': 'd92bbad7a657363b867659e6aa32b80b', 'weight': 0.0, 'pmid': '28118673', 'timestamp': '2017.0', 'source_graph': 'machine_read', 'belief': 0.8}\n",
      "positively regulates\n",
      "{'predicate_key': '3a53c5f9f3dbef1cbf4e2fe575ae8561', 'weight': 0.0, 'pmid': '28118673', 'timestamp': '2017.0', 'source_graph': 'machine_read', 'belief': 0.8}\n",
      "interacts with\n",
      "{'predicate_key': '7888647f27c3d7660ca8662b5d453ca4', 'weight': 0.0, 'pmid': '21278283', 'timestamp': '2011.0', 'source_graph': 'machine_read_INF', 'belief': 0.8}\n"
     ]
    }
   ],
   "source": [
    "edges = nx_graph.get_edge_data(obodict['EGCG'][0], obodict['OATP1B3'][0])\n",
    "if edges:\n",
    "    for item in edges:\n",
    "        print(nodeLabels[str(item)])\n",
    "        print(edges[item])\n",
    "edges = nx_graph.get_edge_data(obodict['EGCG'][1], obodict['OATP1B3'][0])\n",
    "if edges:\n",
    "    for item in edges:\n",
    "        print(nodeLabels[str(item)])\n",
    "        print(edges[item])\n",
    "edges = nx_graph.get_edge_data(obodict['EGCG'][0], obodict['OATP1B3'][1])\n",
    "if edges:\n",
    "    for item in edges:\n",
    "        print(nodeLabels[str(item)])\n",
    "        print(edges[item])\n",
    "edges = nx_graph.get_edge_data(obodict['EGCG'][1], obodict['OATP1B3'][1])\n",
    "if edges:\n",
    "    for item in edges:\n",
    "        print(nodeLabels[str(item)])\n",
    "        print(edges[item])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "id": "7c182257-dc9b-471a-a75e-ca83b6b81361",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "interacts with\n",
      "{'predicate_key': '90a9ba250c26eb2fff346a01179043ad', 'weight': 0.0, 'pmid': '26460146', 'timestamp': '2015.0', 'source_graph': 'machine_read_INF', 'belief': 0.8}\n"
     ]
    }
   ],
   "source": [
    "\n",
    "edges = nx_graph.get_edge_data(obodict['EGCG'][1], obo.CHEBI_17089)\n",
    "if edges:\n",
    "    for item in edges:\n",
    "        print(nodeLabels[str(item)])\n",
    "        print(edges[item])\n",
    "edges = nx_graph.get_edge_data(obodict['EGCG'][1], obo.PR_000037069)\n",
    "if edges:\n",
    "    for item in edges:\n",
    "        print(nodeLabels[str(item)])\n",
    "        print(edges[item])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 134,
   "id": "8f0e5e34-f0ee-4d72-b90d-3d2e80eea40f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "directly negatively regulates activity of\n",
      "{'predicate_key': '54bcb546faa1587a70554e63dbc07307', 'weight': 0.0, 'pmid': '29467215', 'timestamp': '2018.0', 'source_graph': 'machine_read', 'belief': 0.8}\n",
      "positively regulates\n",
      "{'predicate_key': '3cb3d030cf33323c53bb1ab022d4c8b5', 'weight': 0.0, 'source_graph': 'machine_read_INF'}\n"
     ]
    }
   ],
   "source": [
    "edges = nx_graph.get_edge_data(obodict['EGCG'][1], obodict['UGT'][0])\n",
    "if edges:\n",
    "    for item in edges:\n",
    "        print(nodeLabels[str(item)])\n",
    "        print(edges[item])\n",
    "edges = nx_graph.get_edge_data(obodict['EGCG'][1], obodict['UGT'][1])\n",
    "if edges:\n",
    "    for item in edges:\n",
    "        print(nodeLabels[str(item)])\n",
    "        print(edges[item])\n",
    "edges = nx_graph.get_edge_data(obodict['EGCG'][1], obodict['UGT'][2])\n",
    "if edges:\n",
    "    for item in edges:\n",
    "        print(nodeLabels[str(item)])\n",
    "        print(edges[item])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 162,
   "id": "09f218b5-daab-4d6b-aa2a-79175c79b595",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "805"
      ]
     },
     "execution_count": 162,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#how to get through these???\n",
    "len(get_neighbors_filtered(nx_graph, obodict['EGCG'][0]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 170,
   "id": "8c711e6f-d5ab-4a00-af4b-1af271af7b3b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "existence overlaps\n",
      "{'predicate_key': 'cc95487a2b0a7cce0323bafea41baf3b', 'weight': 0.0, 'pmid': '14511674', 'timestamp': '2003.0', 'source_graph': 'machine_read', 'belief': 0.8}\n",
      "directly negatively regulates activity of\n",
      "{'predicate_key': 'b3889a0b7b0cba37bc2e82841d41d1bb', 'weight': 0.0, 'pmid': '12499265', 'timestamp': '2002.0', 'source_graph': 'machine_read', 'belief': 0.8}\n"
     ]
    }
   ],
   "source": [
    "edges = nx_graph.get_edge_data(obodict['EGCG'][0], obodict['PGP'][3])\n",
    "if edges:\n",
    "    for item in edges:\n",
    "        print(nodeLabels[str(item)])\n",
    "        print(edges[item])\n",
    "edges = nx_graph.get_edge_data(obodict['EGCG'][1], obodict['PGP'][3])\n",
    "if edges:\n",
    "    for item in edges:\n",
    "        print(nodeLabels[str(item)])\n",
    "        print(edges[item])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 163,
   "id": "45cb615e-1b1f-4f0e-8cb6-50412672054c",
   "metadata": {},
   "outputs": [],
   "source": [
    "#216 results - saved to file\n",
    "\n",
    "df_neg.loc[df_neg['subject_obo'] == 'napdi_srs_imports:epigallocatechin_gallate'].to_csv('../output_files/EGCG_negated_edges_20220419.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "687144e8-8e2b-4036-a059-cf963e51f485",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bc4231bd-b86f-413f-8c4f-b88b51d1ad0e",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "e2935f66-daa5-431a-ac71-2771b06b76a0",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "#### ECG"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2eb0f105-732b-4792-ad89-1df5e82f0408",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 116,
   "id": "a070fa5b-a992-4002-a6c7-fd4523ca669f",
   "metadata": {},
   "outputs": [],
   "source": [
    "edges = nx_graph.get_edge_data(obodict['ECG'][0], obodict['CYP2E1'][0])\n",
    "if edges:\n",
    "    for item in edges:\n",
    "        print(nodeLabels[str(item)])\n",
    "        print(edges[item])\n",
    "edges = nx_graph.get_edge_data(obodict['ECG'][0], obodict['CYP2E1'][1])\n",
    "if edges:\n",
    "    for item in edges:\n",
    "        print(nodeLabels[str(item)])\n",
    "        print(edges[item])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "id": "3fcf82dc-e03e-4eee-9fd9-95ac8ec829bf",
   "metadata": {},
   "outputs": [],
   "source": [
    "edges = nx_graph.get_edge_data(obodict['ECG'][0], obodict['UGT1A8'][0])\n",
    "if edges:\n",
    "    for item in edges:\n",
    "        print(nodeLabels[str(item)])\n",
    "        print(edges[item])\n",
    "edges = nx_graph.get_edge_data(obodict['ECG'][0], obodict['UGT1A8'][1])\n",
    "if edges:\n",
    "    for item in edges:\n",
    "        print(nodeLabels[str(item)])\n",
    "        print(edges[item])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 127,
   "id": "7a18b57a-f988-40c8-9e18-e20b79b667c9",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "directly negatively regulates activity of\n",
      "{'predicate_key': 'cc72b22e0562a312256062d8d489ae8e', 'weight': 0.0, 'pmid': '29467215', 'timestamp': '2018 May', 'source_graph': 'machine_read', 'belief': 0.65}\n"
     ]
    }
   ],
   "source": [
    "edges = nx_graph.get_edge_data(obodict['ECG'][0], obo.PR_P78381)\n",
    "if edges:\n",
    "    for item in edges:\n",
    "        print(nodeLabels[str(item)])\n",
    "        print(edges[item])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 171,
   "id": "e0516ed0-f384-4bcc-a904-60a0b99e076e",
   "metadata": {},
   "outputs": [],
   "source": [
    "edges = nx_graph.get_edge_data(obodict['ECG'][0], obodict['PGP'][0])\n",
    "if edges:\n",
    "    for item in edges:\n",
    "        print(nodeLabels[str(item)])\n",
    "        print(edges[item])\n",
    "edges = nx_graph.get_edge_data(obodict['ECG'][0], obodict['PGP'][1])\n",
    "if edges:\n",
    "    for item in edges:\n",
    "        print(nodeLabels[str(item)])\n",
    "        print(edges[item])\n",
    "edges = nx_graph.get_edge_data(obodict['ECG'][0], obodict['PGP'][2])\n",
    "if edges:\n",
    "    for item in edges:\n",
    "        print(nodeLabels[str(item)])\n",
    "        print(edges[item])\n",
    "edges = nx_graph.get_edge_data(obodict['ECG'][0], obodict['PGP'][3])\n",
    "if edges:\n",
    "    for item in edges:\n",
    "        print(nodeLabels[str(item)])\n",
    "        print(edges[item])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 121,
   "id": "ca8bc7a7-82e9-4b09-9585-deee575ff5e6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[['(-)-epicatechin-3-O-gallate',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'Activation'},\n",
       "  'mitoxantrone',\n",
       "  'machine_read',\n",
       "  '2017 May'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  'directly negatively regulates activity of',\n",
       "  'episodic kinesigenic dyskinesia 1',\n",
       "  'machine_read',\n",
       "  '2019 Dec'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'Activation'},\n",
       "  'episodic kinesigenic dyskinesia 1',\n",
       "  'machine_read',\n",
       "  '2019 Dec'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'Activation'},\n",
       "  'transport',\n",
       "  'machine_read',\n",
       "  '2018 Dec 21'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  'directly negatively regulates activity of',\n",
       "  'transport',\n",
       "  'machine_read',\n",
       "  '2000 Nov'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  'directly negatively regulates activity of',\n",
       "  'SLCO1A2_gene',\n",
       "  'machine_read',\n",
       "  '2014 Apr'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  'directly negatively regulates activity of',\n",
       "  'sulfate',\n",
       "  'machine_read',\n",
       "  '2011 May'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'Activation'},\n",
       "  'etoposide',\n",
       "  'machine_read',\n",
       "  '2019 Feb 27'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'Activation'},\n",
       "  'proto-oncogene tyrosine-protein kinase Src',\n",
       "  'machine_read',\n",
       "  '2019 Dec'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'IncreaseAmount'},\n",
       "  'Glycine_dehydrogenase_(decarboxylating)',\n",
       "  'machine_read',\n",
       "  '2019 May 15'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  'directly negatively regulates activity of',\n",
       "  'SLC2A1_protein,_human',\n",
       "  'machine_read',\n",
       "  '2018 Dec 21'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'Activation'},\n",
       "  'SLC2A1_protein,_human',\n",
       "  'machine_read',\n",
       "  '2018 Dec 21'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'Activation'},\n",
       "  'sucrose',\n",
       "  'machine_read',\n",
       "  '2018 Dec 21'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  'directly negatively regulates activity of',\n",
       "  'sucrose',\n",
       "  'machine_read',\n",
       "  '2018 Dec 21'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'IncreaseAmount'},\n",
       "  'Adenosine_Triphosphatases',\n",
       "  'machine_read',\n",
       "  '2019 Dec'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  'directly negatively regulates activity of',\n",
       "  'Fluo-3',\n",
       "  'machine_read',\n",
       "  '2011 May'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  'directly negatively regulates activity of',\n",
       "  'potassium iodide',\n",
       "  'machine_read',\n",
       "  '2018 Dec 21'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'Activation'},\n",
       "  'potassium iodide',\n",
       "  'machine_read',\n",
       "  '2018 Dec 21'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'Activation'},\n",
       "  'Drug_resistance',\n",
       "  'machine_read',\n",
       "  '2010 Jul'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'Activation'},\n",
       "  'taurochenodeoxycholate 6alpha-hydroxylase activity',\n",
       "  'machine_read',\n",
       "  '2017 Apr'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  'directly negatively regulates activity of',\n",
       "  'taurochenodeoxycholate 6alpha-hydroxylase activity',\n",
       "  'machine_read',\n",
       "  '2001 Aug 8'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  'directly negatively regulates activity of',\n",
       "  'reactive oxygen species biosynthetic process',\n",
       "  'machine_read',\n",
       "  '2007 Oct'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'IncreaseAmount'},\n",
       "  'adverse event',\n",
       "  'machine_read',\n",
       "  '2019 Dec'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  'directly negatively regulates activity of',\n",
       "  'adverse event',\n",
       "  'machine_read',\n",
       "  '2000 Nov'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  'directly negatively regulates activity of',\n",
       "  'Neoplasm',\n",
       "  'machine_read',\n",
       "  '2003 Dec'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  'directly negatively regulates activity of',\n",
       "  'solute carrier organic anion transporter family member 1B1',\n",
       "  'machine_read',\n",
       "  '2011.0'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'Activation'},\n",
       "  'cell growth',\n",
       "  'machine_read',\n",
       "  '2008 Apr 15'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  'directly negatively regulates activity of',\n",
       "  'solute carrier organic anion transporter family member 1B3',\n",
       "  'machine_read',\n",
       "  '2011 May'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  'directly negatively regulates activity of',\n",
       "  'glucose transmembrane transport',\n",
       "  'machine_read',\n",
       "  '2007 Apr'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'Activation'},\n",
       "  'congenital myasthenic syndrome 5',\n",
       "  'machine_read',\n",
       "  '2013 Jan 1'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  'directly negatively regulates activity of',\n",
       "  'congenital myasthenic syndrome 5',\n",
       "  'machine_read',\n",
       "  '2013 Jan 1'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  'directly negatively regulates activity of',\n",
       "  'CLCA1_gene',\n",
       "  'machine_read',\n",
       "  '2015'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'Activation'},\n",
       "  'ATP hydrolysis activity',\n",
       "  'machine_read',\n",
       "  '2019 Dec'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  'directly negatively regulates activity of',\n",
       "  'conjugation',\n",
       "  'machine_read',\n",
       "  '2007 Oct'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'Activation'},\n",
       "  'tantalum atom',\n",
       "  'machine_read',\n",
       "  '2017 Apr'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  'directly negatively regulates activity of',\n",
       "  'glucose',\n",
       "  'machine_read',\n",
       "  '2000 Nov'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'IncreaseAmount'},\n",
       "  'glucose',\n",
       "  'machine_read',\n",
       "  '2009 Dec'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  'directly negatively regulates activity of',\n",
       "  'UDP-galactose translocator (human)',\n",
       "  'machine_read',\n",
       "  '2018 May'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  'directly negatively regulates activity of',\n",
       "  'estrone 3-sulfate',\n",
       "  'machine_read',\n",
       "  '2011 May'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'IncreaseAmount'},\n",
       "  'ABCB1_gene',\n",
       "  'machine_read',\n",
       "  '2010 Jul'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  'directly negatively regulates activity of',\n",
       "  'Cytochrome_P-450_CYP1A2',\n",
       "  'machine_read',\n",
       "  '2001 Aug 8'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'DecreaseAmount'},\n",
       "  'Cytochrome_P-450_CYP1A2',\n",
       "  'machine_read',\n",
       "  '2006 Sep'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  'directly negatively regulates activity of',\n",
       "  'cytochrome P450 1A1 (human)',\n",
       "  'machine_read',\n",
       "  '2001 Aug 8'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  'directly negatively regulates activity of',\n",
       "  'Bradycardia',\n",
       "  'machine_read',\n",
       "  '2013 Jan 1'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'Activation'},\n",
       "  'Bradycardia',\n",
       "  'machine_read',\n",
       "  '2013 Jan 1'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  'directly negatively regulates activity of',\n",
       "  'cystic fibrosis transmembrane conductance regulator',\n",
       "  'machine_read',\n",
       "  '2015'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  'directly negatively regulates activity of',\n",
       "  'catechin',\n",
       "  'machine_read',\n",
       "  '2001 Aug 8'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  'positively regulates',\n",
       "  'catechin',\n",
       "  'machine_read_INF'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  'directly negatively regulates activity of',\n",
       "  'Mus <genus>',\n",
       "  'machine_read',\n",
       "  '2003 Dec'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  'directly negatively regulates activity of',\n",
       "  '2,6-dichloroisonicotinic acid',\n",
       "  'machine_read',\n",
       "  '2013 Jan 1'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'Activation'},\n",
       "  '2,6-dichloroisonicotinic acid',\n",
       "  'machine_read',\n",
       "  '2013 Jan 1'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  'directly negatively regulates activity of',\n",
       "  'POU domain, class 2, transcription factor 2',\n",
       "  'machine_read',\n",
       "  '2017 Apr'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'Activation'},\n",
       "  '(-)-epigallocatechin 3-gallate',\n",
       "  'machine_read',\n",
       "  '2003 Oct 22'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'Activation'},\n",
       "  'Early_gastric_cancer',\n",
       "  'machine_read',\n",
       "  '2019 Feb 27'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  'directly negatively regulates activity of',\n",
       "  'hydrogen chloride',\n",
       "  'machine_read',\n",
       "  '2010 Nov'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  'directly negatively regulates activity of',\n",
       "  'anoctamin-1 (human)',\n",
       "  'machine_read',\n",
       "  '2015'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  'directly negatively regulates activity of',\n",
       "  'mutagen',\n",
       "  'machine_read',\n",
       "  '2001 Aug 8'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  'directly negatively regulates activity of',\n",
       "  'SLCO2B1_gene',\n",
       "  'machine_read',\n",
       "  '2011 May'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'Activation'},\n",
       "  'transmembrane transport',\n",
       "  'machine_read',\n",
       "  '2019 May 15'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  'directly negatively regulates activity of',\n",
       "  'cholera',\n",
       "  'machine_read',\n",
       "  '2015'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'Activation'},\n",
       "  'Constrictive median neuropathy',\n",
       "  'machine_read',\n",
       "  '2019 Dec'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  'directly negatively regulates activity of',\n",
       "  'fructose import',\n",
       "  'machine_read',\n",
       "  '2018 Apr'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'Activation'},\n",
       "  'cell death',\n",
       "  'machine_read',\n",
       "  '2010 Jul'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  'directly negatively regulates activity of',\n",
       "  'Leu-Asp-Ser',\n",
       "  'machine_read',\n",
       "  '2002 Sep 20'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'Activation'},\n",
       "  'APCS_gene',\n",
       "  'machine_read',\n",
       "  '2013 Jan 1'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  'directly negatively regulates activity of',\n",
       "  'glucose import',\n",
       "  'machine_read',\n",
       "  '2000 Nov'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  'directly negatively regulates activity of',\n",
       "  'lactate',\n",
       "  'machine_read',\n",
       "  '2019 Dec'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  'directly negatively regulates activity of',\n",
       "  'inhibitor',\n",
       "  'machine_read',\n",
       "  '2002 Sep 20'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  'directly negatively regulates activity of',\n",
       "  'Weight loss',\n",
       "  'machine_read',\n",
       "  '2003 Dec'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  'directly negatively regulates activity of',\n",
       "  '4-methylumbelliferone',\n",
       "  'machine_read',\n",
       "  '2018 May'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  'directly negatively regulates activity of',\n",
       "  'fructose',\n",
       "  'machine_read',\n",
       "  '2018 Apr'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  'directly negatively regulates activity of',\n",
       "  'solute carrier family 2, facilitated glucose transporter member 5 (human)',\n",
       "  'machine_read',\n",
       "  '2018 Apr'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'DecreaseAmount'},\n",
       "  'polypeptide',\n",
       "  'machine_read',\n",
       "  '2006 Sep'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'Activation'},\n",
       "  'sodium atom',\n",
       "  'machine_read',\n",
       "  '2013 Jan 1'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  'directly negatively regulates activity of',\n",
       "  'cutaneous fibrous histiocytoma',\n",
       "  'machine_read',\n",
       "  '2009 Nov'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  'directly negatively regulates activity of',\n",
       "  'aromatase',\n",
       "  'machine_read',\n",
       "  '2006 Sep'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'Activation'},\n",
       "  'ryanodine receptor 1',\n",
       "  'machine_read',\n",
       "  '2010 Aug 15'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'Activation'},\n",
       "  'UDP-glucuronosyltransferase 1A1 (human)',\n",
       "  'machine_read',\n",
       "  '2017 Apr'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  'directly negatively regulates activity of',\n",
       "  'SLC5A1_gene',\n",
       "  'machine_read',\n",
       "  '2000 Nov'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  'positively regulates',\n",
       "  'stachyose',\n",
       "  'machine_read',\n",
       "  '2016.0'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  'interacts with',\n",
       "  'antineoplastic agent',\n",
       "  'machine_read_INF',\n",
       "  '2020.0'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  'positively regulates',\n",
       "  'voltage-gated potassium channel subunit KCNQ5',\n",
       "  'machine_read',\n",
       "  '2021.0'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  'interacts with',\n",
       "  'voltage-gated potassium channel subunit KCNQ5',\n",
       "  'machine_read_INF',\n",
       "  '2021.0'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  'directly negatively regulates activity of',\n",
       "  'solute carrier organic anion transporter family member 2B1',\n",
       "  'machine_read',\n",
       "  '2011.0'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  'directly negatively regulates activity of',\n",
       "  'flavonoids',\n",
       "  'machine_read',\n",
       "  '2020.0'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  'directly negatively regulates activity of',\n",
       "  'cathepsin B',\n",
       "  'machine_read',\n",
       "  '2020.0'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  'interacts with',\n",
       "  'cytochrome P450',\n",
       "  'machine_read_INF',\n",
       "  '2006.0'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'augments'},\n",
       "  'chromaffin cell',\n",
       "  'machine_read',\n",
       "  '2007.0'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  'directly negatively regulates activity of',\n",
       "  'glycoprotein',\n",
       "  'machine_read',\n",
       "  '2020.0'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  'directly negatively regulates activity of',\n",
       "  'cytochrome P450 1A2',\n",
       "  'machine_read',\n",
       "  '2006.0'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  'directly negatively regulates activity of',\n",
       "  'solute carrier organic anion transporter family member 1A2',\n",
       "  'machine_read',\n",
       "  '2011.0'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  'interacts with',\n",
       "  '(-)-epigallocatechin gallate',\n",
       "  'machine_read_INF',\n",
       "  '2013.0'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  'positively regulates',\n",
       "  'protein kinase C epsilon',\n",
       "  'machine_read',\n",
       "  '2019.0'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  'directly negatively regulates activity of',\n",
       "  'procathepsin L',\n",
       "  'machine_read',\n",
       "  '2020.0'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  'directly negatively regulates activity of',\n",
       "  'transmembrane protease serine 2',\n",
       "  'machine_read',\n",
       "  '2020.0'],\n",
       " ['(-)-epicatechin-3-O-gallate',\n",
       "  'negatively regulates',\n",
       "  'import into cell',\n",
       "  'machine_read',\n",
       "  '2018.0'],\n",
       " ['(-)-epicatechin-3-O-gallate', 'has functional parent', 'gallic acid'],\n",
       " ['(-)-epicatechin-3-O-gallate', 'has functional parent', '(-)-epicatechin']]"
      ]
     },
     "execution_count": 121,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "get_neighbors_filtered(nx_graph, obodict['ECG'][0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 172,
   "id": "b78410ca-4924-4155-a738-03172f9f6255",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pmid</th>\n",
       "      <th>subject_cui</th>\n",
       "      <th>subject_type</th>\n",
       "      <th>object_cui</th>\n",
       "      <th>object_type</th>\n",
       "      <th>year</th>\n",
       "      <th>subject_name</th>\n",
       "      <th>predicate</th>\n",
       "      <th>object_name</th>\n",
       "      <th>source_text</th>\n",
       "      <th>Unnamed: 10</th>\n",
       "      <th>predicate_obo</th>\n",
       "      <th>subject_obo</th>\n",
       "      <th>object_obo</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1047</th>\n",
       "      <td>22969825</td>\n",
       "      <td>C0059422</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>C1148926</td>\n",
       "      <td>moft</td>\n",
       "      <td>2012.0</td>\n",
       "      <td>Epicatechin-3-Gallate</td>\n",
       "      <td>neg_disrupts</td>\n",
       "      <td>adenosinetriphosphatase activity</td>\n",
       "      <td>22969825_ascii.txt.tx.1 Of these, sennoside A,...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CHEBI_70255</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1137</th>\n",
       "      <td>14647974</td>\n",
       "      <td>C0059422</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>C0596235</td>\n",
       "      <td>elii,phsu</td>\n",
       "      <td>2004.0</td>\n",
       "      <td>Epicatechin-3-Gallate</td>\n",
       "      <td>neg_interacts_with</td>\n",
       "      <td>Calcium ion</td>\n",
       "      <td>14647974_ascii.txt.tx.1 EGCG has no effects on...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CHEBI_70255</td>\n",
       "      <td>CHEBI_39124</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          pmid subject_cui subject_type object_cui object_type    year  \\\n",
       "1047  22969825    C0059422    orch,phsu   C1148926        moft  2012.0   \n",
       "1137  14647974    C0059422    orch,phsu   C0596235   elii,phsu  2004.0   \n",
       "\n",
       "               subject_name           predicate  \\\n",
       "1047  Epicatechin-3-Gallate        neg_disrupts   \n",
       "1137  Epicatechin-3-Gallate  neg_interacts_with   \n",
       "\n",
       "                           object_name  \\\n",
       "1047  adenosinetriphosphatase activity   \n",
       "1137                       Calcium ion   \n",
       "\n",
       "                                            source_text  Unnamed: 10  \\\n",
       "1047  22969825_ascii.txt.tx.1 Of these, sennoside A,...          NaN   \n",
       "1137  14647974_ascii.txt.tx.1 EGCG has no effects on...          NaN   \n",
       "\n",
       "      predicate_obo  subject_obo   object_obo  \n",
       "1047            NaN  CHEBI_70255          NaN  \n",
       "1137            NaN  CHEBI_70255  CHEBI_39124  "
      ]
     },
     "execution_count": 172,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_neg.loc[df_neg['subject_obo'] == 'CHEBI_70255']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d8123d2f-9768-43ee-8999-7ce3e52f2e62",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "e9f5f3c3-941f-429f-bfff-8a7ea02bc5d0",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "#### Epicatechin"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 136,
   "id": "843ba469-c066-4406-8341-ca6f5e5c3ce3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "directly negatively regulates activity of\n",
      "{'predicate_key': '9284bde1b3f4fbdf0a5d44c3211e50c0', 'weight': 0.0, 'pmid': '20583320', 'timestamp': '2011.0', 'source_graph': 'machine_read', 'belief': 0.8}\n"
     ]
    }
   ],
   "source": [
    "edges = nx_graph.get_edge_data(obodict['epicatechin'][0], obodict['CYP1A1'][0])\n",
    "if edges:\n",
    "    for item in edges:\n",
    "        print(nodeLabels[str(item)])\n",
    "        print(edges[item])\n",
    "edges = nx_graph.get_edge_data(obodict['epicatechin'][0], obodict['CYP1A1'][1])\n",
    "if edges:\n",
    "    for item in edges:\n",
    "        print(nodeLabels[str(item)])\n",
    "        print(edges[item])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 140,
   "id": "d967625c-32ac-4835-99eb-d12654679399",
   "metadata": {},
   "outputs": [],
   "source": [
    "edges = nx_graph.get_edge_data(obodict['epicatechin'][0], obodict['UGT1A8'][0])\n",
    "if edges:\n",
    "    for item in edges:\n",
    "        print(nodeLabels[str(item)])\n",
    "        print(edges[item])\n",
    "edges = nx_graph.get_edge_data(obodict['epicatechin'][0], obodict['UGT1A8'][1])\n",
    "if edges:\n",
    "    for item in edges:\n",
    "        print(nodeLabels[str(item)])\n",
    "        print(edges[item])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 173,
   "id": "9c63015c-0d29-486d-b415-76d1cd84acd4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pmid</th>\n",
       "      <th>subject_cui</th>\n",
       "      <th>subject_type</th>\n",
       "      <th>object_cui</th>\n",
       "      <th>object_type</th>\n",
       "      <th>year</th>\n",
       "      <th>subject_name</th>\n",
       "      <th>predicate</th>\n",
       "      <th>object_name</th>\n",
       "      <th>source_text</th>\n",
       "      <th>Unnamed: 10</th>\n",
       "      <th>predicate_obo</th>\n",
       "      <th>subject_obo</th>\n",
       "      <th>object_obo</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>110</th>\n",
       "      <td>29193528</td>\n",
       "      <td>C0014485</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>C0596235</td>\n",
       "      <td>elii,phsu</td>\n",
       "      <td>2018.0</td>\n",
       "      <td>Epicatechin</td>\n",
       "      <td>neg_interacts_with</td>\n",
       "      <td>Calcium ion</td>\n",
       "      <td>29193528_ascii.txt.tx.2 Therefore, the follow-...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CHEBI_90</td>\n",
       "      <td>CHEBI_39124</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>111</th>\n",
       "      <td>29193528</td>\n",
       "      <td>C0014485</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>C3543842</td>\n",
       "      <td>phsu</td>\n",
       "      <td>2018.0</td>\n",
       "      <td>Epicatechin</td>\n",
       "      <td>neg_interacts_with</td>\n",
       "      <td>TONICS</td>\n",
       "      <td>29193528_ascii.txt.tx.1 3 | RESULTS 3.1 | The ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CHEBI_90</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>190</th>\n",
       "      <td>22015496</td>\n",
       "      <td>C0014485</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>C0030055</td>\n",
       "      <td>clna</td>\n",
       "      <td>2011.0</td>\n",
       "      <td>Epicatechin</td>\n",
       "      <td>neg_affects</td>\n",
       "      <td>Oxygen Consumption</td>\n",
       "      <td>22015496_ascii.txt.tx.1 Epicatechin, quercetin...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CHEBI_90</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>243</th>\n",
       "      <td>29205863</td>\n",
       "      <td>C0014485</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>C0174599</td>\n",
       "      <td>aapp,gngm,bacs</td>\n",
       "      <td>2018.0</td>\n",
       "      <td>Epicatechin</td>\n",
       "      <td>neg_interacts_with</td>\n",
       "      <td>SLC2A2 protein, human</td>\n",
       "      <td>29205863_ascii.txt.tx.3 Interestingly,  EC  an...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CHEBI_90</td>\n",
       "      <td>PR_P11168</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>244</th>\n",
       "      <td>29205863</td>\n",
       "      <td>C0014485</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>C1420201</td>\n",
       "      <td>gngm,aapp</td>\n",
       "      <td>2018.0</td>\n",
       "      <td>Epicatechin</td>\n",
       "      <td>neg_interacts_with</td>\n",
       "      <td>SLC5A2 gene</td>\n",
       "      <td>29205863_ascii.txt.tx.3 Interestingly,  EC  an...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CHEBI_90</td>\n",
       "      <td>PR_000015169</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>285</th>\n",
       "      <td>16978655</td>\n",
       "      <td>C0014485</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>C0007620</td>\n",
       "      <td>celf</td>\n",
       "      <td>2006.0</td>\n",
       "      <td>Epicatechin</td>\n",
       "      <td>neg_affects</td>\n",
       "      <td>Cell Survival</td>\n",
       "      <td>16978655_ascii.txt.tx.3 Catechin and epicatech...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CHEBI_90</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>624</th>\n",
       "      <td>17077187</td>\n",
       "      <td>C0014485</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>C0761993</td>\n",
       "      <td>aapp,gngm,bacs</td>\n",
       "      <td>2007.0</td>\n",
       "      <td>Epicatechin</td>\n",
       "      <td>neg_stimulates</td>\n",
       "      <td>ABCG2 protein, human</td>\n",
       "      <td>17077187_ascii.txt.tx.1 Monofunctional inducer...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CHEBI_90</td>\n",
       "      <td>PR_Q9UNQ0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>672</th>\n",
       "      <td>23252598</td>\n",
       "      <td>C0014485</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>C0034693</td>\n",
       "      <td>mamm</td>\n",
       "      <td>2013.0</td>\n",
       "      <td>Epicatechin</td>\n",
       "      <td>neg_administered_to</td>\n",
       "      <td>Rattus norvegicus</td>\n",
       "      <td>23252598_ascii.txt.tx.2 Capillarity was quanti...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CHEBI_90</td>\n",
       "      <td>NCBITaxon_10116</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>673</th>\n",
       "      <td>23252598</td>\n",
       "      <td>C0014485</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>C2699239</td>\n",
       "      <td>mamm</td>\n",
       "      <td>2013.0</td>\n",
       "      <td>Epicatechin</td>\n",
       "      <td>neg_administered_to</td>\n",
       "      <td>SD, Rat Strain</td>\n",
       "      <td>23252598_ascii.txt.tx.2 Capillarity was quanti...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CHEBI_90</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>675</th>\n",
       "      <td>23252598</td>\n",
       "      <td>C0014485</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>C0429627</td>\n",
       "      <td>fndg</td>\n",
       "      <td>2013.0</td>\n",
       "      <td>Epicatechin</td>\n",
       "      <td>neg_augments</td>\n",
       "      <td>Oxygen uptake</td>\n",
       "      <td>23252598_ascii.txt.tx.2 These results suggest ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CHEBI_90</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>676</th>\n",
       "      <td>23252598</td>\n",
       "      <td>C0014485</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>C0006903</td>\n",
       "      <td>npop</td>\n",
       "      <td>2013.0</td>\n",
       "      <td>Epicatechin</td>\n",
       "      <td>neg_affects</td>\n",
       "      <td>Capillarity</td>\n",
       "      <td>23252598_ascii.txt.tx.4 Since there are no emp...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CHEBI_90</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>916</th>\n",
       "      <td>17656102</td>\n",
       "      <td>C0014485</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>C0020517</td>\n",
       "      <td>patf</td>\n",
       "      <td>2007.0</td>\n",
       "      <td>Epicatechin</td>\n",
       "      <td>neg_affects</td>\n",
       "      <td>Hypersensitivity</td>\n",
       "      <td>17656102_ascii.txt.tx.3 Interestingly, except ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CHEBI_90</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>944</th>\n",
       "      <td>17298385</td>\n",
       "      <td>C0014485</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>C0031715</td>\n",
       "      <td>moft</td>\n",
       "      <td>2007.0</td>\n",
       "      <td>Epicatechin</td>\n",
       "      <td>neg_augments</td>\n",
       "      <td>Phosphorylation</td>\n",
       "      <td>17298385_ascii.txt.tx.1 In contrast at a conce...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CHEBI_90</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1024</th>\n",
       "      <td>28863367</td>\n",
       "      <td>C0014485</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>C0073591</td>\n",
       "      <td>antb,orch</td>\n",
       "      <td>2017.0</td>\n",
       "      <td>Epicatechin</td>\n",
       "      <td>neg_stimulates</td>\n",
       "      <td>rosoxacin</td>\n",
       "      <td>28863367_ascii.txt.tx.3 Furthermore, INS-1E ce...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CHEBI_90</td>\n",
       "      <td>CHEBI_131715</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1046</th>\n",
       "      <td>22969825</td>\n",
       "      <td>C0014485</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>C1148926</td>\n",
       "      <td>moft</td>\n",
       "      <td>2012.0</td>\n",
       "      <td>Epicatechin</td>\n",
       "      <td>neg_affects</td>\n",
       "      <td>adenosinetriphosphatase activity</td>\n",
       "      <td>22969825_ascii.txt.tx.3 Sennoside A, sennoside...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CHEBI_90</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1074</th>\n",
       "      <td>12127645</td>\n",
       "      <td>C0014485</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>C1881340</td>\n",
       "      <td>phsu</td>\n",
       "      <td>2002.0</td>\n",
       "      <td>Epicatechin</td>\n",
       "      <td>neg_interacts_with</td>\n",
       "      <td>L-Epicatechin</td>\n",
       "      <td>12127645_ascii.txt.tx.1 6B).No obvious effect ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CHEBI_90</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          pmid subject_cui subject_type object_cui     object_type    year  \\\n",
       "110   29193528    C0014485    orch,phsu   C0596235       elii,phsu  2018.0   \n",
       "111   29193528    C0014485    orch,phsu   C3543842            phsu  2018.0   \n",
       "190   22015496    C0014485    orch,phsu   C0030055            clna  2011.0   \n",
       "243   29205863    C0014485    orch,phsu   C0174599  aapp,gngm,bacs  2018.0   \n",
       "244   29205863    C0014485    orch,phsu   C1420201       gngm,aapp  2018.0   \n",
       "285   16978655    C0014485    orch,phsu   C0007620            celf  2006.0   \n",
       "624   17077187    C0014485    orch,phsu   C0761993  aapp,gngm,bacs  2007.0   \n",
       "672   23252598    C0014485    orch,phsu   C0034693            mamm  2013.0   \n",
       "673   23252598    C0014485    orch,phsu   C2699239            mamm  2013.0   \n",
       "675   23252598    C0014485    orch,phsu   C0429627            fndg  2013.0   \n",
       "676   23252598    C0014485    orch,phsu   C0006903            npop  2013.0   \n",
       "916   17656102    C0014485    orch,phsu   C0020517            patf  2007.0   \n",
       "944   17298385    C0014485    orch,phsu   C0031715            moft  2007.0   \n",
       "1024  28863367    C0014485    orch,phsu   C0073591       antb,orch  2017.0   \n",
       "1046  22969825    C0014485    orch,phsu   C1148926            moft  2012.0   \n",
       "1074  12127645    C0014485    orch,phsu   C1881340            phsu  2002.0   \n",
       "\n",
       "     subject_name            predicate                       object_name  \\\n",
       "110   Epicatechin   neg_interacts_with                       Calcium ion   \n",
       "111   Epicatechin   neg_interacts_with                            TONICS   \n",
       "190   Epicatechin          neg_affects                Oxygen Consumption   \n",
       "243   Epicatechin   neg_interacts_with             SLC2A2 protein, human   \n",
       "244   Epicatechin   neg_interacts_with                       SLC5A2 gene   \n",
       "285   Epicatechin          neg_affects                     Cell Survival   \n",
       "624   Epicatechin       neg_stimulates              ABCG2 protein, human   \n",
       "672   Epicatechin  neg_administered_to                 Rattus norvegicus   \n",
       "673   Epicatechin  neg_administered_to                    SD, Rat Strain   \n",
       "675   Epicatechin         neg_augments                     Oxygen uptake   \n",
       "676   Epicatechin          neg_affects                       Capillarity   \n",
       "916   Epicatechin          neg_affects                  Hypersensitivity   \n",
       "944   Epicatechin         neg_augments                   Phosphorylation   \n",
       "1024  Epicatechin       neg_stimulates                         rosoxacin   \n",
       "1046  Epicatechin          neg_affects  adenosinetriphosphatase activity   \n",
       "1074  Epicatechin   neg_interacts_with                     L-Epicatechin   \n",
       "\n",
       "                                            source_text  Unnamed: 10  \\\n",
       "110   29193528_ascii.txt.tx.2 Therefore, the follow-...          NaN   \n",
       "111   29193528_ascii.txt.tx.1 3 | RESULTS 3.1 | The ...          NaN   \n",
       "190   22015496_ascii.txt.tx.1 Epicatechin, quercetin...          NaN   \n",
       "243   29205863_ascii.txt.tx.3 Interestingly,  EC  an...          NaN   \n",
       "244   29205863_ascii.txt.tx.3 Interestingly,  EC  an...          NaN   \n",
       "285   16978655_ascii.txt.tx.3 Catechin and epicatech...          NaN   \n",
       "624   17077187_ascii.txt.tx.1 Monofunctional inducer...          NaN   \n",
       "672   23252598_ascii.txt.tx.2 Capillarity was quanti...          NaN   \n",
       "673   23252598_ascii.txt.tx.2 Capillarity was quanti...          NaN   \n",
       "675   23252598_ascii.txt.tx.2 These results suggest ...          NaN   \n",
       "676   23252598_ascii.txt.tx.4 Since there are no emp...          NaN   \n",
       "916   17656102_ascii.txt.tx.3 Interestingly, except ...          NaN   \n",
       "944   17298385_ascii.txt.tx.1 In contrast at a conce...          NaN   \n",
       "1024  28863367_ascii.txt.tx.3 Furthermore, INS-1E ce...          NaN   \n",
       "1046  22969825_ascii.txt.tx.3 Sennoside A, sennoside...          NaN   \n",
       "1074  12127645_ascii.txt.tx.1 6B).No obvious effect ...          NaN   \n",
       "\n",
       "      predicate_obo subject_obo       object_obo  \n",
       "110             NaN    CHEBI_90      CHEBI_39124  \n",
       "111             NaN    CHEBI_90              NaN  \n",
       "190             NaN    CHEBI_90              NaN  \n",
       "243             NaN    CHEBI_90        PR_P11168  \n",
       "244             NaN    CHEBI_90     PR_000015169  \n",
       "285             NaN    CHEBI_90              NaN  \n",
       "624             NaN    CHEBI_90        PR_Q9UNQ0  \n",
       "672             NaN    CHEBI_90  NCBITaxon_10116  \n",
       "673             NaN    CHEBI_90              NaN  \n",
       "675             NaN    CHEBI_90              NaN  \n",
       "676             NaN    CHEBI_90              NaN  \n",
       "916             NaN    CHEBI_90              NaN  \n",
       "944             NaN    CHEBI_90              NaN  \n",
       "1024            NaN    CHEBI_90     CHEBI_131715  \n",
       "1046            NaN    CHEBI_90              NaN  \n",
       "1074            NaN    CHEBI_90              NaN  "
      ]
     },
     "execution_count": 173,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_neg.loc[df_neg['subject_obo'] == 'CHEBI_90']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 141,
   "id": "6d03efc8-b05e-409d-8b64-12796b4fb80b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[['(-)-epicatechin',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'Activation'},\n",
       "  'Peroxidase',\n",
       "  'machine_read',\n",
       "  '2011 Dec'],\n",
       " ['(-)-epicatechin',\n",
       "  'directly negatively regulates activity of',\n",
       "  'Cytochromes_c',\n",
       "  'machine_read',\n",
       "  '2011 Dec'],\n",
       " ['(-)-epicatechin',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'Activation'},\n",
       "  'Cytochromes_c',\n",
       "  'machine_read',\n",
       "  '2011 Dec'],\n",
       " ['(-)-epicatechin',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'Activation'},\n",
       "  'time',\n",
       "  'machine_read',\n",
       "  '2015 Jan'],\n",
       " ['(-)-epicatechin',\n",
       "  'directly negatively regulates activity of',\n",
       "  'ubiquinone-1',\n",
       "  'machine_read',\n",
       "  '2011 Dec'],\n",
       " ['(-)-epicatechin',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'Activation'},\n",
       "  'transport',\n",
       "  'machine_read',\n",
       "  '2008 Feb'],\n",
       " ['(-)-epicatechin',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'Activation'},\n",
       "  'N(gamma)-nitro-L-arginine methyl ester',\n",
       "  'machine_read',\n",
       "  '1999 Apr 19'],\n",
       " ['(-)-epicatechin',\n",
       "  'positively regulates',\n",
       "  'N(gamma)-nitro-L-arginine methyl ester',\n",
       "  'machine_read',\n",
       "  '1999.0'],\n",
       " ['(-)-epicatechin',\n",
       "  'directly negatively regulates activity of',\n",
       "  'NADH_dehydrogenase_(ubiquinone)',\n",
       "  'machine_read',\n",
       "  '2011 Dec'],\n",
       " ['(-)-epicatechin',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'IncreaseAmount'},\n",
       "  'GRIA2_gene',\n",
       "  'machine_read',\n",
       "  '2007 Jun'],\n",
       " ['(-)-epicatechin',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'Activation'},\n",
       "  'GRIA2_gene',\n",
       "  'machine_read',\n",
       "  '2007 Jun'],\n",
       " ['(-)-epicatechin',\n",
       "  'directly negatively regulates activity of',\n",
       "  'Cytochrome_P-450_CYP2E1',\n",
       "  'machine_read',\n",
       "  '2021 Jun'],\n",
       " ['(-)-epicatechin',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'Activation'},\n",
       "  'Leu-Asp-Ser',\n",
       "  'machine_read',\n",
       "  '2002 Sep 20'],\n",
       " ['(-)-epicatechin',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'DecreaseAmount'},\n",
       "  'etoposide',\n",
       "  'machine_read',\n",
       "  '2004 Feb 1'],\n",
       " ['(-)-epicatechin',\n",
       "  'directly negatively regulates activity of',\n",
       "  'sodium atom',\n",
       "  'machine_read',\n",
       "  '2001'],\n",
       " ['(-)-epicatechin',\n",
       "  'directly negatively regulates activity of',\n",
       "  'phosphatidic acid',\n",
       "  'machine_read',\n",
       "  '2009 Aug'],\n",
       " ['(-)-epicatechin',\n",
       "  'directly negatively regulates activity of',\n",
       "  'reactive oxygen species biosynthetic process',\n",
       "  'machine_read',\n",
       "  '2006 Nov 10'],\n",
       " ['(-)-epicatechin',\n",
       "  'directly negatively regulates activity of',\n",
       "  'solute carrier family 22 member 8 (human)',\n",
       "  'machine_read',\n",
       "  '2021 Feb 16'],\n",
       " ['(-)-epicatechin',\n",
       "  'directly negatively regulates activity of',\n",
       "  'calcium(2+)',\n",
       "  'machine_read',\n",
       "  '2006 Nov 10'],\n",
       " ['(-)-epicatechin',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'Activation'},\n",
       "  'vasodilation',\n",
       "  'machine_read',\n",
       "  '2019 Jan 16'],\n",
       " ['(-)-epicatechin',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'Activation'},\n",
       "  'adverse event',\n",
       "  'machine_read',\n",
       "  '2019 Jan 16'],\n",
       " ['(-)-epicatechin',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'Activation'},\n",
       "  'relaxation of smooth muscle',\n",
       "  'machine_read',\n",
       "  '1999 Apr 19'],\n",
       " ['(-)-epicatechin',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'Activation'},\n",
       "  'chemical synaptic transmission',\n",
       "  'machine_read',\n",
       "  '2007 Jun'],\n",
       " ['(-)-epicatechin',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'Activation'},\n",
       "  'GADD45A_gene',\n",
       "  'machine_read',\n",
       "  '2014'],\n",
       " ['(-)-epicatechin',\n",
       "  'directly negatively regulates activity of',\n",
       "  'apoptotic process',\n",
       "  'machine_read',\n",
       "  '2006 Nov 10'],\n",
       " ['(-)-epicatechin',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'Activation'},\n",
       "  'apoptotic process',\n",
       "  'machine_read',\n",
       "  '2014'],\n",
       " ['(-)-epicatechin',\n",
       "  'directly negatively regulates activity of',\n",
       "  'Cytochrome_P-450_CYP1A2',\n",
       "  'machine_read',\n",
       "  '2012'],\n",
       " ['(-)-epicatechin',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'Activation'},\n",
       "  'DDIT3_gene',\n",
       "  'machine_read',\n",
       "  '2014'],\n",
       " ['(-)-epicatechin',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'Activation'},\n",
       "  \"3',5'-cyclic GMP\",\n",
       "  'machine_read',\n",
       "  '1999 Apr 19'],\n",
       " ['(-)-epicatechin',\n",
       "  'positively regulates',\n",
       "  \"3',5'-cyclic GMP\",\n",
       "  'machine_read_INF',\n",
       "  '1999.0'],\n",
       " ['(-)-epicatechin',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'Activation'},\n",
       "  'calcium ion',\n",
       "  'machine_read',\n",
       "  '2003 Mar 7'],\n",
       " ['(-)-epicatechin',\n",
       "  'positively regulates',\n",
       "  'calcium ion',\n",
       "  'machine_read_INF',\n",
       "  '2018.0'],\n",
       " ['(-)-epicatechin',\n",
       "  'directly negatively regulates activity of',\n",
       "  'calcium ion',\n",
       "  'machine_read',\n",
       "  '2006.0'],\n",
       " ['(-)-epicatechin',\n",
       "  'directly negatively regulates activity of',\n",
       "  'fructose import',\n",
       "  'machine_read',\n",
       "  '2018 Apr'],\n",
       " ['(-)-epicatechin',\n",
       "  'directly negatively regulates activity of',\n",
       "  'NADH',\n",
       "  'machine_read',\n",
       "  '2011 Dec'],\n",
       " ['(-)-epicatechin',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'Activation'},\n",
       "  '3-(4,5-dimethylthiazol-2-yl)-2,5-diphenyltetrazolium bromide',\n",
       "  'machine_read',\n",
       "  '2006 Nov 10'],\n",
       " ['(-)-epicatechin',\n",
       "  'directly negatively regulates activity of',\n",
       "  'doxorubicin',\n",
       "  'machine_read',\n",
       "  '2004 May 28'],\n",
       " ['(-)-epicatechin',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'Activation'},\n",
       "  'doxorubicin',\n",
       "  'machine_read',\n",
       "  '2018 Nov 26'],\n",
       " ['(-)-epicatechin',\n",
       "  'directly negatively regulates activity of',\n",
       "  'DNA-Directed_DNA_Polymerase',\n",
       "  'machine_read',\n",
       "  '2015 Jul 17'],\n",
       " ['(-)-epicatechin',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'Activation'},\n",
       "  'peroxisome biogenesis factor 2 (human)',\n",
       "  'machine_read',\n",
       "  '2018 Feb'],\n",
       " ['(-)-epicatechin',\n",
       "  'directly negatively regulates activity of',\n",
       "  'neuron death',\n",
       "  'machine_read',\n",
       "  '2006 Nov 10'],\n",
       " ['(-)-epicatechin',\n",
       "  'directly negatively regulates activity of',\n",
       "  'SLCO2B1_gene',\n",
       "  'machine_read',\n",
       "  '2019 Sep'],\n",
       " ['(-)-epicatechin',\n",
       "  'directly negatively regulates activity of',\n",
       "  'Central sleep apnea',\n",
       "  'machine_read',\n",
       "  '2014'],\n",
       " ['(-)-epicatechin',\n",
       "  'directly negatively regulates activity of',\n",
       "  'tube formation',\n",
       "  'machine_read',\n",
       "  '2017 Jan 20'],\n",
       " ['(-)-epicatechin',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'Activation'},\n",
       "  'Adenosine_Triphosphatases',\n",
       "  'machine_read',\n",
       "  '2008 Oct'],\n",
       " ['(-)-epicatechin',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'Activation'},\n",
       "  'ketoconazole',\n",
       "  'machine_read',\n",
       "  '2008 Feb'],\n",
       " ['(-)-epicatechin',\n",
       "  'directly negatively regulates activity of',\n",
       "  'aromatase',\n",
       "  'machine_read',\n",
       "  '2006 May 17'],\n",
       " ['(-)-epicatechin',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'Activation'},\n",
       "  'death',\n",
       "  'machine_read',\n",
       "  '2006 Nov 10'],\n",
       " ['(-)-epicatechin',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'Activation'},\n",
       "  'phenylephrine',\n",
       "  'machine_read',\n",
       "  '2018 Feb'],\n",
       " ['(-)-epicatechin',\n",
       "  'directly negatively regulates activity of',\n",
       "  'angiogenesis',\n",
       "  'machine_read',\n",
       "  '2015 Jul 17'],\n",
       " ['(-)-epicatechin',\n",
       "  'directly negatively regulates activity of',\n",
       "  'isoprenaline',\n",
       "  'machine_read',\n",
       "  '2013 Mar'],\n",
       " ['(-)-epicatechin',\n",
       "  'positively regulates',\n",
       "  'isoprenaline',\n",
       "  'machine_read',\n",
       "  '2013.0'],\n",
       " ['(-)-epicatechin',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'IncreaseAmount'},\n",
       "  'voltage-gated potassium channel KCNH2',\n",
       "  'machine_read',\n",
       "  '2019 Apr 1'],\n",
       " ['(-)-epicatechin',\n",
       "  'positively regulates',\n",
       "  'voltage-gated potassium channel KCNH2',\n",
       "  'machine_read',\n",
       "  '2019.0'],\n",
       " ['(-)-epicatechin',\n",
       "  'directly negatively regulates activity of',\n",
       "  'Laccase',\n",
       "  'machine_read',\n",
       "  '2021 Jan 13'],\n",
       " ['(-)-epicatechin',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'Activation'},\n",
       "  'generation of precursor metabolites and energy',\n",
       "  'machine_read',\n",
       "  '2012 Apr'],\n",
       " ['(-)-epicatechin',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'DecreaseAmount'},\n",
       "  'hydrogen chloride',\n",
       "  'machine_read',\n",
       "  '2005 Oct'],\n",
       " ['(-)-epicatechin',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'Activation'},\n",
       "  \"3',5'-cyclic AMP\",\n",
       "  'machine_read',\n",
       "  '2007 Jun'],\n",
       " ['(-)-epicatechin',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'Activation'},\n",
       "  'Mus <genus>',\n",
       "  'machine_read',\n",
       "  '2012 Nov 15'],\n",
       " ['(-)-epicatechin',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'Activation'},\n",
       "  'mitogen-activated protein kinase 3',\n",
       "  'machine_read',\n",
       "  '2007 Jun'],\n",
       " ['(-)-epicatechin',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'IncreaseAmount'},\n",
       "  'mitogen-activated protein kinase 3',\n",
       "  'machine_read',\n",
       "  '2007 Jun'],\n",
       " ['(-)-epicatechin',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'Activation'},\n",
       "  'phosphatidylinositol',\n",
       "  'machine_read',\n",
       "  '2007 Jun'],\n",
       " ['(-)-epicatechin',\n",
       "  'directly negatively regulates activity of',\n",
       "  'sodium/hydrogen exchanger 1 (human)',\n",
       "  'machine_read',\n",
       "  '2001'],\n",
       " ['(-)-epicatechin',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'Activation'},\n",
       "  'gene expression',\n",
       "  'machine_read',\n",
       "  '2007 Jun'],\n",
       " ['(-)-epicatechin',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'Activation'},\n",
       "  'synapse assembly',\n",
       "  'machine_read',\n",
       "  '2007 Jun'],\n",
       " ['(-)-epicatechin',\n",
       "  'directly negatively regulates activity of',\n",
       "  'caspase-3',\n",
       "  'machine_read',\n",
       "  '2006 Nov 10'],\n",
       " ['(-)-epicatechin',\n",
       "  'positively regulates',\n",
       "  'caspase-3',\n",
       "  'machine_read_INF',\n",
       "  '2006.0'],\n",
       " ['(-)-epicatechin',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'IncreaseAmount'},\n",
       "  'taurochenodeoxycholate 6alpha-hydroxylase activity',\n",
       "  'machine_read',\n",
       "  '2006 May'],\n",
       " ['(-)-epicatechin',\n",
       "  'positively regulates',\n",
       "  'glucose',\n",
       "  'machine_read',\n",
       "  '2007.0'],\n",
       " ['(-)-epicatechin',\n",
       "  'interacts with',\n",
       "  'glucose',\n",
       "  'machine_read_INF',\n",
       "  '2014.0'],\n",
       " ['(-)-epicatechin',\n",
       "  'interacts with',\n",
       "  'insulin',\n",
       "  'machine_read_INF',\n",
       "  '2001.0'],\n",
       " ['(-)-epicatechin', 'positively regulates', 'insulin', 'machine_read_INF'],\n",
       " ['(-)-epicatechin',\n",
       "  'interacts with',\n",
       "  'canalicular multispecific organic anion transporter 1',\n",
       "  'machine_read_INF',\n",
       "  '2003.0'],\n",
       " ['(-)-epicatechin',\n",
       "  'directly negatively regulates activity of',\n",
       "  'cytochrome P450 1A1',\n",
       "  'machine_read',\n",
       "  '2011.0'],\n",
       " ['(-)-epicatechin',\n",
       "  'positively regulates',\n",
       "  'cytochrome c',\n",
       "  'machine_read',\n",
       "  '2021.0'],\n",
       " ['(-)-epicatechin',\n",
       "  'interacts with',\n",
       "  'cytochrome c',\n",
       "  'machine_read_INF',\n",
       "  '2011.0'],\n",
       " ['(-)-epicatechin',\n",
       "  'interacts with',\n",
       "  'catechin',\n",
       "  'machine_read_INF',\n",
       "  '2001.0'],\n",
       " ['(-)-epicatechin', 'positively regulates', 'catechin', 'machine_read_INF'],\n",
       " ['(-)-epicatechin',\n",
       "  'directly negatively regulates activity of',\n",
       "  'gamma-aminobutyric acid',\n",
       "  'machine_read',\n",
       "  '2015.0'],\n",
       " ['(-)-epicatechin',\n",
       "  'positively regulates',\n",
       "  '4-methylumbelliferone',\n",
       "  'machine_read',\n",
       "  '2003.0'],\n",
       " ['(-)-epicatechin',\n",
       "  'interacts with',\n",
       "  '(-)-epigallocatechin gallate',\n",
       "  'machine_read_INF',\n",
       "  '2011.0'],\n",
       " ['(-)-epicatechin',\n",
       "  'positively regulates',\n",
       "  '(-)-epigallocatechin gallate',\n",
       "  'machine_read_INF'],\n",
       " ['(-)-epicatechin',\n",
       "  'directly negatively regulates activity of',\n",
       "  {'entity_type': 'NODES', 'label': 'ABCG2 gene', 'cui': 'C1332002'},\n",
       "  'machine_read',\n",
       "  '2010.0'],\n",
       " ['(-)-epicatechin',\n",
       "  'interacts with',\n",
       "  'succinate--CoA ligase [ADP-forming] subunit beta, mitochondrial',\n",
       "  'machine_read_INF',\n",
       "  '2006.0'],\n",
       " ['(-)-epicatechin',\n",
       "  'directly negatively regulates activity of',\n",
       "  'succinate--CoA ligase [ADP-forming] subunit beta, mitochondrial',\n",
       "  'machine_read',\n",
       "  '2006.0'],\n",
       " ['(-)-epicatechin',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'affects'},\n",
       "  'epithelial cell',\n",
       "  'machine_read',\n",
       "  '2008.0'],\n",
       " ['(-)-epicatechin',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'affects'},\n",
       "  'diabetes mellitus',\n",
       "  'machine_read',\n",
       "  '2013.0'],\n",
       " ['(-)-epicatechin',\n",
       "  'interacts with',\n",
       "  \"5'-AMP-activated protein kinase catalytic subunit alpha-1\",\n",
       "  'machine_read_INF',\n",
       "  '2014.0'],\n",
       " ['(-)-epicatechin',\n",
       "  'positively regulates',\n",
       "  \"5'-AMP-activated protein kinase catalytic subunit alpha-1\",\n",
       "  'machine_read_INF'],\n",
       " ['(-)-epicatechin',\n",
       "  'directly negatively regulates activity of',\n",
       "  'hydroxyl',\n",
       "  'machine_read',\n",
       "  '2011.0'],\n",
       " ['(-)-epicatechin',\n",
       "  'positively regulates',\n",
       "  'neuronitis',\n",
       "  'machine_read',\n",
       "  '2006.0'],\n",
       " ['(-)-epicatechin',\n",
       "  'directly negatively regulates activity of',\n",
       "  'superoxide',\n",
       "  'machine_read',\n",
       "  '2011.0'],\n",
       " ['(-)-epicatechin',\n",
       "  'positively regulates',\n",
       "  'apoptosis regulator BAX',\n",
       "  'machine_read_INF',\n",
       "  '2021.0'],\n",
       " ['(-)-epicatechin',\n",
       "  'negatively regulates',\n",
       "  'import into cell',\n",
       "  'machine_read',\n",
       "  '2013.0'],\n",
       " ['(-)-epicatechin',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'augments'},\n",
       "  'import into cell',\n",
       "  'machine_read',\n",
       "  '2015.0'],\n",
       " ['(-)-epicatechin',\n",
       "  'directly negatively regulates activity of',\n",
       "  'sodium/hydrogen exchanger 1',\n",
       "  'machine_read',\n",
       "  '2001.0'],\n",
       " ['(-)-epicatechin',\n",
       "  'positively regulates',\n",
       "  'sodium/hydrogen exchanger 1',\n",
       "  'machine_read_INF'],\n",
       " ['(-)-epicatechin',\n",
       "  'positively regulates',\n",
       "  '4-aminopyridine',\n",
       "  'machine_read',\n",
       "  '2018.0'],\n",
       " ['(-)-epicatechin',\n",
       "  'interacts with',\n",
       "  '4-aminopyridine',\n",
       "  'machine_read_INF',\n",
       "  '2015.0'],\n",
       " ['(-)-epicatechin',\n",
       "  'interacts with',\n",
       "  'fatty acid',\n",
       "  'machine_read_INF',\n",
       "  '2015.0'],\n",
       " ['(-)-epicatechin',\n",
       "  'interacts with',\n",
       "  'thrombospondin-1',\n",
       "  'machine_read_INF',\n",
       "  '2012.0'],\n",
       " ['(-)-epicatechin',\n",
       "  'positively regulates',\n",
       "  'thrombospondin-1',\n",
       "  'machine_read_INF',\n",
       "  '2013.0'],\n",
       " ['(-)-epicatechin',\n",
       "  'interacts with',\n",
       "  'vascular endothelial growth factor A',\n",
       "  'machine_read_INF',\n",
       "  '2012.0'],\n",
       " ['(-)-epicatechin',\n",
       "  'positively regulates',\n",
       "  'vascular endothelial growth factor A',\n",
       "  'machine_read',\n",
       "  '2013.0'],\n",
       " ['(-)-epicatechin',\n",
       "  'directly negatively regulates activity of',\n",
       "  'dopamine receptor',\n",
       "  'machine_read',\n",
       "  '2015.0'],\n",
       " ['(-)-epicatechin',\n",
       "  'directly negatively regulates activity of',\n",
       "  'cytochrome P450 2E1',\n",
       "  'machine_read',\n",
       "  '2021.0'],\n",
       " ['(-)-epicatechin',\n",
       "  'interacts with',\n",
       "  'ellagic acid',\n",
       "  'machine_read_INF',\n",
       "  '2015.0'],\n",
       " ['(-)-epicatechin',\n",
       "  'interacts with',\n",
       "  'phenolate',\n",
       "  'machine_read_INF',\n",
       "  '2019.0'],\n",
       " ['(-)-epicatechin',\n",
       "  'directly negatively regulates activity of',\n",
       "  'rhodamine 123',\n",
       "  'machine_read',\n",
       "  '2020.0'],\n",
       " ['(-)-epicatechin',\n",
       "  'positively regulates',\n",
       "  'glutamate receptor 2',\n",
       "  'machine_read',\n",
       "  '2007.0'],\n",
       " ['(-)-epicatechin',\n",
       "  'interacts with',\n",
       "  'flavonoids',\n",
       "  'machine_read_INF',\n",
       "  '2015.0'],\n",
       " ['(-)-epicatechin', 'positively regulates', 'flavonoids', 'machine_read_INF'],\n",
       " ['(-)-epicatechin',\n",
       "  'positively regulates',\n",
       "  'arginine',\n",
       "  'machine_read',\n",
       "  '1999.0'],\n",
       " ['(-)-epicatechin',\n",
       "  'interacts with',\n",
       "  'procyanidin',\n",
       "  'machine_read_INF',\n",
       "  '2016.0'],\n",
       " ['(-)-epicatechin',\n",
       "  'directly negatively regulates activity of',\n",
       "  'cytochrome P450',\n",
       "  'machine_read',\n",
       "  '2003.0'],\n",
       " ['(-)-epicatechin',\n",
       "  'positively regulates',\n",
       "  'stachyose',\n",
       "  'machine_read',\n",
       "  '2016.0'],\n",
       " ['(-)-epicatechin',\n",
       "  'directly negatively regulates activity of',\n",
       "  'anion',\n",
       "  'machine_read',\n",
       "  '2007.0'],\n",
       " ['(-)-epicatechin',\n",
       "  'interacts with',\n",
       "  'anthocyanin',\n",
       "  'machine_read_INF',\n",
       "  '2009.0'],\n",
       " ['(-)-epicatechin',\n",
       "  'interacts with',\n",
       "  'ephrin type-B receptor 2 (human)',\n",
       "  'machine_read_INF',\n",
       "  '2007.0'],\n",
       " ['(-)-epicatechin',\n",
       "  'positively regulates',\n",
       "  'ephrin type-B receptor 2 (human)',\n",
       "  'machine_read',\n",
       "  '2007.0'],\n",
       " ['(-)-epicatechin',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'augments'},\n",
       "  'central core myopathy',\n",
       "  'machine_read',\n",
       "  '2013.0'],\n",
       " ['(-)-epicatechin',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'affects'},\n",
       "  'herpes simplex infectious disease',\n",
       "  'machine_read',\n",
       "  '2018.0'],\n",
       " ['(-)-epicatechin',\n",
       "  'directly negatively regulates activity of',\n",
       "  'testosterone',\n",
       "  'machine_read',\n",
       "  '2013.0'],\n",
       " ['(-)-epicatechin',\n",
       "  'positively regulates',\n",
       "  'nitric oxide',\n",
       "  'machine_read',\n",
       "  '2015.0'],\n",
       " ['(-)-epicatechin', 'interacts with', 'acid', 'machine_read_INF', '2002.0'],\n",
       " ['(-)-epicatechin',\n",
       "  'directly negatively regulates activity of',\n",
       "  'mutagen',\n",
       "  'machine_read',\n",
       "  '2003.0'],\n",
       " ['(-)-epicatechin',\n",
       "  'interacts with',\n",
       "  {'entity_type': 'NODES',\n",
       "   'label': 'Phosphoenolpyruvate Carboxylase',\n",
       "   'cui': 'C0031644'},\n",
       "  'machine_read_INF',\n",
       "  '2013.0'],\n",
       " ['(-)-epicatechin',\n",
       "  'positively regulates',\n",
       "  {'entity_type': 'NODES',\n",
       "   'label': 'Phosphoenolpyruvate Carboxylase',\n",
       "   'cui': 'C0031644'},\n",
       "  'machine_read_INF'],\n",
       " ['(-)-epicatechin',\n",
       "  'interacts with',\n",
       "  'nanos homolog 3 (human)',\n",
       "  'machine_read_INF',\n",
       "  '2020.0'],\n",
       " ['(-)-epicatechin',\n",
       "  'positively regulates',\n",
       "  'nanos homolog 3 (human)',\n",
       "  'machine_read_INF'],\n",
       " ['(-)-epicatechin',\n",
       "  'interacts with',\n",
       "  'ATP-sensitive inward rectifier potassium channel 11',\n",
       "  'machine_read_INF',\n",
       "  '2005.0'],\n",
       " ['(-)-epicatechin',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'affects'},\n",
       "  'saphenous vein',\n",
       "  'machine_read',\n",
       "  '2018.0'],\n",
       " ['(-)-epicatechin',\n",
       "  'interacts with',\n",
       "  'hydrogen peroxide',\n",
       "  'machine_read_INF',\n",
       "  '2011.0'],\n",
       " ['(-)-epicatechin',\n",
       "  'directly negatively regulates activity of',\n",
       "  'hydrogen peroxide',\n",
       "  'machine_read',\n",
       "  '2011.0'],\n",
       " ['(-)-epicatechin',\n",
       "  'directly negatively regulates activity of',\n",
       "  '2-thiobarbituric acid',\n",
       "  'machine_read',\n",
       "  '2013.0'],\n",
       " ['(-)-epicatechin',\n",
       "  'interacts with',\n",
       "  'narrow pore, gated channel activity',\n",
       "  'machine_read_INF',\n",
       "  '2011.0'],\n",
       " ['(-)-epicatechin',\n",
       "  'interacts with',\n",
       "  'pyrimidine',\n",
       "  'machine_read_INF',\n",
       "  '2011.0'],\n",
       " ['(-)-epicatechin',\n",
       "  'positively regulates',\n",
       "  'AMPA glutamate receptor complex',\n",
       "  'machine_read',\n",
       "  '2007.0'],\n",
       " ['(-)-epicatechin',\n",
       "  'directly negatively regulates activity of',\n",
       "  'kaempferol',\n",
       "  'machine_read',\n",
       "  '2011.0'],\n",
       " ['(-)-epicatechin',\n",
       "  'interacts with',\n",
       "  'voltage-gated potassium channel subunit KCNQ5',\n",
       "  'machine_read_INF',\n",
       "  '2021.0'],\n",
       " ['(-)-epicatechin', 'positively regulates', 'CRE', 'machine_read', '2007.0'],\n",
       " ['(-)-epicatechin',\n",
       "  'interacts with',\n",
       "  'hypoglycemic agent',\n",
       "  'machine_read_INF',\n",
       "  '2004.0'],\n",
       " ['(-)-epicatechin',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'affects'},\n",
       "  'neuron',\n",
       "  'machine_read',\n",
       "  '2007.0'],\n",
       " ['(-)-epicatechin', 'positively regulates', 'neuron', 'machine_read_INF'],\n",
       " ['(-)-epicatechin',\n",
       "  'positively regulates',\n",
       "  'laxative',\n",
       "  'machine_read',\n",
       "  '2015.0'],\n",
       " ['(-)-epicatechin',\n",
       "  'interacts with',\n",
       "  'hydroxyflavan',\n",
       "  'machine_read_INF',\n",
       "  '2018.0'],\n",
       " ['(-)-epicatechin',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'augments'},\n",
       "  'myocardium',\n",
       "  'machine_read',\n",
       "  '2013.0'],\n",
       " ['(-)-epicatechin',\n",
       "  'positively regulates',\n",
       "  'peroxisome proliferator-activated receptor gamma',\n",
       "  'machine_read',\n",
       "  '2015.0'],\n",
       " ['(-)-epicatechin',\n",
       "  'interacts with',\n",
       "  'retinoblastoma-like protein 2',\n",
       "  'machine_read_INF',\n",
       "  '2000.0'],\n",
       " ['(-)-epicatechin', 'positively regulates', 'polyphenol', 'machine_read_INF'],\n",
       " ['(-)-epicatechin',\n",
       "  'positively regulates',\n",
       "  'glycogen synthase kinase-3 beta',\n",
       "  'machine_read_INF'],\n",
       " ['(-)-epicatechin',\n",
       "  'positively regulates',\n",
       "  'phospholipid-transporting ATPase IB',\n",
       "  'machine_read_INF'],\n",
       " ['(-)-epicatechin', 'positively regulates', 'caspase-9', 'machine_read_INF'],\n",
       " ['(-)-epicatechin',\n",
       "  'positively regulates',\n",
       "  'brown adipose tissue',\n",
       "  'machine_read_INF'],\n",
       " ['(-)-epicatechin',\n",
       "  'positively regulates',\n",
       "  'proline-rich transmembrane protein 2',\n",
       "  'machine_read_INF'],\n",
       " ['(-)-epicatechin',\n",
       "  'positively regulates',\n",
       "  'thapsigargin',\n",
       "  'machine_read_INF'],\n",
       " ['(-)-epicatechin',\n",
       "  'positively regulates',\n",
       "  'serine/threonine-protein kinase mTOR (human)',\n",
       "  'machine_read_INF'],\n",
       " ['(-)-epicatechin',\n",
       "  'positively regulates',\n",
       "  'N-methyl-D-aspartic acid',\n",
       "  'machine_read_INF'],\n",
       " ['(-)-epicatechin',\n",
       "  'positively regulates',\n",
       "  'zinc fingers and homeoboxes protein 2',\n",
       "  'machine_read_INF'],\n",
       " ['(-)-epicatechin', 'positively regulates', 'glycine', 'machine_read_INF'],\n",
       " ['(-)-epicatechin', 'positively regulates', 'dopamine', 'machine_read_INF'],\n",
       " ['(-)-epicatechin',\n",
       "  'positively regulates',\n",
       "  'glycogen synthase kinase 3',\n",
       "  'machine_read_INF'],\n",
       " ['(-)-epicatechin',\n",
       "  'positively regulates',\n",
       "  'sodium/glucose cotransporter 1 (human)',\n",
       "  'machine_read_INF'],\n",
       " ['(-)-epicatechin',\n",
       "  'positively regulates',\n",
       "  'Glucagon-like peptide 1',\n",
       "  'machine_read_INF'],\n",
       " ['(-)-epicatechin',\n",
       "  'positively regulates',\n",
       "  'programmed cell death protein 1',\n",
       "  'machine_read_INF'],\n",
       " ['(-)-epicatechin',\n",
       "  'positively regulates',\n",
       "  'transient receptor potential cation channel TRPV1',\n",
       "  'machine_read_INF'],\n",
       " ['(-)-epicatechin',\n",
       "  'positively regulates',\n",
       "  'beta-sarcoglycan',\n",
       "  'machine_read_INF'],\n",
       " ['(-)-epicatechin', 'positively regulates', 'rosoxacin', 'machine_read_INF'],\n",
       " ['(-)-epicatechin',\n",
       "  'positively regulates',\n",
       "  'DNA fragmentation factor subunit alpha',\n",
       "  'machine_read_INF'],\n",
       " ['(-)-epicatechin', 'positively regulates', 'caspase-6', 'machine_read_INF'],\n",
       " ['(-)-epicatechin',\n",
       "  'positively regulates',\n",
       "  'Bcl-2-like protein 1',\n",
       "  'machine_read_INF'],\n",
       " ['(-)-epicatechin',\n",
       "  'positively regulates',\n",
       "  'apoptosis regulator Bcl-2',\n",
       "  'machine_read_INF'],\n",
       " ['(-)-epicatechin', 'positively regulates', 'curcumin', 'machine_read_INF'],\n",
       " ['(-)-epicatechin',\n",
       "  'positively regulates',\n",
       "  'sphingosine 1-phosphate',\n",
       "  'machine_read_INF'],\n",
       " ['(-)-epicatechin',\n",
       "  'positively regulates',\n",
       "  'ryanodine receptor 1',\n",
       "  'machine_read_INF']]"
      ]
     },
     "execution_count": 141,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "get_neighbors_filtered(nx_graph,obodict['epicatechin'][0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 144,
   "id": "c7e6fadc-f0c7-4610-bdf6-d00cfbc5433d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[]"
      ]
     },
     "execution_count": 144,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#enantiomer\n",
    "get_neighbors_filtered(nx_graph, obo.CHEBI_76125)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fc26bdc7-f695-4058-a53d-d584f53b409e",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "#### Epigallocatechin"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 182,
   "id": "d6778d02-cdbf-444e-8a05-9d53d8c00f48",
   "metadata": {},
   "outputs": [],
   "source": [
    "edges = nx_graph.get_edge_data(obodict['epigallocatechin'][0], obodict['UGT'][0])\n",
    "if edges:\n",
    "    for item in edges:\n",
    "        print(nodeLabels[str(item)])\n",
    "        print(edges[item])\n",
    "edges = nx_graph.get_edge_data(obodict['epigallocatechin'][0], obodict['UGT'][1])\n",
    "if edges:\n",
    "    for item in edges:\n",
    "        print(nodeLabels[str(item)])\n",
    "        print(edges[item])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 176,
   "id": "6f9e0286-56a0-4c74-96f8-46bd9ccb2bf0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pmid</th>\n",
       "      <th>subject_cui</th>\n",
       "      <th>subject_type</th>\n",
       "      <th>object_cui</th>\n",
       "      <th>object_type</th>\n",
       "      <th>year</th>\n",
       "      <th>subject_name</th>\n",
       "      <th>predicate</th>\n",
       "      <th>object_name</th>\n",
       "      <th>source_text</th>\n",
       "      <th>Unnamed: 10</th>\n",
       "      <th>predicate_obo</th>\n",
       "      <th>subject_obo</th>\n",
       "      <th>object_obo</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>456</th>\n",
       "      <td>19111536</td>\n",
       "      <td>C2348459</td>\n",
       "      <td>orch</td>\n",
       "      <td>C0020517</td>\n",
       "      <td>patf</td>\n",
       "      <td>2009.0</td>\n",
       "      <td>Epigallocatechin</td>\n",
       "      <td>neg_treats</td>\n",
       "      <td>Hypersensitivity</td>\n",
       "      <td>19111536_ascii.txt.tx.1 The apparent dissociat...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CHEBI_42255</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>518</th>\n",
       "      <td>15277314</td>\n",
       "      <td>C2348459</td>\n",
       "      <td>orch</td>\n",
       "      <td>C1519726</td>\n",
       "      <td>moft</td>\n",
       "      <td>2004.0</td>\n",
       "      <td>Epigallocatechin</td>\n",
       "      <td>neg_disrupts</td>\n",
       "      <td>Tyrosine Phosphorylation</td>\n",
       "      <td>15277314_ascii.txt.tx.1 Activation of is contr...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CHEBI_42255</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>857</th>\n",
       "      <td>15729621</td>\n",
       "      <td>C2348459</td>\n",
       "      <td>orch</td>\n",
       "      <td>C1412077</td>\n",
       "      <td>gngm,aapp</td>\n",
       "      <td>2005.0</td>\n",
       "      <td>Epigallocatechin</td>\n",
       "      <td>neg_interacts_with</td>\n",
       "      <td>ABCC2 gene</td>\n",
       "      <td>15729621_ascii.txt.tx.1 Func- tional assays in...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CHEBI_42255</td>\n",
       "      <td>PR_000003558</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1215</th>\n",
       "      <td>28346686</td>\n",
       "      <td>C2348459</td>\n",
       "      <td>orch</td>\n",
       "      <td>C4055506</td>\n",
       "      <td>fndg</td>\n",
       "      <td>2017.0</td>\n",
       "      <td>Epigallocatechin</td>\n",
       "      <td>neg_augments</td>\n",
       "      <td>Accumulation</td>\n",
       "      <td>28346686_ascii.txt.tx.1 In contrast, other stu...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CHEBI_42255</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1502</th>\n",
       "      <td>15277314</td>\n",
       "      <td>C2348459</td>\n",
       "      <td>orch</td>\n",
       "      <td>C1519726</td>\n",
       "      <td>moft</td>\n",
       "      <td>2004.0</td>\n",
       "      <td>Epigallocatechin</td>\n",
       "      <td>neg_disrupts</td>\n",
       "      <td>Tyrosine Phosphorylation</td>\n",
       "      <td>15277314_ascii.txt.tx.1|relation|C1120843|mito...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CHEBI_42255</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1648</th>\n",
       "      <td>28346686</td>\n",
       "      <td>C2348459</td>\n",
       "      <td>orch</td>\n",
       "      <td>C4055506</td>\n",
       "      <td>fndg</td>\n",
       "      <td>2017.0</td>\n",
       "      <td>Epigallocatechin</td>\n",
       "      <td>neg_augments</td>\n",
       "      <td>Accumulation</td>\n",
       "      <td>28346686_ascii.txt.tx.4 Some studies reported ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CHEBI_42255</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          pmid subject_cui subject_type object_cui object_type    year  \\\n",
       "456   19111536    C2348459         orch   C0020517        patf  2009.0   \n",
       "518   15277314    C2348459         orch   C1519726        moft  2004.0   \n",
       "857   15729621    C2348459         orch   C1412077   gngm,aapp  2005.0   \n",
       "1215  28346686    C2348459         orch   C4055506        fndg  2017.0   \n",
       "1502  15277314    C2348459         orch   C1519726        moft  2004.0   \n",
       "1648  28346686    C2348459         orch   C4055506        fndg  2017.0   \n",
       "\n",
       "          subject_name           predicate               object_name  \\\n",
       "456   Epigallocatechin          neg_treats          Hypersensitivity   \n",
       "518   Epigallocatechin        neg_disrupts  Tyrosine Phosphorylation   \n",
       "857   Epigallocatechin  neg_interacts_with                ABCC2 gene   \n",
       "1215  Epigallocatechin        neg_augments              Accumulation   \n",
       "1502  Epigallocatechin        neg_disrupts  Tyrosine Phosphorylation   \n",
       "1648  Epigallocatechin        neg_augments              Accumulation   \n",
       "\n",
       "                                            source_text  Unnamed: 10  \\\n",
       "456   19111536_ascii.txt.tx.1 The apparent dissociat...          NaN   \n",
       "518   15277314_ascii.txt.tx.1 Activation of is contr...          NaN   \n",
       "857   15729621_ascii.txt.tx.1 Func- tional assays in...          NaN   \n",
       "1215  28346686_ascii.txt.tx.1 In contrast, other stu...          NaN   \n",
       "1502  15277314_ascii.txt.tx.1|relation|C1120843|mito...          NaN   \n",
       "1648  28346686_ascii.txt.tx.4 Some studies reported ...          NaN   \n",
       "\n",
       "      predicate_obo  subject_obo    object_obo  \n",
       "456             NaN  CHEBI_42255           NaN  \n",
       "518             NaN  CHEBI_42255           NaN  \n",
       "857             NaN  CHEBI_42255  PR_000003558  \n",
       "1215            NaN  CHEBI_42255           NaN  \n",
       "1502            NaN  CHEBI_42255           NaN  \n",
       "1648            NaN  CHEBI_42255           NaN  "
      ]
     },
     "execution_count": 176,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_neg.loc[df_neg['subject_obo'] == 'CHEBI_42255']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 183,
   "id": "7d96af93-33bf-4063-b88c-e764a04c8f99",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[['(-)-epigallocatechin',\n",
       "  'positively regulates',\n",
       "  'stachyose',\n",
       "  'machine_read',\n",
       "  '2016.0'],\n",
       " ['(-)-epigallocatechin',\n",
       "  'negatively regulates',\n",
       "  'import into cell',\n",
       "  'machine_read',\n",
       "  '2011.0'],\n",
       " ['(-)-epigallocatechin',\n",
       "  'directly negatively regulates activity of',\n",
       "  'potassium atom',\n",
       "  'machine_read',\n",
       "  '2007.0'],\n",
       " ['(-)-epigallocatechin',\n",
       "  'interacts with',\n",
       "  '(-)-epigallocatechin gallate',\n",
       "  'machine_read_INF',\n",
       "  '2013.0'],\n",
       " ['(-)-epigallocatechin',\n",
       "  {'entity_type': 'RELATIONS', 'label': 'affects'},\n",
       "  'blood vessel',\n",
       "  'machine_read',\n",
       "  '2003.0'],\n",
       " ['(-)-epigallocatechin',\n",
       "  'interacts with',\n",
       "  'voltage-gated potassium channel subunit KCNQ5',\n",
       "  'machine_read_INF',\n",
       "  '2021.0'],\n",
       " ['(-)-epigallocatechin',\n",
       "  'interacts with',\n",
       "  'procyanidin B2',\n",
       "  'machine_read_INF',\n",
       "  '2020.0'],\n",
       " ['(-)-epigallocatechin',\n",
       "  'positively regulates',\n",
       "  'catechin',\n",
       "  'machine_read_INF']]"
      ]
     },
     "execution_count": 183,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "get_neighbors_filtered(nx_graph, obodict['epigallocatechin'][0])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f75b9efe-0db5-4a65-b9cd-e0c3c7ded2df",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "#### Catechin"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 188,
   "id": "14ccc4ba-9de3-421d-9e2e-ec75d026c94d",
   "metadata": {},
   "outputs": [],
   "source": [
    "edges = nx_graph.get_edge_data(obodict['catechin'][0], obodict['UGT1A8'][0])\n",
    "if edges:\n",
    "    for item in edges:\n",
    "        print(nodeLabels[str(item)])\n",
    "        print(edges[item])\n",
    "edges = nx_graph.get_edge_data(obodict['catechin'][0], obodict['UGT1A8'][1])\n",
    "if edges:\n",
    "    for item in edges:\n",
    "        print(nodeLabels[str(item)])\n",
    "        print(edges[item])\n",
    "edges = nx_graph.get_edge_data(obodict['catechin'][1], obodict['UGT1A8'][0])\n",
    "if edges:\n",
    "    for item in edges:\n",
    "        print(nodeLabels[str(item)])\n",
    "        print(edges[item])\n",
    "edges = nx_graph.get_edge_data(obodict['catechin'][1], obodict['UGT1A8'][1])\n",
    "if edges:\n",
    "    for item in edges:\n",
    "        print(nodeLabels[str(item)])\n",
    "        print(edges[item])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 190,
   "id": "d5c512dc-8868-4e29-a7ea-a513f3746adc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pmid</th>\n",
       "      <th>subject_cui</th>\n",
       "      <th>subject_type</th>\n",
       "      <th>object_cui</th>\n",
       "      <th>object_type</th>\n",
       "      <th>year</th>\n",
       "      <th>subject_name</th>\n",
       "      <th>predicate</th>\n",
       "      <th>object_name</th>\n",
       "      <th>source_text</th>\n",
       "      <th>Unnamed: 10</th>\n",
       "      <th>predicate_obo</th>\n",
       "      <th>subject_obo</th>\n",
       "      <th>object_obo</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>12659723</td>\n",
       "      <td>C0007404</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>C1420188</td>\n",
       "      <td>gngm,aapp</td>\n",
       "      <td>2003.0</td>\n",
       "      <td>Catechin</td>\n",
       "      <td>neg_interacts_with</td>\n",
       "      <td>SLC35A2 gene</td>\n",
       "      <td>12659723_ascii.txt.tx.2 Lhoste et al./ Food an...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CHEBI_23053</td>\n",
       "      <td>PR_000015090</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>20471964</td>\n",
       "      <td>C0007404</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>C0243144</td>\n",
       "      <td>phsf</td>\n",
       "      <td>2010.0</td>\n",
       "      <td>Catechin</td>\n",
       "      <td>neg_disrupts</td>\n",
       "      <td>Uptake</td>\n",
       "      <td>20471964_ascii.txt.tx.2 These results further ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CHEBI_23053</td>\n",
       "      <td>GO_0098657</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76</th>\n",
       "      <td>15104253</td>\n",
       "      <td>C0007404</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>C0537026</td>\n",
       "      <td>gngm,aapp</td>\n",
       "      <td>2004.0</td>\n",
       "      <td>Catechin</td>\n",
       "      <td>neg_stimulates(spec)</td>\n",
       "      <td>UGT1A1 gene</td>\n",
       "      <td>15104253_ascii.txt.tx.2 Here, we investigated ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CHEBI_23053</td>\n",
       "      <td>PR_000017048</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>88</th>\n",
       "      <td>21776468</td>\n",
       "      <td>C0007404</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>C0166441</td>\n",
       "      <td>aapp,gngm,bacs</td>\n",
       "      <td>2010.0</td>\n",
       "      <td>Catechin</td>\n",
       "      <td>neg_interacts_with</td>\n",
       "      <td>Glucose Transporter Type 4</td>\n",
       "      <td>21776468_ascii.txt.tx.2 This is the first repo...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CHEBI_23053</td>\n",
       "      <td>PR_000013058</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>131</th>\n",
       "      <td>18936213</td>\n",
       "      <td>C0007404</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>C3842459</td>\n",
       "      <td>fndg</td>\n",
       "      <td>2008.0</td>\n",
       "      <td>Catechin</td>\n",
       "      <td>neg_affects</td>\n",
       "      <td>Energetic</td>\n",
       "      <td>18936213_ascii.txt.tx.1 From modulation of foo...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CHEBI_23053</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>133</th>\n",
       "      <td>18936213</td>\n",
       "      <td>C0007404</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>C0014939</td>\n",
       "      <td>horm,orch,phsu</td>\n",
       "      <td>2008.0</td>\n",
       "      <td>Catechin</td>\n",
       "      <td>neg_interacts_with</td>\n",
       "      <td>Estrogens</td>\n",
       "      <td>18936213_ascii.txt.tx.1 In that experimental s...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CHEBI_23053</td>\n",
       "      <td>CHEBI_50114</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>172</th>\n",
       "      <td>15041478</td>\n",
       "      <td>C0007404</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>C0380168</td>\n",
       "      <td>aapp,gngm,bacs</td>\n",
       "      <td>2004.0</td>\n",
       "      <td>Catechin</td>\n",
       "      <td>neg_inhibits</td>\n",
       "      <td>glutathione S-conjugate export pump</td>\n",
       "      <td>15041478_ascii.txt.tx.1 Flavonoids without a C...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CHEBI_23053</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>220</th>\n",
       "      <td>27374085</td>\n",
       "      <td>C0007404</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>C1538577</td>\n",
       "      <td>gngm,aapp</td>\n",
       "      <td>2016.0</td>\n",
       "      <td>Catechin</td>\n",
       "      <td>neg_stimulates</td>\n",
       "      <td>PARP1 gene</td>\n",
       "      <td>27374085_ascii.txt.tx.1 A significant increase...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CHEBI_23053</td>\n",
       "      <td>PR_000012289</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>221</th>\n",
       "      <td>27374085</td>\n",
       "      <td>C0007404</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>C0020517</td>\n",
       "      <td>patf</td>\n",
       "      <td>2016.0</td>\n",
       "      <td>Catechin</td>\n",
       "      <td>neg_affects</td>\n",
       "      <td>Hypersensitivity</td>\n",
       "      <td>27374085_ascii.txt.tx.3 Neuroblastoma cells wi...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CHEBI_23053</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>223</th>\n",
       "      <td>27374085</td>\n",
       "      <td>C0007404</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>C0027819</td>\n",
       "      <td>neop</td>\n",
       "      <td>2016.0</td>\n",
       "      <td>Catechin</td>\n",
       "      <td>neg_treats</td>\n",
       "      <td>Neuroblastoma</td>\n",
       "      <td>27374085_ascii.txt.tx.1 To determine whether C...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CHEBI_23053</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>226</th>\n",
       "      <td>27374085</td>\n",
       "      <td>C0007404</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>C0030551</td>\n",
       "      <td>famg,humn</td>\n",
       "      <td>2016.0</td>\n",
       "      <td>Catechin</td>\n",
       "      <td>neg_treats</td>\n",
       "      <td>parent</td>\n",
       "      <td>27374085_ascii.txt.tx.2 Importantly,  Dextran-...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CHEBI_23053</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>227</th>\n",
       "      <td>27374085</td>\n",
       "      <td>C0007404</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>C0815000</td>\n",
       "      <td>cell</td>\n",
       "      <td>2016.0</td>\n",
       "      <td>Catechin</td>\n",
       "      <td>neg_affects</td>\n",
       "      <td>neuroblastoma cell</td>\n",
       "      <td>27374085_ascii.txt.tx.3 (C) Cisplatin-resistan...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CHEBI_23053</td>\n",
       "      <td>MONDO_0005072</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>228</th>\n",
       "      <td>27374085</td>\n",
       "      <td>C0007404</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>C0242606</td>\n",
       "      <td>comd</td>\n",
       "      <td>2016.0</td>\n",
       "      <td>Catechin</td>\n",
       "      <td>neg_causes</td>\n",
       "      <td>Oxidative Stress</td>\n",
       "      <td>27374085_ascii.txt.tx.1 Taken together these r...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CHEBI_23053</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>229</th>\n",
       "      <td>27374085</td>\n",
       "      <td>C0007404</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>C0879626</td>\n",
       "      <td>patf</td>\n",
       "      <td>2016.0</td>\n",
       "      <td>Catechin</td>\n",
       "      <td>neg_causes</td>\n",
       "      <td>Adverse effects</td>\n",
       "      <td>27374085_ascii.txt.tx.2 Dextran-Catechin 300 u...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CHEBI_23053</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>230</th>\n",
       "      <td>27374085</td>\n",
       "      <td>C0007404</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>C0242606</td>\n",
       "      <td>comd</td>\n",
       "      <td>2016.0</td>\n",
       "      <td>Catechin</td>\n",
       "      <td>neg_causes</td>\n",
       "      <td>Oxidative Stress</td>\n",
       "      <td>27374085_ascii.txt.tx.2 Therefore, Dextran-Cat...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CHEBI_23053</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>236</th>\n",
       "      <td>29466429</td>\n",
       "      <td>C0007404</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>C1420131</td>\n",
       "      <td>gngm,aapp</td>\n",
       "      <td>2018.0</td>\n",
       "      <td>Catechin</td>\n",
       "      <td>neg_inhibits</td>\n",
       "      <td>SLCO1B1 gene</td>\n",
       "      <td>29466429_ascii.txt.tx.1 In addition we found t...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CHEBI_23053</td>\n",
       "      <td>PR_000015223</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>284</th>\n",
       "      <td>16978655</td>\n",
       "      <td>C0007404</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>C0007620</td>\n",
       "      <td>celf</td>\n",
       "      <td>2006.0</td>\n",
       "      <td>Catechin</td>\n",
       "      <td>neg_affects</td>\n",
       "      <td>Cell Survival</td>\n",
       "      <td>16978655_ascii.txt.tx.3 Catechin and epicatech...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CHEBI_23053</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>370</th>\n",
       "      <td>14511674</td>\n",
       "      <td>C0007404</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>C0210995</td>\n",
       "      <td>aapp,gngm,bacs</td>\n",
       "      <td>2003.0</td>\n",
       "      <td>Catechin</td>\n",
       "      <td>neg_interacts_with</td>\n",
       "      <td>mismatch repair protein 1, human</td>\n",
       "      <td>14511674_ascii.txt.tx.1 At 10 lM, ())-epigallo...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CHEBI_23053</td>\n",
       "      <td>PR_P20585</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>371</th>\n",
       "      <td>14511674</td>\n",
       "      <td>C0007404</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>C1412077</td>\n",
       "      <td>gngm,aapp</td>\n",
       "      <td>2003.0</td>\n",
       "      <td>Catechin</td>\n",
       "      <td>neg_interacts_with</td>\n",
       "      <td>ABCC2 gene</td>\n",
       "      <td>14511674_ascii.txt.tx.1 At 10 lM, ())-epigallo...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CHEBI_23053</td>\n",
       "      <td>PR_000003558</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>494</th>\n",
       "      <td>14557796</td>\n",
       "      <td>C0007404</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>C0026879</td>\n",
       "      <td>hops</td>\n",
       "      <td>2003.0</td>\n",
       "      <td>Catechin</td>\n",
       "      <td>neg_coexists_with</td>\n",
       "      <td>Mutagens</td>\n",
       "      <td>14557796_ascii.txt.tx.1 No relationship could ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CHEBI_23053</td>\n",
       "      <td>CHEBI_25435</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>662</th>\n",
       "      <td>16762473</td>\n",
       "      <td>C0007404</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>C0919426</td>\n",
       "      <td>gngm,aapp</td>\n",
       "      <td>2006.0</td>\n",
       "      <td>Catechin</td>\n",
       "      <td>neg_interacts_with</td>\n",
       "      <td>CYP2E1 gene</td>\n",
       "      <td>16762473_ascii.txt.tx.4 However the results we...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CHEBI_23053</td>\n",
       "      <td>PR_000006122</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>687</th>\n",
       "      <td>24225153</td>\n",
       "      <td>C0007404</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>C0596973</td>\n",
       "      <td>chvs</td>\n",
       "      <td>2014.0</td>\n",
       "      <td>Catechin</td>\n",
       "      <td>neg_same_as</td>\n",
       "      <td>monomer</td>\n",
       "      <td>24225153_ascii.txt.tx.1|relation|C0694883|STK1...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CHEBI_23053</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>755</th>\n",
       "      <td>17108060</td>\n",
       "      <td>C0007404</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gngm,aapp</td>\n",
       "      <td>2007.0</td>\n",
       "      <td>Catechin</td>\n",
       "      <td>neg_stimulates</td>\n",
       "      <td>NaN</td>\n",
       "      <td>17108060_ascii.txt.tx.1 Catechins do not induc...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CHEBI_23053</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>760</th>\n",
       "      <td>17108060</td>\n",
       "      <td>C0007404</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>C0537026</td>\n",
       "      <td>gngm,aapp</td>\n",
       "      <td>2007.0</td>\n",
       "      <td>Catechin</td>\n",
       "      <td>neg_stimulates</td>\n",
       "      <td>UGT1A1 gene</td>\n",
       "      <td>17108060_ascii.txt.tx.1 Taken together, our re...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CHEBI_23053</td>\n",
       "      <td>PR_000017048</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>762</th>\n",
       "      <td>17108060</td>\n",
       "      <td>C0007404</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>C0332149</td>\n",
       "      <td>fndg</td>\n",
       "      <td>2007.0</td>\n",
       "      <td>Catechin</td>\n",
       "      <td>neg_affects</td>\n",
       "      <td>Possible</td>\n",
       "      <td>17108060_ascii.txt.tx.1 Third, we used catechi...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CHEBI_23053</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>768</th>\n",
       "      <td>22429924</td>\n",
       "      <td>C0007404</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>C1421339</td>\n",
       "      <td>gngm,aapp</td>\n",
       "      <td>2012.0</td>\n",
       "      <td>Catechin</td>\n",
       "      <td>neg_interacts_with</td>\n",
       "      <td>UGT2B17 gene</td>\n",
       "      <td>22429924_ascii.txt.tx.1 This further supports ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CHEBI_23053</td>\n",
       "      <td>PR_000017062</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>836</th>\n",
       "      <td>29703388</td>\n",
       "      <td>C0007404</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>C0026259</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>2018.0</td>\n",
       "      <td>Catechin</td>\n",
       "      <td>neg_interacts_with</td>\n",
       "      <td>Mitoxantrone</td>\n",
       "      <td>29703388_ascii.txt.tx.1 In contrast, GT catech...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CHEBI_23053</td>\n",
       "      <td>CHEBI_50729</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>872</th>\n",
       "      <td>27518169</td>\n",
       "      <td>C0007404</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>C0010762</td>\n",
       "      <td>aapp,gngm,enzy</td>\n",
       "      <td>2016.0</td>\n",
       "      <td>Catechin</td>\n",
       "      <td>neg_inhibits</td>\n",
       "      <td>Cytochrome P450</td>\n",
       "      <td>27518169_ascii.txt.tx.1 Catechins without the ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CHEBI_23053</td>\n",
       "      <td>CHEBI_38559</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>887</th>\n",
       "      <td>26393568</td>\n",
       "      <td>C0007404</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>C1819360</td>\n",
       "      <td>celf</td>\n",
       "      <td>2015.0</td>\n",
       "      <td>Catechin</td>\n",
       "      <td>neg_affects</td>\n",
       "      <td>fructose import</td>\n",
       "      <td>26393568_ascii.txt.tx.2 Consistent with this r...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CHEBI_23053</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>890</th>\n",
       "      <td>28118673</td>\n",
       "      <td>C0007404</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>C0010762</td>\n",
       "      <td>aapp,gngm,enzy</td>\n",
       "      <td>2017.0</td>\n",
       "      <td>Catechin</td>\n",
       "      <td>neg_inhibits</td>\n",
       "      <td>Cytochrome P450</td>\n",
       "      <td>28118673_ascii.txt.tx.1 EGCG (Wako Pure Chemic...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CHEBI_23053</td>\n",
       "      <td>CHEBI_38559</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>892</th>\n",
       "      <td>28118673</td>\n",
       "      <td>C0007404</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>C0243144</td>\n",
       "      <td>phsf</td>\n",
       "      <td>2017.0</td>\n",
       "      <td>Catechin</td>\n",
       "      <td>neg_disrupts</td>\n",
       "      <td>Uptake</td>\n",
       "      <td>28118673_ascii.txt.tx.2 However, other catechi...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CHEBI_23053</td>\n",
       "      <td>GO_0098657</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>898</th>\n",
       "      <td>17174345</td>\n",
       "      <td>C0007404</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>C0018810</td>\n",
       "      <td>clna</td>\n",
       "      <td>2007.0</td>\n",
       "      <td>Catechin</td>\n",
       "      <td>neg_affects</td>\n",
       "      <td>heart rate</td>\n",
       "      <td>17174345_ascii.txt.tx.1 These mechan- isms cou...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CHEBI_23053</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>937</th>\n",
       "      <td>21081808</td>\n",
       "      <td>C0007404</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>C0007587</td>\n",
       "      <td>celf</td>\n",
       "      <td>2010.0</td>\n",
       "      <td>Catechin</td>\n",
       "      <td>neg_causes</td>\n",
       "      <td>Cell Death</td>\n",
       "      <td>21081808_ascii.txt.tx.1 On  the  other  hand, ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CHEBI_23053</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>938</th>\n",
       "      <td>21081808</td>\n",
       "      <td>C0007404</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>C0007620</td>\n",
       "      <td>celf</td>\n",
       "      <td>2010.0</td>\n",
       "      <td>Catechin</td>\n",
       "      <td>neg_causes</td>\n",
       "      <td>Cell Survival</td>\n",
       "      <td>21081808_ascii.txt.tx.1 On  the  other  hand, ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CHEBI_23053</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1010</th>\n",
       "      <td>26193264</td>\n",
       "      <td>C0007404</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>C1159527</td>\n",
       "      <td>celf</td>\n",
       "      <td>2015.0</td>\n",
       "      <td>Catechin</td>\n",
       "      <td>neg_augments</td>\n",
       "      <td>glucose uptake</td>\n",
       "      <td>26193264_ascii.txt.tx.1 Conversely, 3-O-myrist...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CHEBI_23053</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1012</th>\n",
       "      <td>26193264</td>\n",
       "      <td>C0007404</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>C1159527</td>\n",
       "      <td>celf</td>\n",
       "      <td>2015.0</td>\n",
       "      <td>Catechin</td>\n",
       "      <td>neg_augments</td>\n",
       "      <td>glucose uptake</td>\n",
       "      <td>26193264_ascii.txt.tx.3 Mol.Sci.2015, 16  is  ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CHEBI_23053</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1073</th>\n",
       "      <td>3624765</td>\n",
       "      <td>C0007404</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>C0025519</td>\n",
       "      <td>orgf</td>\n",
       "      <td>1987.0</td>\n",
       "      <td>Catechin</td>\n",
       "      <td>neg_affects</td>\n",
       "      <td>Metabolism</td>\n",
       "      <td>3624765_ascii.txt.tx.2 This may explain the ap...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CHEBI_23053</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1090</th>\n",
       "      <td>29368187</td>\n",
       "      <td>C0007404</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>C4699564</td>\n",
       "      <td>fndg</td>\n",
       "      <td>2018.0</td>\n",
       "      <td>Catechin</td>\n",
       "      <td>neg_causes</td>\n",
       "      <td>Significant activity</td>\n",
       "      <td>29368187_ascii.txt.tx.1 In conclusion, co-admi...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CHEBI_23053</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1106</th>\n",
       "      <td>30022812</td>\n",
       "      <td>C0007404</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>C0012265</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>2018.0</td>\n",
       "      <td>Catechin</td>\n",
       "      <td>neg_interacts_with</td>\n",
       "      <td>Digoxin</td>\n",
       "      <td>30022812_ascii.txt.tx.4 Also, green tea produc...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CHEBI_23053</td>\n",
       "      <td>CHEBI_145795</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1153</th>\n",
       "      <td>11064004</td>\n",
       "      <td>C0007404</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>C0919425</td>\n",
       "      <td>gngm,aapp</td>\n",
       "      <td>2000.0</td>\n",
       "      <td>Catechin</td>\n",
       "      <td>neg_stimulates</td>\n",
       "      <td>CYP1A1 gene</td>\n",
       "      <td>11064004_ascii.txt.tx.1 :Chemico-Biological In...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CHEBI_23053</td>\n",
       "      <td>PR_000006101</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1209</th>\n",
       "      <td>27782875</td>\n",
       "      <td>C0007404</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>C0010762</td>\n",
       "      <td>aapp,gngm,enzy</td>\n",
       "      <td>2016.0</td>\n",
       "      <td>Catechin</td>\n",
       "      <td>neg_interacts_with</td>\n",
       "      <td>Cytochrome P450</td>\n",
       "      <td>27782875_ascii.txt.tx.1 Although catechins are...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CHEBI_23053</td>\n",
       "      <td>CHEBI_38559</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1443</th>\n",
       "      <td>15041478</td>\n",
       "      <td>C0007404</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>C0380168</td>\n",
       "      <td>aapp,gngm,bacs</td>\n",
       "      <td>2004.0</td>\n",
       "      <td>Catechin</td>\n",
       "      <td>neg_inhibits</td>\n",
       "      <td>glutathione S-conjugate export pump</td>\n",
       "      <td>15041478_ascii.txt.tx.3 Flavonoids without a C...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CHEBI_23053</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1479</th>\n",
       "      <td>14511674</td>\n",
       "      <td>C0007404</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>C0210995</td>\n",
       "      <td>aapp,gngm,bacs</td>\n",
       "      <td>2003.0</td>\n",
       "      <td>Catechin</td>\n",
       "      <td>neg_interacts_with</td>\n",
       "      <td>mismatch repair protein 1, human</td>\n",
       "      <td>14511674_ascii.txt.tx.1 At 10 lM, ())-epigallo...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CHEBI_23053</td>\n",
       "      <td>PR_P20585</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1480</th>\n",
       "      <td>14511674</td>\n",
       "      <td>C0007404</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>C1412077</td>\n",
       "      <td>gngm,aapp</td>\n",
       "      <td>2003.0</td>\n",
       "      <td>Catechin</td>\n",
       "      <td>neg_interacts_with</td>\n",
       "      <td>ABCC2 gene</td>\n",
       "      <td>14511674_ascii.txt.tx.1 At 10 lM, ())-epigallo...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CHEBI_23053</td>\n",
       "      <td>PR_000003558</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1561</th>\n",
       "      <td>17108060</td>\n",
       "      <td>C0007404</td>\n",
       "      <td>orch,phsu</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gngm,aapp</td>\n",
       "      <td>2007.0</td>\n",
       "      <td>Catechin</td>\n",
       "      <td>neg_stimulates</td>\n",
       "      <td>NaN</td>\n",
       "      <td>17108060_ascii.txt.tx.3 Catechins do not induc...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CHEBI_23053</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          pmid subject_cui subject_type object_cui     object_type    year  \\\n",
       "1     12659723    C0007404    orch,phsu   C1420188       gngm,aapp  2003.0   \n",
       "17    20471964    C0007404    orch,phsu   C0243144            phsf  2010.0   \n",
       "76    15104253    C0007404    orch,phsu   C0537026       gngm,aapp  2004.0   \n",
       "88    21776468    C0007404    orch,phsu   C0166441  aapp,gngm,bacs  2010.0   \n",
       "131   18936213    C0007404    orch,phsu   C3842459            fndg  2008.0   \n",
       "133   18936213    C0007404    orch,phsu   C0014939  horm,orch,phsu  2008.0   \n",
       "172   15041478    C0007404    orch,phsu   C0380168  aapp,gngm,bacs  2004.0   \n",
       "220   27374085    C0007404    orch,phsu   C1538577       gngm,aapp  2016.0   \n",
       "221   27374085    C0007404    orch,phsu   C0020517            patf  2016.0   \n",
       "223   27374085    C0007404    orch,phsu   C0027819            neop  2016.0   \n",
       "226   27374085    C0007404    orch,phsu   C0030551       famg,humn  2016.0   \n",
       "227   27374085    C0007404    orch,phsu   C0815000            cell  2016.0   \n",
       "228   27374085    C0007404    orch,phsu   C0242606            comd  2016.0   \n",
       "229   27374085    C0007404    orch,phsu   C0879626            patf  2016.0   \n",
       "230   27374085    C0007404    orch,phsu   C0242606            comd  2016.0   \n",
       "236   29466429    C0007404    orch,phsu   C1420131       gngm,aapp  2018.0   \n",
       "284   16978655    C0007404    orch,phsu   C0007620            celf  2006.0   \n",
       "370   14511674    C0007404    orch,phsu   C0210995  aapp,gngm,bacs  2003.0   \n",
       "371   14511674    C0007404    orch,phsu   C1412077       gngm,aapp  2003.0   \n",
       "494   14557796    C0007404    orch,phsu   C0026879            hops  2003.0   \n",
       "662   16762473    C0007404    orch,phsu   C0919426       gngm,aapp  2006.0   \n",
       "687   24225153    C0007404    orch,phsu   C0596973            chvs  2014.0   \n",
       "755   17108060    C0007404    orch,phsu        NaN       gngm,aapp  2007.0   \n",
       "760   17108060    C0007404    orch,phsu   C0537026       gngm,aapp  2007.0   \n",
       "762   17108060    C0007404    orch,phsu   C0332149            fndg  2007.0   \n",
       "768   22429924    C0007404    orch,phsu   C1421339       gngm,aapp  2012.0   \n",
       "836   29703388    C0007404    orch,phsu   C0026259       orch,phsu  2018.0   \n",
       "872   27518169    C0007404    orch,phsu   C0010762  aapp,gngm,enzy  2016.0   \n",
       "887   26393568    C0007404    orch,phsu   C1819360            celf  2015.0   \n",
       "890   28118673    C0007404    orch,phsu   C0010762  aapp,gngm,enzy  2017.0   \n",
       "892   28118673    C0007404    orch,phsu   C0243144            phsf  2017.0   \n",
       "898   17174345    C0007404    orch,phsu   C0018810            clna  2007.0   \n",
       "937   21081808    C0007404    orch,phsu   C0007587            celf  2010.0   \n",
       "938   21081808    C0007404    orch,phsu   C0007620            celf  2010.0   \n",
       "1010  26193264    C0007404    orch,phsu   C1159527            celf  2015.0   \n",
       "1012  26193264    C0007404    orch,phsu   C1159527            celf  2015.0   \n",
       "1073   3624765    C0007404    orch,phsu   C0025519            orgf  1987.0   \n",
       "1090  29368187    C0007404    orch,phsu   C4699564            fndg  2018.0   \n",
       "1106  30022812    C0007404    orch,phsu   C0012265       orch,phsu  2018.0   \n",
       "1153  11064004    C0007404    orch,phsu   C0919425       gngm,aapp  2000.0   \n",
       "1209  27782875    C0007404    orch,phsu   C0010762  aapp,gngm,enzy  2016.0   \n",
       "1443  15041478    C0007404    orch,phsu   C0380168  aapp,gngm,bacs  2004.0   \n",
       "1479  14511674    C0007404    orch,phsu   C0210995  aapp,gngm,bacs  2003.0   \n",
       "1480  14511674    C0007404    orch,phsu   C1412077       gngm,aapp  2003.0   \n",
       "1561  17108060    C0007404    orch,phsu        NaN       gngm,aapp  2007.0   \n",
       "\n",
       "     subject_name             predicate                          object_name  \\\n",
       "1        Catechin    neg_interacts_with                         SLC35A2 gene   \n",
       "17       Catechin          neg_disrupts                               Uptake   \n",
       "76       Catechin  neg_stimulates(spec)                          UGT1A1 gene   \n",
       "88       Catechin    neg_interacts_with           Glucose Transporter Type 4   \n",
       "131      Catechin           neg_affects                            Energetic   \n",
       "133      Catechin    neg_interacts_with                            Estrogens   \n",
       "172      Catechin          neg_inhibits  glutathione S-conjugate export pump   \n",
       "220      Catechin        neg_stimulates                           PARP1 gene   \n",
       "221      Catechin           neg_affects                     Hypersensitivity   \n",
       "223      Catechin            neg_treats                        Neuroblastoma   \n",
       "226      Catechin            neg_treats                               parent   \n",
       "227      Catechin           neg_affects                   neuroblastoma cell   \n",
       "228      Catechin            neg_causes                     Oxidative Stress   \n",
       "229      Catechin            neg_causes                      Adverse effects   \n",
       "230      Catechin            neg_causes                     Oxidative Stress   \n",
       "236      Catechin          neg_inhibits                         SLCO1B1 gene   \n",
       "284      Catechin           neg_affects                        Cell Survival   \n",
       "370      Catechin    neg_interacts_with     mismatch repair protein 1, human   \n",
       "371      Catechin    neg_interacts_with                           ABCC2 gene   \n",
       "494      Catechin     neg_coexists_with                             Mutagens   \n",
       "662      Catechin    neg_interacts_with                          CYP2E1 gene   \n",
       "687      Catechin           neg_same_as                              monomer   \n",
       "755      Catechin        neg_stimulates                                  NaN   \n",
       "760      Catechin        neg_stimulates                          UGT1A1 gene   \n",
       "762      Catechin           neg_affects                             Possible   \n",
       "768      Catechin    neg_interacts_with                         UGT2B17 gene   \n",
       "836      Catechin    neg_interacts_with                         Mitoxantrone   \n",
       "872      Catechin          neg_inhibits                      Cytochrome P450   \n",
       "887      Catechin           neg_affects                      fructose import   \n",
       "890      Catechin          neg_inhibits                      Cytochrome P450   \n",
       "892      Catechin          neg_disrupts                               Uptake   \n",
       "898      Catechin           neg_affects                           heart rate   \n",
       "937      Catechin            neg_causes                           Cell Death   \n",
       "938      Catechin            neg_causes                        Cell Survival   \n",
       "1010     Catechin          neg_augments                       glucose uptake   \n",
       "1012     Catechin          neg_augments                       glucose uptake   \n",
       "1073     Catechin           neg_affects                           Metabolism   \n",
       "1090     Catechin            neg_causes                 Significant activity   \n",
       "1106     Catechin    neg_interacts_with                              Digoxin   \n",
       "1153     Catechin        neg_stimulates                          CYP1A1 gene   \n",
       "1209     Catechin    neg_interacts_with                      Cytochrome P450   \n",
       "1443     Catechin          neg_inhibits  glutathione S-conjugate export pump   \n",
       "1479     Catechin    neg_interacts_with     mismatch repair protein 1, human   \n",
       "1480     Catechin    neg_interacts_with                           ABCC2 gene   \n",
       "1561     Catechin        neg_stimulates                                  NaN   \n",
       "\n",
       "                                            source_text  Unnamed: 10  \\\n",
       "1     12659723_ascii.txt.tx.2 Lhoste et al./ Food an...          NaN   \n",
       "17    20471964_ascii.txt.tx.2 These results further ...          NaN   \n",
       "76    15104253_ascii.txt.tx.2 Here, we investigated ...          NaN   \n",
       "88    21776468_ascii.txt.tx.2 This is the first repo...          NaN   \n",
       "131   18936213_ascii.txt.tx.1 From modulation of foo...          NaN   \n",
       "133   18936213_ascii.txt.tx.1 In that experimental s...          NaN   \n",
       "172   15041478_ascii.txt.tx.1 Flavonoids without a C...          NaN   \n",
       "220   27374085_ascii.txt.tx.1 A significant increase...          NaN   \n",
       "221   27374085_ascii.txt.tx.3 Neuroblastoma cells wi...          NaN   \n",
       "223   27374085_ascii.txt.tx.1 To determine whether C...          NaN   \n",
       "226   27374085_ascii.txt.tx.2 Importantly,  Dextran-...          NaN   \n",
       "227   27374085_ascii.txt.tx.3 (C) Cisplatin-resistan...          NaN   \n",
       "228   27374085_ascii.txt.tx.1 Taken together these r...          NaN   \n",
       "229   27374085_ascii.txt.tx.2 Dextran-Catechin 300 u...          NaN   \n",
       "230   27374085_ascii.txt.tx.2 Therefore, Dextran-Cat...          NaN   \n",
       "236   29466429_ascii.txt.tx.1 In addition we found t...          NaN   \n",
       "284   16978655_ascii.txt.tx.3 Catechin and epicatech...          NaN   \n",
       "370   14511674_ascii.txt.tx.1 At 10 lM, ())-epigallo...          NaN   \n",
       "371   14511674_ascii.txt.tx.1 At 10 lM, ())-epigallo...          NaN   \n",
       "494   14557796_ascii.txt.tx.1 No relationship could ...          NaN   \n",
       "662   16762473_ascii.txt.tx.4 However the results we...          NaN   \n",
       "687   24225153_ascii.txt.tx.1|relation|C0694883|STK1...          NaN   \n",
       "755   17108060_ascii.txt.tx.1 Catechins do not induc...          NaN   \n",
       "760   17108060_ascii.txt.tx.1 Taken together, our re...          NaN   \n",
       "762   17108060_ascii.txt.tx.1 Third, we used catechi...          NaN   \n",
       "768   22429924_ascii.txt.tx.1 This further supports ...          NaN   \n",
       "836   29703388_ascii.txt.tx.1 In contrast, GT catech...          NaN   \n",
       "872   27518169_ascii.txt.tx.1 Catechins without the ...          NaN   \n",
       "887   26393568_ascii.txt.tx.2 Consistent with this r...          NaN   \n",
       "890   28118673_ascii.txt.tx.1 EGCG (Wako Pure Chemic...          NaN   \n",
       "892   28118673_ascii.txt.tx.2 However, other catechi...          NaN   \n",
       "898   17174345_ascii.txt.tx.1 These mechan- isms cou...          NaN   \n",
       "937   21081808_ascii.txt.tx.1 On  the  other  hand, ...          NaN   \n",
       "938   21081808_ascii.txt.tx.1 On  the  other  hand, ...          NaN   \n",
       "1010  26193264_ascii.txt.tx.1 Conversely, 3-O-myrist...          NaN   \n",
       "1012  26193264_ascii.txt.tx.3 Mol.Sci.2015, 16  is  ...          NaN   \n",
       "1073  3624765_ascii.txt.tx.2 This may explain the ap...          NaN   \n",
       "1090  29368187_ascii.txt.tx.1 In conclusion, co-admi...          NaN   \n",
       "1106  30022812_ascii.txt.tx.4 Also, green tea produc...          NaN   \n",
       "1153  11064004_ascii.txt.tx.1 :Chemico-Biological In...          NaN   \n",
       "1209  27782875_ascii.txt.tx.1 Although catechins are...          NaN   \n",
       "1443  15041478_ascii.txt.tx.3 Flavonoids without a C...          NaN   \n",
       "1479  14511674_ascii.txt.tx.1 At 10 lM, ())-epigallo...          NaN   \n",
       "1480  14511674_ascii.txt.tx.1 At 10 lM, ())-epigallo...          NaN   \n",
       "1561  17108060_ascii.txt.tx.3 Catechins do not induc...          NaN   \n",
       "\n",
       "      predicate_obo  subject_obo     object_obo  \n",
       "1               NaN  CHEBI_23053   PR_000015090  \n",
       "17              NaN  CHEBI_23053     GO_0098657  \n",
       "76              NaN  CHEBI_23053   PR_000017048  \n",
       "88              NaN  CHEBI_23053   PR_000013058  \n",
       "131             NaN  CHEBI_23053            NaN  \n",
       "133             NaN  CHEBI_23053    CHEBI_50114  \n",
       "172             NaN  CHEBI_23053            NaN  \n",
       "220             NaN  CHEBI_23053   PR_000012289  \n",
       "221             NaN  CHEBI_23053            NaN  \n",
       "223             NaN  CHEBI_23053            NaN  \n",
       "226             NaN  CHEBI_23053            NaN  \n",
       "227             NaN  CHEBI_23053  MONDO_0005072  \n",
       "228             NaN  CHEBI_23053            NaN  \n",
       "229             NaN  CHEBI_23053            NaN  \n",
       "230             NaN  CHEBI_23053            NaN  \n",
       "236             NaN  CHEBI_23053   PR_000015223  \n",
       "284             NaN  CHEBI_23053            NaN  \n",
       "370             NaN  CHEBI_23053      PR_P20585  \n",
       "371             NaN  CHEBI_23053   PR_000003558  \n",
       "494             NaN  CHEBI_23053    CHEBI_25435  \n",
       "662             NaN  CHEBI_23053   PR_000006122  \n",
       "687             NaN  CHEBI_23053            NaN  \n",
       "755             NaN  CHEBI_23053            NaN  \n",
       "760             NaN  CHEBI_23053   PR_000017048  \n",
       "762             NaN  CHEBI_23053            NaN  \n",
       "768             NaN  CHEBI_23053   PR_000017062  \n",
       "836             NaN  CHEBI_23053    CHEBI_50729  \n",
       "872             NaN  CHEBI_23053    CHEBI_38559  \n",
       "887             NaN  CHEBI_23053            NaN  \n",
       "890             NaN  CHEBI_23053    CHEBI_38559  \n",
       "892             NaN  CHEBI_23053     GO_0098657  \n",
       "898             NaN  CHEBI_23053            NaN  \n",
       "937             NaN  CHEBI_23053            NaN  \n",
       "938             NaN  CHEBI_23053            NaN  \n",
       "1010            NaN  CHEBI_23053            NaN  \n",
       "1012            NaN  CHEBI_23053            NaN  \n",
       "1073            NaN  CHEBI_23053            NaN  \n",
       "1090            NaN  CHEBI_23053            NaN  \n",
       "1106            NaN  CHEBI_23053   CHEBI_145795  \n",
       "1153            NaN  CHEBI_23053   PR_000006101  \n",
       "1209            NaN  CHEBI_23053    CHEBI_38559  \n",
       "1443            NaN  CHEBI_23053            NaN  \n",
       "1479            NaN  CHEBI_23053      PR_P20585  \n",
       "1480            NaN  CHEBI_23053   PR_000003558  \n",
       "1561            NaN  CHEBI_23053            NaN  "
      ]
     },
     "execution_count": 190,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_neg.loc[df_neg['subject_obo'] == 'CHEBI_23053']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 196,
   "id": "1e778007-adb2-4b1c-becf-bf4a66be1b06",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1023"
      ]
     },
     "execution_count": 196,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(get_neighbors_filtered(nx_graph, obodict['catechin'][0]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 197,
   "id": "efa3c257-1916-40b3-88ca-581fcbbac5e8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "502"
      ]
     },
     "execution_count": 197,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(get_neighbors_filtered(nx_graph, obodict['catechin'][1]))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7b4b1fae-f17e-48d2-a798-ceea204fdae9",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "#### Gallocatechin"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 201,
   "id": "eec88ae4-f521-473d-90a2-30eac7edadad",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[['gallocatechin',\n",
       "  'directly negatively regulates activity of',\n",
       "  'Tea',\n",
       "  'machine_read',\n",
       "  '2014'],\n",
       " ['gallocatechin',\n",
       "  'existence overlaps',\n",
       "  'flavonoids',\n",
       "  'machine_read',\n",
       "  '2016.0'],\n",
       " ['gallocatechin', 'interacts with', 'chelator', 'machine_read_INF', '2011.0'],\n",
       " ['gallocatechin',\n",
       "  'positively regulates',\n",
       "  'very-long-chain enoyl-CoA reductase',\n",
       "  'machine_read',\n",
       "  '2004.0'],\n",
       " ['gallocatechin', 'interacts with', 'interleukin-6 (human)'],\n",
       " ['gallocatechin', 'subClassOf', 'chemical entity'],\n",
       " ['gallocatechin', 'has role', 'metabolite'],\n",
       " ['gallocatechin', 'interacts with', 'transcription factor p65 (human)'],\n",
       " ['gallocatechin', 'subClassOf', 'catechin'],\n",
       " ['gallocatechin', 'subClassOf', \"flavan-3,3',4',5,5',7-hexol\"],\n",
       " ['gallocatechin',\n",
       "  'interacts with',\n",
       "  'glutathione reductase, mitochondrial (human)'],\n",
       " ['gallocatechin', 'interacts with', 'Toll-like receptor 4 (human)'],\n",
       " ['gallocatechin', 'interacts with', 'myeloperoxidase (human)'],\n",
       " ['gallocatechin', 'interacts with', 'bile acid receptor (human)'],\n",
       " ['gallocatechin', 'interacts with', 'tumor necrosis factor (human)']]"
      ]
     },
     "execution_count": 201,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "get_neighbors(nx_graph, obodict['gallocatechin'][1])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 207,
   "id": "5abdf508-e989-47cc-898e-9e1292a53b53",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pmid</th>\n",
       "      <th>subject_cui</th>\n",
       "      <th>subject_type</th>\n",
       "      <th>object_cui</th>\n",
       "      <th>object_type</th>\n",
       "      <th>year</th>\n",
       "      <th>subject_name</th>\n",
       "      <th>predicate</th>\n",
       "      <th>object_name</th>\n",
       "      <th>source_text</th>\n",
       "      <th>Unnamed: 10</th>\n",
       "      <th>predicate_obo</th>\n",
       "      <th>subject_obo</th>\n",
       "      <th>object_obo</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [pmid, subject_cui, subject_type, object_cui, object_type, year, subject_name, predicate, object_name, source_text, Unnamed: 10, predicate_obo, subject_obo, object_obo]\n",
       "Index: []"
      ]
     },
     "execution_count": 207,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_neg.loc[df_neg['object_name'] == 'gallocatechin']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fac8656f-1b03-4718-a4d1-17724379d2ec",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "558ead08-f9b4-4b73-a545-1dd65fad6d47",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f76fd32e-af16-47b5-ba64-b09cfdbae7ae",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "36cdfb9c-729b-42c8-8087-db49bef29218",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
